{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pumped Catchment eDNA metabarcoding data processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I will use [metaBEAT](https://github.com/HullUni-bioinformatics/metaBEAT), a tool tailored towards reproducible and efficient analyses of metabarcoding data that was developed by Dr. Christoph Hahn (University of Graz) for the EvoHull group at University of Hull. The pipeline is still under active development and will likely be extended further in the future. The pipeline is available in a Docker container with all necessary dependencies. The Docker image builds on [ReproPhylo](https://github.com/HullUni-bioinformatics/ReproPhylo).\n",
    "\n",
    "The metaBEAT tool is designed for complete bioinformatic analysis from raw data, and performs (optionally) de-multiplexing, quality filtering, chimera detection, clustering, and taxononomic assignment (outputs in `.biom` and `.tsv` formats). It currently supports BLAST, Kraken and phylogenetic placement (pplacer)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data input"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook will perform basic processing (trimming, merging, chimera removal, clustering and taxonomic assignment) of the metabarcoding data.\n",
    "\n",
    "Minimum input for an analysis is a set of query sequences in one or several files (a number of file formats are accepted, e.g. `.fasta`, `.fastq`). These will be run through the pipeline sequentially.\n",
    "\n",
    "Information on the nature and location of the query sequence files must be provided in a separate tab-delimited text file via the -Q flags.\n",
    "\n",
    "Each line in this text file should look as follows: unique sample_ID, format, file1, file2\n",
    "\n",
    "The required text files can be generated in any text editor. So theoretically, nano could be used in the terminal to construct the text file. For reproducibility and ease, a simple program can be used to generate the required file.\n",
    "\n",
    "In the cell below, it is produced using a simple python script. The script will list all files in the location to which you downloaded your Illumina data (specified via the 'datadir' variable). It assumes that there is a file ending in `_R1.fastq` for each sample. For each such file, it will extract the sample name from the filename and format the required line for the text file accordingly. The resulting file is called `Querymap.txt` (specified in the 'to' variable)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/1-trimming\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘1-trimming’: File exists\r\n"
     ]
    }
   ],
   "source": [
    "!mkdir 1-trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/1-trimming\n"
     ]
    }
   ],
   "source": [
    "cd 1-trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24-blank.R1.fastq.gz\r\n",
      "24-blank.R2.fastq.gz\r\n",
      "24-S1.R1.fastq.gz\r\n",
      "24-S1.R2.fastq.gz\r\n",
      "24-S2.R1.fastq.gz\r\n",
      "24-S2.R2.fastq.gz\r\n",
      "24-S3.R1.fastq.gz\r\n",
      "24-S3.R2.fastq.gz\r\n",
      "24-S4.R1.fastq.gz\r\n",
      "24-S4.R2.fastq.gz\r\n",
      "24-S5.R1.fastq.gz\r\n",
      "24-S5.R2.fastq.gz\r\n",
      "25-blank.R1.fastq.gz\r\n",
      "25-blank.R2.fastq.gz\r\n",
      "25-S1.R1.fastq.gz\r\n",
      "25-S1.R2.fastq.gz\r\n",
      "25-S2.R1.fastq.gz\r\n",
      "25-S2.R2.fastq.gz\r\n",
      "25-S3.R1.fastq.gz\r\n",
      "25-S3.R2.fastq.gz\r\n",
      "25-S4.R1.fastq.gz\r\n",
      "25-S4.R2.fastq.gz\r\n",
      "25-S5.R1.fastq.gz\r\n",
      "25-S5.R2.fastq.gz\r\n",
      "26-blank.R1.fastq.gz\r\n",
      "26-blank.R2.fastq.gz\r\n",
      "26-S1.R1.fastq.gz\r\n",
      "26-S1.R2.fastq.gz\r\n",
      "26-S2.R1.fastq.gz\r\n",
      "26-S2.R2.fastq.gz\r\n",
      "26-S3.R1.fastq.gz\r\n",
      "26-S3.R2.fastq.gz\r\n",
      "26-S4.R1.fastq.gz\r\n",
      "26-S4.R2.fastq.gz\r\n",
      "26-S5.R1.fastq.gz\r\n",
      "26-S5.R2.fastq.gz\r\n",
      "BL3-blank.R1.fastq.gz\r\n",
      "BL3-blank.R2.fastq.gz\r\n",
      "BL3-S1.R1.fastq.gz\r\n",
      "BL3-S1.R2.fastq.gz\r\n",
      "BL3-S2.R1.fastq.gz\r\n",
      "BL3-S2.R2.fastq.gz\r\n",
      "BL3-S3.R1.fastq.gz\r\n",
      "BL3-S3.R2.fastq.gz\r\n",
      "BL3-S4.R1.fastq.gz\r\n",
      "BL3-S4.R2.fastq.gz\r\n",
      "BL3-S5.R1.fastq.gz\r\n",
      "BL3-S5.R2.fastq.gz\r\n",
      "BL4-blank.R1.fastq.gz\r\n",
      "BL4-blank.R2.fastq.gz\r\n",
      "BL4-S1.R1.fastq.gz\r\n",
      "BL4-S1.R2.fastq.gz\r\n",
      "BL4-S2.R1.fastq.gz\r\n",
      "BL4-S2.R2.fastq.gz\r\n",
      "BL4-S3.R1.fastq.gz\r\n",
      "BL4-S3.R2.fastq.gz\r\n",
      "BL4-S4.R1.fastq.gz\r\n",
      "BL4-S4.R2.fastq.gz\r\n",
      "BL4-S5.R1.fastq.gz\r\n",
      "BL4-S5.R2.fastq.gz\r\n",
      "C-01.R1.fastq.gz\r\n",
      "C-01.R2.fastq.gz\r\n",
      "C-02.R1.fastq.gz\r\n",
      "C-02.R2.fastq.gz\r\n",
      "C-03.R1.fastq.gz\r\n",
      "C-03.R2.fastq.gz\r\n",
      "C-04.R1.fastq.gz\r\n",
      "C-04.R2.fastq.gz\r\n",
      "C-05.R1.fastq.gz\r\n",
      "C-05.R2.fastq.gz\r\n",
      "D-01.R1.fastq.gz\r\n",
      "D-01.R2.fastq.gz\r\n",
      "D-02.R1.fastq.gz\r\n",
      "D-02.R2.fastq.gz\r\n",
      "D-03.R1.fastq.gz\r\n",
      "D-03.R2.fastq.gz\r\n",
      "D-04.R1.fastq.gz\r\n",
      "D-04.R2.fastq.gz\r\n",
      "D-05.R1.fastq.gz\r\n",
      "D-05.R2.fastq.gz\r\n",
      "D-B1.R1.fastq.gz\r\n",
      "D-B1.R2.fastq.gz\r\n",
      "LB-01.R1.fastq.gz\r\n",
      "LB-01.R2.fastq.gz\r\n",
      "LB-02.R1.fastq.gz\r\n",
      "LB-02.R2.fastq.gz\r\n",
      "LB-03.R1.fastq.gz\r\n",
      "LB-03.R2.fastq.gz\r\n",
      "LB-04.R1.fastq.gz\r\n",
      "LB-04.R2.fastq.gz\r\n",
      "LB-05.R1.fastq.gz\r\n",
      "LB-05.R2.fastq.gz\r\n",
      "LB-06.R1.fastq.gz\r\n",
      "LB-06.R2.fastq.gz\r\n",
      "LB-07.R1.fastq.gz\r\n",
      "LB-07.R2.fastq.gz\r\n",
      "LB-08.R1.fastq.gz\r\n",
      "LB-08.R2.fastq.gz\r\n",
      "LB-09.R1.fastq.gz\r\n",
      "LB-09.R2.fastq.gz\r\n",
      "LB-10.R1.fastq.gz\r\n",
      "LB-10.R2.fastq.gz\r\n",
      "LB-B1.R1.fastq.gz\r\n",
      "LB-B1.R2.fastq.gz\r\n",
      "LH-01.R1.fastq.gz\r\n",
      "LH-01.R2.fastq.gz\r\n",
      "LH-02.R1.fastq.gz\r\n",
      "LH-02.R2.fastq.gz\r\n",
      "LH-03.R1.fastq.gz\r\n",
      "LH-03.R2.fastq.gz\r\n",
      "LH-04.R1.fastq.gz\r\n",
      "LH-04.R2.fastq.gz\r\n",
      "LH-05.R1.fastq.gz\r\n",
      "LH-05.R2.fastq.gz\r\n",
      "N-01.R1.fastq.gz\r\n",
      "N-01.R2.fastq.gz\r\n",
      "N-02.R1.fastq.gz\r\n",
      "N-02.R2.fastq.gz\r\n",
      "N-03.R1.fastq.gz\r\n",
      "N-03.R2.fastq.gz\r\n",
      "N-04.R1.fastq.gz\r\n",
      "N-04.R2.fastq.gz\r\n",
      "N-05.R1.fastq.gz\r\n",
      "N-05.R2.fastq.gz\r\n",
      "N+C-B1.R1.fastq.gz\r\n",
      "N+C-B1.R2.fastq.gz\r\n",
      "neg1.R1.fastq.gz\r\n",
      "neg1.R2.fastq.gz\r\n",
      "neg2.R1.fastq.gz\r\n",
      "neg2.R2.fastq.gz\r\n",
      "neg3.R1.fastq.gz\r\n",
      "neg3.R2.fastq.gz\r\n",
      "neg4.R1.fastq.gz\r\n",
      "neg4.R2.fastq.gz\r\n",
      "neg5.R1.fastq.gz\r\n",
      "neg5.R2.fastq.gz\r\n",
      "PH-01.R1.fastq.gz\r\n",
      "PH-01.R2.fastq.gz\r\n",
      "PH-02.R1.fastq.gz\r\n",
      "PH-02.R2.fastq.gz\r\n",
      "PH-03.R1.fastq.gz\r\n",
      "PH-03.R2.fastq.gz\r\n",
      "PH-04.R1.fastq.gz\r\n",
      "PH-04.R2.fastq.gz\r\n",
      "PH-05.R1.fastq.gz\r\n",
      "PH-05.R2.fastq.gz\r\n",
      "PH-06.R1.fastq.gz\r\n",
      "PH-06.R2.fastq.gz\r\n",
      "PH-07.R1.fastq.gz\r\n",
      "PH-07.R2.fastq.gz\r\n",
      "PH-08.R1.fastq.gz\r\n",
      "PH-08.R2.fastq.gz\r\n",
      "PH-09.R1.fastq.gz\r\n",
      "PH-09.R2.fastq.gz\r\n",
      "PH-10.R1.fastq.gz\r\n",
      "PH-10.R2.fastq.gz\r\n",
      "PH-B1.R1.fastq.gz\r\n",
      "PH-B1.R2.fastq.gz\r\n",
      "pos1.R1.fastq.gz\r\n",
      "pos1.R2.fastq.gz\r\n",
      "pos2.R1.fastq.gz\r\n",
      "pos2.R2.fastq.gz\r\n",
      "pos3.R1.fastq.gz\r\n",
      "pos3.R2.fastq.gz\r\n",
      "pos4.R1.fastq.gz\r\n",
      "pos4.R2.fastq.gz\r\n",
      "pos5.R1.fastq.gz\r\n",
      "pos5.R2.fastq.gz\r\n",
      "S-01.R1.fastq.gz\r\n",
      "S-01.R2.fastq.gz\r\n",
      "S-02.R1.fastq.gz\r\n",
      "S-02.R2.fastq.gz\r\n",
      "S-03.R1.fastq.gz\r\n",
      "S-03.R2.fastq.gz\r\n",
      "S-04.R1.fastq.gz\r\n",
      "S-04.R2.fastq.gz\r\n",
      "S-05.R1.fastq.gz\r\n",
      "S-05.R2.fastq.gz\r\n",
      "SG10-blank.R1.fastq.gz\r\n",
      "SG10-blank.R2.fastq.gz\r\n",
      "SG10-S1.R1.fastq.gz\r\n",
      "SG10-S1.R2.fastq.gz\r\n",
      "SG10-S2.R1.fastq.gz\r\n",
      "SG10-S2.R2.fastq.gz\r\n",
      "SG10-S3.R1.fastq.gz\r\n",
      "SG10-S3.R2.fastq.gz\r\n",
      "SG10-S4.R1.fastq.gz\r\n",
      "SG10-S4.R2.fastq.gz\r\n",
      "SG10-S5.R1.fastq.gz\r\n",
      "SG10-S5.R2.fastq.gz\r\n",
      "SG1-blank.R1.fastq.gz\r\n",
      "SG1-blank.R2.fastq.gz\r\n",
      "SG1-S1.R1.fastq.gz\r\n",
      "SG1-S1.R2.fastq.gz\r\n",
      "SG1-S2.R1.fastq.gz\r\n",
      "SG1-S2.R2.fastq.gz\r\n",
      "SG1-S3.R1.fastq.gz\r\n",
      "SG1-S3.R2.fastq.gz\r\n",
      "SG1-S4.R1.fastq.gz\r\n",
      "SG1-S4.R2.fastq.gz\r\n",
      "SG1-S5.R1.fastq.gz\r\n",
      "SG1-S5.R2.fastq.gz\r\n",
      "SG21-blank.R1.fastq.gz\r\n",
      "SG21-blank.R2.fastq.gz\r\n",
      "SG21-S1.R1.fastq.gz\r\n",
      "SG21-S1.R2.fastq.gz\r\n",
      "SG21-S2.R1.fastq.gz\r\n",
      "SG21-S2.R2.fastq.gz\r\n",
      "SG21-S3.R1.fastq.gz\r\n",
      "SG21-S3.R2.fastq.gz\r\n",
      "SG21-S4.R1.fastq.gz\r\n",
      "SG21-S4.R2.fastq.gz\r\n",
      "SG21-S5.R1.fastq.gz\r\n",
      "SG21-S5.R2.fastq.gz\r\n",
      "SG22-blank.R1.fastq.gz\r\n",
      "SG22-blank.R2.fastq.gz\r\n",
      "SG22-S1.R1.fastq.gz\r\n",
      "SG22-S1.R2.fastq.gz\r\n",
      "SG22-S2.R1.fastq.gz\r\n",
      "SG22-S2.R2.fastq.gz\r\n",
      "SG22-S3.R1.fastq.gz\r\n",
      "SG22-S3.R2.fastq.gz\r\n",
      "SG22-S4.R1.fastq.gz\r\n",
      "SG22-S4.R2.fastq.gz\r\n",
      "SG22-S5.R1.fastq.gz\r\n",
      "SG22-S5.R2.fastq.gz\r\n",
      "SG23-blank.R1.fastq.gz\r\n",
      "SG23-blank.R2.fastq.gz\r\n",
      "SG23-S1.R1.fastq.gz\r\n",
      "SG23-S1.R2.fastq.gz\r\n",
      "SG23-S2.R1.fastq.gz\r\n",
      "SG23-S2.R2.fastq.gz\r\n",
      "SG23-S3.R1.fastq.gz\r\n",
      "SG23-S3.R2.fastq.gz\r\n",
      "SG23-S4.R1.fastq.gz\r\n",
      "SG23-S4.R2.fastq.gz\r\n",
      "SG23-S5.R1.fastq.gz\r\n",
      "SG23-S5.R2.fastq.gz\r\n",
      "SG2-blank.R1.fastq.gz\r\n",
      "SG2-blank.R2.fastq.gz\r\n",
      "SG2-S1.R1.fastq.gz\r\n",
      "SG2-S1.R2.fastq.gz\r\n",
      "SG2-S2.R1.fastq.gz\r\n",
      "SG2-S2.R2.fastq.gz\r\n",
      "SG2-S3.R1.fastq.gz\r\n",
      "SG2-S3.R2.fastq.gz\r\n",
      "SG2-S4.R1.fastq.gz\r\n",
      "SG2-S4.R2.fastq.gz\r\n",
      "SG2-S5.R1.fastq.gz\r\n",
      "SG2-S5.R2.fastq.gz\r\n",
      "SG5-blank.R1.fastq.gz\r\n",
      "SG5-blank.R2.fastq.gz\r\n",
      "SG5-S1.R1.fastq.gz\r\n",
      "SG5-S1.R2.fastq.gz\r\n",
      "SG5-S2.R1.fastq.gz\r\n",
      "SG5-S2.R2.fastq.gz\r\n",
      "SG5-S3.R1.fastq.gz\r\n",
      "SG5-S3.R2.fastq.gz\r\n",
      "SG5-S4.R1.fastq.gz\r\n",
      "SG5-S4.R2.fastq.gz\r\n",
      "SG5-S5.R1.fastq.gz\r\n",
      "SG5-S5.R2.fastq.gz\r\n",
      "SG6-blank.R1.fastq.gz\r\n",
      "SG6-blank.R2.fastq.gz\r\n",
      "SG6-S1.R1.fastq.gz\r\n",
      "SG6-S1.R2.fastq.gz\r\n",
      "SG6-S2.R1.fastq.gz\r\n",
      "SG6-S2.R2.fastq.gz\r\n",
      "SG6-S3.R1.fastq.gz\r\n",
      "SG6-S3.R2.fastq.gz\r\n",
      "SG6-S4.R1.fastq.gz\r\n",
      "SG6-S4.R2.fastq.gz\r\n",
      "SG6-S5.R1.fastq.gz\r\n",
      "SG6-S5.R2.fastq.gz\r\n",
      "SG7-blank.R1.fastq.gz\r\n",
      "SG7-blank.R2.fastq.gz\r\n",
      "SG7-S1.R1.fastq.gz\r\n",
      "SG7-S1.R2.fastq.gz\r\n",
      "SG7-S2.R1.fastq.gz\r\n",
      "SG7-S2.R2.fastq.gz\r\n",
      "SG7-S3.R1.fastq.gz\r\n",
      "SG7-S3.R2.fastq.gz\r\n",
      "SG7-S4.R1.fastq.gz\r\n",
      "SG7-S4.R2.fastq.gz\r\n",
      "SG7-S5.R1.fastq.gz\r\n",
      "SG7-S5.R2.fastq.gz\r\n",
      "SG8-blank.R1.fastq.gz\r\n",
      "SG8-blank.R2.fastq.gz\r\n",
      "SG8-S1.R1.fastq.gz\r\n",
      "SG8-S1.R2.fastq.gz\r\n",
      "SG8-S2.R1.fastq.gz\r\n",
      "SG8-S2.R2.fastq.gz\r\n",
      "SG8-S3.R1.fastq.gz\r\n",
      "SG8-S3.R2.fastq.gz\r\n",
      "SG8-S4.R1.fastq.gz\r\n",
      "SG8-S4.R2.fastq.gz\r\n",
      "SG8-S5.R1.fastq.gz\r\n",
      "SG8-S5.R2.fastq.gz\r\n",
      "SG9-blank.R1.fastq.gz\r\n",
      "SG9-blank.R2.fastq.gz\r\n",
      "SG9-S1.R1.fastq.gz\r\n",
      "SG9-S1.R2.fastq.gz\r\n",
      "SG9-S2.R1.fastq.gz\r\n",
      "SG9-S2.R2.fastq.gz\r\n",
      "SG9-S3.R1.fastq.gz\r\n",
      "SG9-S3.R2.fastq.gz\r\n",
      "SG9-S4.R1.fastq.gz\r\n",
      "SG9-S4.R2.fastq.gz\r\n",
      "SG9-S5.R1.fastq.gz\r\n",
      "SG9-S5.R2.fastq.gz\r\n",
      "T-01.R1.fastq.gz\r\n",
      "T-01.R2.fastq.gz\r\n",
      "T-02.R1.fastq.gz\r\n",
      "T-02.R2.fastq.gz\r\n",
      "T-03.R1.fastq.gz\r\n",
      "T-03.R2.fastq.gz\r\n",
      "T-04.R1.fastq.gz\r\n",
      "T-04.R2.fastq.gz\r\n",
      "T-05.R1.fastq.gz\r\n",
      "T-05.R2.fastq.gz\r\n",
      "T-B1.R1.fastq.gz\r\n",
      "T-B1.R2.fastq.gz\r\n",
      "W-01.R1.fastq.gz\r\n",
      "W-01.R2.fastq.gz\r\n",
      "W-02.R1.fastq.gz\r\n",
      "W-02.R2.fastq.gz\r\n",
      "W-03.R1.fastq.gz\r\n",
      "W-03.R2.fastq.gz\r\n",
      "W-04.R1.fastq.gz\r\n",
      "W-04.R2.fastq.gz\r\n",
      "W-05.R1.fastq.gz\r\n",
      "W-05.R2.fastq.gz\r\n",
      "W-06.R1.fastq.gz\r\n",
      "W-06.R2.fastq.gz\r\n",
      "W-07.R1.fastq.gz\r\n",
      "W-07.R2.fastq.gz\r\n",
      "W-08.R1.fastq.gz\r\n",
      "W-08.R2.fastq.gz\r\n",
      "W-09.R1.fastq.gz\r\n",
      "W-09.R2.fastq.gz\r\n",
      "W-10.R1.fastq.gz\r\n",
      "W-10.R2.fastq.gz\r\n",
      "W11-blank.R1.fastq.gz\r\n",
      "W11-blank.R2.fastq.gz\r\n",
      "W11-S1.R1.fastq.gz\r\n",
      "W11-S1.R2.fastq.gz\r\n",
      "W11-S2.R1.fastq.gz\r\n",
      "W11-S2.R2.fastq.gz\r\n",
      "W11-S3.R1.fastq.gz\r\n",
      "W11-S3.R2.fastq.gz\r\n",
      "W11-S4.R1.fastq.gz\r\n",
      "W11-S4.R2.fastq.gz\r\n",
      "W11-S5.R1.fastq.gz\r\n",
      "W11-S5.R2.fastq.gz\r\n",
      "W13-blank.R1.fastq.gz\r\n",
      "W13-blank.R2.fastq.gz\r\n",
      "W13-S1.R1.fastq.gz\r\n",
      "W13-S1.R2.fastq.gz\r\n",
      "W13-S2.R1.fastq.gz\r\n",
      "W13-S2.R2.fastq.gz\r\n",
      "W13-S3.R1.fastq.gz\r\n",
      "W13-S3.R2.fastq.gz\r\n",
      "W13-S4.R1.fastq.gz\r\n",
      "W13-S4.R2.fastq.gz\r\n",
      "W13-S5.R1.fastq.gz\r\n",
      "W13-S5.R2.fastq.gz\r\n",
      "W14-blank.R1.fastq.gz\r\n",
      "W14-blank.R2.fastq.gz\r\n",
      "W14-S1.R1.fastq.gz\r\n",
      "W14-S1.R2.fastq.gz\r\n",
      "W14-S2.R1.fastq.gz\r\n",
      "W14-S2.R2.fastq.gz\r\n",
      "W14-S3.R1.fastq.gz\r\n",
      "W14-S3.R2.fastq.gz\r\n",
      "W14-S4.R1.fastq.gz\r\n",
      "W14-S4.R2.fastq.gz\r\n",
      "W14-S5.R1.fastq.gz\r\n",
      "W14-S5.R2.fastq.gz\r\n",
      "W15-blank.R1.fastq.gz\r\n",
      "W15-blank.R2.fastq.gz\r\n",
      "W15-S1.R1.fastq.gz\r\n",
      "W15-S1.R2.fastq.gz\r\n",
      "W15-S2.R1.fastq.gz\r\n",
      "W15-S2.R2.fastq.gz\r\n",
      "W15-S3.R1.fastq.gz\r\n",
      "W15-S3.R2.fastq.gz\r\n",
      "W15-S4.R1.fastq.gz\r\n",
      "W15-S4.R2.fastq.gz\r\n",
      "W15-S5.R1.fastq.gz\r\n",
      "W15-S5.R2.fastq.gz\r\n",
      "W16-blank.R1.fastq.gz\r\n",
      "W16-blank.R2.fastq.gz\r\n",
      "W16-S1.R1.fastq.gz\r\n",
      "W16-S1.R2.fastq.gz\r\n",
      "W16-S2.R1.fastq.gz\r\n",
      "W16-S2.R2.fastq.gz\r\n",
      "W16-S3.R1.fastq.gz\r\n",
      "W16-S3.R2.fastq.gz\r\n",
      "W16-S4.R1.fastq.gz\r\n",
      "W16-S4.R2.fastq.gz\r\n",
      "W16-S5.R1.fastq.gz\r\n",
      "W16-S5.R2.fastq.gz\r\n",
      "W17-blank.R1.fastq.gz\r\n",
      "W17-blank.R2.fastq.gz\r\n",
      "W17-S1.R1.fastq.gz\r\n",
      "W17-S1.R2.fastq.gz\r\n",
      "W17-S2.R1.fastq.gz\r\n",
      "W17-S2.R2.fastq.gz\r\n",
      "W17-S3.R1.fastq.gz\r\n",
      "W17-S3.R2.fastq.gz\r\n",
      "W17-S4.R1.fastq.gz\r\n",
      "W17-S4.R2.fastq.gz\r\n",
      "W17-S5.R1.fastq.gz\r\n",
      "W17-S5.R2.fastq.gz\r\n",
      "W18-blank.R1.fastq.gz\r\n",
      "W18-blank.R2.fastq.gz\r\n",
      "W18-S1.R1.fastq.gz\r\n",
      "W18-S1.R2.fastq.gz\r\n",
      "W18-S2.R1.fastq.gz\r\n",
      "W18-S2.R2.fastq.gz\r\n",
      "W18-S3.R1.fastq.gz\r\n",
      "W18-S3.R2.fastq.gz\r\n",
      "W18-S4.R1.fastq.gz\r\n",
      "W18-S4.R2.fastq.gz\r\n",
      "W18-S5.R1.fastq.gz\r\n",
      "W18-S5.R2.fastq.gz\r\n",
      "W19-blank.R1.fastq.gz\r\n",
      "W19-blank.R2.fastq.gz\r\n",
      "W19-S1.R1.fastq.gz\r\n",
      "W19-S1.R2.fastq.gz\r\n",
      "W19-S2.R1.fastq.gz\r\n",
      "W19-S2.R2.fastq.gz\r\n",
      "W19-S3.R1.fastq.gz\r\n",
      "W19-S3.R2.fastq.gz\r\n",
      "W19-S4.R1.fastq.gz\r\n",
      "W19-S4.R2.fastq.gz\r\n",
      "W19-S5.R1.fastq.gz\r\n",
      "W19-S5.R2.fastq.gz\r\n",
      "W20-blank.R1.fastq.gz\r\n",
      "W20-blank.R2.fastq.gz\r\n",
      "W20-S1.R1.fastq.gz\r\n",
      "W20-S1.R2.fastq.gz\r\n",
      "W20-S2.R1.fastq.gz\r\n",
      "W20-S2.R2.fastq.gz\r\n",
      "W20-S3.R1.fastq.gz\r\n",
      "W20-S3.R2.fastq.gz\r\n",
      "W20-S4.R1.fastq.gz\r\n",
      "W20-S4.R2.fastq.gz\r\n",
      "W20-S5.R1.fastq.gz\r\n",
      "W20-S5.R2.fastq.gz\r\n",
      "W27-blank.R1.fastq.gz\r\n",
      "W27-blank.R2.fastq.gz\r\n",
      "W27-S1.R1.fastq.gz\r\n",
      "W27-S1.R2.fastq.gz\r\n",
      "W27-S2.R1.fastq.gz\r\n",
      "W27-S2.R2.fastq.gz\r\n",
      "W27-S3.R1.fastq.gz\r\n",
      "W27-S3.R2.fastq.gz\r\n",
      "W27-S4.R1.fastq.gz\r\n",
      "W27-S4.R2.fastq.gz\r\n",
      "W27-S5.R1.fastq.gz\r\n",
      "W27-S5.R2.fastq.gz\r\n",
      "W-B1.R1.fastq.gz\r\n",
      "W-B1.R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls -1 ../raw_reads/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare a text file specifying the samples to be processed, including the format and location of the reads - the querymap.\n",
    "\n",
    "The next command expects two `.fastq` files (forward and reverse reads) per sample in the directory `../raw_reads/`. It expects the files to be named 'SampleID', followed by '_R1' or '_R2' to identify the forward/reverse read file respectively.\n",
    "\n",
    "The raw data have been downloaded and demultiplexed. They can be found in `../raw_reads/`.\n",
    "\n",
    "SampleID must correspond to the first column in a file called `Sample_accessions.tsv`. This will either be pre-made to correspond to downloading read data from the NCBI Sequence Read Archive, or you will have to make it. The marker is '12S'.\n",
    "\n",
    "If the `Sample_accessions.tsv` was pre-made, use this code to proceed:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cat: ../Data/Sample_accessions.tsv: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "#%%bash\n",
    "#\n",
    "#for a in $(cat ../Data/Sample_accessions.tsv | grep \"12S\" | cut -f 1 | grep \"SampleID\" -v)\n",
    "#do\n",
    "#    R1=$(ls -1 ../raw_reads/$a-12S_* | grep \"_R1.fastq\")\n",
    "#    R2=$(ls -1 ../raw_reads/$a-12S_* | grep \"_R2.fastq\")\n",
    "#\n",
    "#    echo -e \"$a\\tfastq\\t$R1\\t$R2\"\n",
    "#done > Querymap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!head -n 10 Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**OR...**\n",
    "\n",
    "To make the `Sample_accessions.tsv` file, use:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!echo \"SampleID\" > ../1-trimming/Sample_accessions.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "for a in $(ls ../raw_reads/ | grep -w \"R1\" | cut -d '.' -f 1)\n",
    "do \n",
    "   SampleID=$a\n",
    "   \n",
    "   echo -e \"$SampleID\"\n",
    "done >> ../1-trimming/Sample_accessions.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SampleID\r\n",
      "24-blank\r\n",
      "24-S1\r\n",
      "24-S2\r\n",
      "24-S3\r\n",
      "24-S4\r\n",
      "24-S5\r\n",
      "25-blank\r\n",
      "25-S1\r\n",
      "25-S2\r\n",
      "25-S3\r\n",
      "25-S4\r\n",
      "25-S5\r\n",
      "26-blank\r\n",
      "26-S1\r\n",
      "26-S2\r\n",
      "26-S3\r\n",
      "26-S4\r\n",
      "26-S5\r\n",
      "BL3-blank\r\n",
      "BL3-S1\r\n",
      "BL3-S2\r\n",
      "BL3-S3\r\n",
      "BL3-S4\r\n",
      "BL3-S5\r\n",
      "BL4-blank\r\n",
      "BL4-S1\r\n",
      "BL4-S2\r\n",
      "BL4-S3\r\n",
      "BL4-S4\r\n",
      "BL4-S5\r\n",
      "C-01\r\n",
      "C-02\r\n",
      "C-03\r\n",
      "C-04\r\n",
      "C-05\r\n",
      "D-01\r\n",
      "D-02\r\n",
      "D-03\r\n",
      "D-04\r\n",
      "D-05\r\n",
      "D-B1\r\n",
      "LB-01\r\n",
      "LB-02\r\n",
      "LB-03\r\n",
      "LB-04\r\n",
      "LB-05\r\n",
      "LB-06\r\n",
      "LB-07\r\n",
      "LB-08\r\n",
      "LB-09\r\n",
      "LB-10\r\n",
      "LB-B1\r\n",
      "LH-01\r\n",
      "LH-02\r\n",
      "LH-03\r\n",
      "LH-04\r\n",
      "LH-05\r\n",
      "N-01\r\n",
      "N-02\r\n",
      "N-03\r\n",
      "N-04\r\n",
      "N-05\r\n",
      "N+C-B1\r\n",
      "neg1\r\n",
      "neg2\r\n",
      "neg3\r\n",
      "neg4\r\n",
      "neg5\r\n",
      "PH-01\r\n",
      "PH-02\r\n",
      "PH-03\r\n",
      "PH-04\r\n",
      "PH-05\r\n",
      "PH-06\r\n",
      "PH-07\r\n",
      "PH-08\r\n",
      "PH-09\r\n",
      "PH-10\r\n",
      "PH-B1\r\n",
      "pos1\r\n",
      "pos2\r\n",
      "pos3\r\n",
      "pos4\r\n",
      "pos5\r\n",
      "S-01\r\n",
      "S-02\r\n",
      "S-03\r\n",
      "S-04\r\n",
      "S-05\r\n",
      "SG10-blank\r\n",
      "SG10-S1\r\n",
      "SG10-S2\r\n",
      "SG10-S3\r\n",
      "SG10-S4\r\n",
      "SG10-S5\r\n",
      "SG1-blank\r\n",
      "SG1-S1\r\n",
      "SG1-S2\r\n",
      "SG1-S3\r\n",
      "SG1-S4\r\n",
      "SG1-S5\r\n",
      "SG21-blank\r\n",
      "SG21-S1\r\n",
      "SG21-S2\r\n",
      "SG21-S3\r\n",
      "SG21-S4\r\n",
      "SG21-S5\r\n",
      "SG22-blank\r\n",
      "SG22-S1\r\n",
      "SG22-S2\r\n",
      "SG22-S3\r\n",
      "SG22-S4\r\n",
      "SG22-S5\r\n",
      "SG23-blank\r\n",
      "SG23-S1\r\n",
      "SG23-S2\r\n",
      "SG23-S3\r\n",
      "SG23-S4\r\n",
      "SG23-S5\r\n",
      "SG2-blank\r\n",
      "SG2-S1\r\n",
      "SG2-S2\r\n",
      "SG2-S3\r\n",
      "SG2-S4\r\n",
      "SG2-S5\r\n",
      "SG5-blank\r\n",
      "SG5-S1\r\n",
      "SG5-S2\r\n",
      "SG5-S3\r\n",
      "SG5-S4\r\n",
      "SG5-S5\r\n",
      "SG6-blank\r\n",
      "SG6-S1\r\n",
      "SG6-S2\r\n",
      "SG6-S3\r\n",
      "SG6-S4\r\n",
      "SG6-S5\r\n",
      "SG7-blank\r\n",
      "SG7-S1\r\n",
      "SG7-S2\r\n",
      "SG7-S3\r\n",
      "SG7-S4\r\n",
      "SG7-S5\r\n",
      "SG8-blank\r\n",
      "SG8-S1\r\n",
      "SG8-S2\r\n",
      "SG8-S3\r\n",
      "SG8-S4\r\n",
      "SG8-S5\r\n",
      "SG9-blank\r\n",
      "SG9-S1\r\n",
      "SG9-S2\r\n",
      "SG9-S3\r\n",
      "SG9-S4\r\n",
      "SG9-S5\r\n",
      "T-01\r\n",
      "T-02\r\n",
      "T-03\r\n",
      "T-04\r\n",
      "T-05\r\n",
      "T-B1\r\n",
      "W-01\r\n",
      "W-02\r\n",
      "W-03\r\n",
      "W-04\r\n",
      "W-05\r\n",
      "W-06\r\n",
      "W-07\r\n",
      "W-08\r\n",
      "W-09\r\n",
      "W-10\r\n",
      "W11-blank\r\n",
      "W11-S1\r\n",
      "W11-S2\r\n",
      "W11-S3\r\n",
      "W11-S4\r\n",
      "W11-S5\r\n",
      "W13-blank\r\n",
      "W13-S1\r\n",
      "W13-S2\r\n",
      "W13-S3\r\n",
      "W13-S4\r\n",
      "W13-S5\r\n",
      "W14-blank\r\n",
      "W14-S1\r\n",
      "W14-S2\r\n",
      "W14-S3\r\n",
      "W14-S4\r\n",
      "W14-S5\r\n",
      "W15-blank\r\n",
      "W15-S1\r\n",
      "W15-S2\r\n",
      "W15-S3\r\n",
      "W15-S4\r\n",
      "W15-S5\r\n",
      "W16-blank\r\n",
      "W16-S1\r\n",
      "W16-S2\r\n",
      "W16-S3\r\n",
      "W16-S4\r\n",
      "W16-S5\r\n",
      "W17-blank\r\n",
      "W17-S1\r\n",
      "W17-S2\r\n",
      "W17-S3\r\n",
      "W17-S4\r\n",
      "W17-S5\r\n",
      "W18-blank\r\n",
      "W18-S1\r\n",
      "W18-S2\r\n",
      "W18-S3\r\n",
      "W18-S4\r\n",
      "W18-S5\r\n",
      "W19-blank\r\n",
      "W19-S1\r\n",
      "W19-S2\r\n",
      "W19-S3\r\n",
      "W19-S4\r\n",
      "W19-S5\r\n",
      "W20-blank\r\n",
      "W20-S1\r\n",
      "W20-S2\r\n",
      "W20-S3\r\n",
      "W20-S4\r\n",
      "W20-S5\r\n",
      "W27-blank\r\n",
      "W27-S1\r\n",
      "W27-S2\r\n",
      "W27-S3\r\n",
      "W27-S4\r\n",
      "W27-S5\r\n",
      "W-B1\r\n"
     ]
    }
   ],
   "source": [
    "!cat ../1-trimming/Sample_accessions.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for a in $(cat ../1-trimming/Sample_accessions.tsv | grep \"SampleID\" -v)\n",
    "do\n",
    "    R1=$(ls -1 ../raw_reads/$a.* | grep -w \"R1\")\n",
    "    R2=$(ls -1  ../raw_reads/$a.* | grep -w \"R2\")\n",
    "\n",
    "    echo -e \"$a\\tfastq\\t$R1\\t$R2\"\n",
    "done > Querymap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24-blank\tfastq\t../raw_reads/24-blank.R1.fastq.gz\t../raw_reads/24-blank.R2.fastq.gz\r\n",
      "24-S1\tfastq\t../raw_reads/24-S1.R1.fastq.gz\t../raw_reads/24-S1.R2.fastq.gz\r\n",
      "24-S2\tfastq\t../raw_reads/24-S2.R1.fastq.gz\t../raw_reads/24-S2.R2.fastq.gz\r\n",
      "24-S3\tfastq\t../raw_reads/24-S3.R1.fastq.gz\t../raw_reads/24-S3.R2.fastq.gz\r\n",
      "24-S4\tfastq\t../raw_reads/24-S4.R1.fastq.gz\t../raw_reads/24-S4.R2.fastq.gz\r\n",
      "24-S5\tfastq\t../raw_reads/24-S5.R1.fastq.gz\t../raw_reads/24-S5.R2.fastq.gz\r\n",
      "25-blank\tfastq\t../raw_reads/25-blank.R1.fastq.gz\t../raw_reads/25-blank.R2.fastq.gz\r\n",
      "25-S1\tfastq\t../raw_reads/25-S1.R1.fastq.gz\t../raw_reads/25-S1.R2.fastq.gz\r\n",
      "25-S2\tfastq\t../raw_reads/25-S2.R1.fastq.gz\t../raw_reads/25-S2.R2.fastq.gz\r\n",
      "25-S3\tfastq\t../raw_reads/25-S3.R1.fastq.gz\t../raw_reads/25-S3.R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 10 Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To the `Querymap.txt` file, add two columns which specify the number of bases to remove from the forward and reverse read. In our case, we want to remove 18 bp to ensure that there is no forward or reverse primer left."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "sed 's/$/&\\t18/' Querymap.txt > Querymap_new.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Have a look (note that the output is probably line-wrapped):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24-blank\tfastq\t../raw_reads/24-blank.R1.fastq.gz\t../raw_reads/24-blank.R2.fastq.gz\t18\r\n",
      "24-S1\tfastq\t../raw_reads/24-S1.R1.fastq.gz\t../raw_reads/24-S1.R2.fastq.gz\t18\r\n",
      "24-S2\tfastq\t../raw_reads/24-S2.R1.fastq.gz\t../raw_reads/24-S2.R2.fastq.gz\t18\r\n",
      "24-S3\tfastq\t../raw_reads/24-S3.R1.fastq.gz\t../raw_reads/24-S3.R2.fastq.gz\t18\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 4 Querymap_new.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "sed 's/$/&\\t18/' Querymap_new.txt > Querymap_final.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24-blank\tfastq\t../raw_reads/24-blank.R1.fastq.gz\t../raw_reads/24-blank.R2.fastq.gz\t18\t18\r\n",
      "24-S1\tfastq\t../raw_reads/24-S1.R1.fastq.gz\t../raw_reads/24-S1.R2.fastq.gz\t18\t18\r\n",
      "24-S2\tfastq\t../raw_reads/24-S2.R1.fastq.gz\t../raw_reads/24-S2.R2.fastq.gz\t18\t18\r\n",
      "24-S3\tfastq\t../raw_reads/24-S3.R1.fastq.gz\t../raw_reads/24-S3.R2.fastq.gz\t18\t18\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 4 Querymap_final.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Raw read processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, perform basic quality trimming and clipping (Trimmomatic) and paired-end read merging (flash). metaBEAT will be used to process all samples in one go."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: metaBEAT.py [-h] [-Q <FILE>] [-B <FILE>] [--g_queries <FILE>] [-v] [-s]\r\n",
      "                   [-f] [-p] [-k] [-t] [-b] [-m <string>] [-n <INT>] [-E] [-e]\r\n",
      "                   [--read_stats_off] [--PCR_primer <FILE>] [--bc_dist <INT>]\r\n",
      "                   [--trim_adapter <FILE>] [--trim_qual <INT>] [--phred <INT>]\r\n",
      "                   [--trim_window <INT>] [--read_crop <INT>]\r\n",
      "                   [--trim_minlength <INT>] [--merge] [--product_length <INT>]\r\n",
      "                   [--merged_only] [--forward_only] [--length_filter <INT>]\r\n",
      "                   [--length_deviation <FLOAT>] [-R <FILE>] [--gb_out <FILE>]\r\n",
      "                   [--rec_check] [--gb_to_taxid <FILE>] [--cluster]\r\n",
      "                   [--clust_match <FLOAT>] [--clust_cov <INT>]\r\n",
      "                   [--blast_db <PATH>] [--blast_xml <PATH>]\r\n",
      "                   [--update_taxonomy] [--taxonomy_db <FILE>]\r\n",
      "                   [--min_ident <FLOAT>] [--min_ali_length <FLOAT>]\r\n",
      "                   [--bitscore_skim_LCA <FLOAT>] [--bitscore_skim_adjust_off]\r\n",
      "                   [--min_bit <INT>] [--refpkg <DIR>] [--jplace <FILE>]\r\n",
      "                   [--jellyfish_hash_size <STR>] [--kraken_db <DIR>]\r\n",
      "                   [--kraken_score_threshold <FLOAT>] [--rm_kraken_db]\r\n",
      "                   [-o OUTPUT_PREFIX] [--metadata METADATA]\r\n",
      "                   [-@ <email-address>] [--version]\r\n",
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool version:\r\n",
      "v.0.97.11-global\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -Q <FILE>, --querylist <FILE>\r\n",
      "                        file containing a list of query files\r\n",
      "  -B <FILE>, --BIOM_input <FILE>\r\n",
      "                        OTU table in BIOM format\r\n",
      "  --g_queries <FILE>    fasta file containing query sequences (in combination\r\n",
      "                        with '-B' sequence headers are expected to match)\r\n",
      "  -v, --verbose         turn verbose output on\r\n",
      "  -s, --seqinfo         write out seq_info.csv file\r\n",
      "  -f, --fasta           write out ref.fasta file\r\n",
      "  -p, --pplace          perform phylogenetic placement\r\n",
      "  -k, --kraken          perform phylogenetic placement\r\n",
      "  -t, --taxids          write out taxid.txt file\r\n",
      "  -b, --blast           compile local blast db and blast queries\r\n",
      "  -m <string>, --marker <string>\r\n",
      "                        marker ID (default: marker)\r\n",
      "  -n <INT>, --n_threads <INT>\r\n",
      "                        Number of threads (default: 1)\r\n",
      "  -E, --extract_centroid_reads\r\n",
      "                        extract centroid reads to files\r\n",
      "  -e, --extract_all_reads\r\n",
      "                        extract reads to files\r\n",
      "  --read_stats_off      ommit writing read stats to file\r\n",
      "  --version             show program's version number and exit\r\n",
      "\r\n",
      "Query preprocessing:\r\n",
      "  The parameters in this group affect how the query sequences are processed\r\n",
      "\r\n",
      "  --PCR_primer <FILE>   PCR primers (provided in fasta file) to be clipped\r\n",
      "                        from reads\r\n",
      "  --bc_dist <INT>       Number of mismatches allowed in barcode sequences\r\n",
      "  --trim_adapter <FILE>\r\n",
      "                        trim adapters provided in file\r\n",
      "  --trim_qual <INT>     minimum phred quality score (default: 30)\r\n",
      "  --phred <INT>         phred quality score offset - 33 or 64 (default: 33)\r\n",
      "  --trim_window <INT>   sliding window size (default: 5) for trimming; if\r\n",
      "                        average quality drops below the specified minimum\r\n",
      "                        quality all subsequent bases are removed from the\r\n",
      "                        reads\r\n",
      "  --read_crop <INT>     Crop reads to this length if they are longer than that\r\n",
      "                        (default: off)\r\n",
      "  --trim_minlength <INT>\r\n",
      "                        minimum length of reads to be retained after trimming\r\n",
      "                        (default: 50)\r\n",
      "  --merge               attempt to merge paired-end reads\r\n",
      "  --product_length <INT>\r\n",
      "                        estimated length of PCR product (specifying this\r\n",
      "                        option increases merging efficiency)\r\n",
      "  --merged_only         only process successfully merged read-pairs\r\n",
      "  --forward_only        only process sequences that contain forward reads\r\n",
      "                        (i.e. unmerged forward reads and merged reads)\r\n",
      "  --length_filter <INT>\r\n",
      "                        only process reads, which are within +/- 10 percent of\r\n",
      "                        this length\r\n",
      "  --length_deviation <FLOAT>\r\n",
      "                        allowed deviation (in percent) from length specified\r\n",
      "                        by --length_filter (default=0.1)\r\n",
      "\r\n",
      "Reference:\r\n",
      "  The parameters in this group affect the reference to be used in the\r\n",
      "  analyses\r\n",
      "\r\n",
      "  -R <FILE>, --REFlist <FILE>\r\n",
      "                        file containing a list of files to be used as\r\n",
      "                        reference sequences\r\n",
      "  --gb_out <FILE>       output the corrected gb file\r\n",
      "  --rec_check           check records to be used as reference\r\n",
      "  --gb_to_taxid <FILE>  comma delimited file containing 'gb accession,taxid'\r\n",
      "                        for a list of taxa\r\n",
      "\r\n",
      "Query clustering options:\r\n",
      "  The parameters in this group affect read clustering\r\n",
      "\r\n",
      "  --cluster             perform clustering of query sequences using vsearch\r\n",
      "  --clust_match <FLOAT>\r\n",
      "                        identity threshold for clustering in percent (default:\r\n",
      "                        1)\r\n",
      "  --clust_cov <INT>     minimum number of records in cluster (default: 1)\r\n",
      "\r\n",
      "BLAST search:\r\n",
      "  The parameters in this group affect BLAST search and BLAST based taxonomic\r\n",
      "  assignment\r\n",
      "\r\n",
      "  --blast_db <PATH>     path to precompiled blast database\r\n",
      "  --blast_xml <PATH>    path to Blast result in xml format\r\n",
      "  --update_taxonomy     Download/update taxonomy database. Database will be\r\n",
      "                        called 'taxonomy.db' and will be compiled in the same\r\n",
      "                        location as the metaBEAT.py script.\r\n",
      "  --taxonomy_db <FILE>  taxonomy database file location. In case it's not the\r\n",
      "                        default, which is 'taxonomy.db' in the same directory\r\n",
      "                        as the metaBEAT.py script.\r\n",
      "  --min_ident <FLOAT>   minimum identity threshold in percent (default: 0.80)\r\n",
      "  --min_ali_length <FLOAT>\r\n",
      "                        minimum alignment length in percent of total query\r\n",
      "                        length (default: 0.95)\r\n",
      "  --bitscore_skim_LCA <FLOAT>\r\n",
      "                        Only BLAST hits with bitscores differing by less than\r\n",
      "                        this factor from the top hit (bitscore skim window)\r\n",
      "                        will be considered for LCA (0-1; default: 0.1)\r\n",
      "  --bitscore_skim_adjust_off\r\n",
      "                        Per default a 100% identity BLAST top hit across the\r\n",
      "                        minimum alignment length triggers an adjustment of the\r\n",
      "                        bitscore skim window to '0', i.e. only hits with\r\n",
      "                        bitscores as good as the top hit are considered for\r\n",
      "                        LCA. This flag switches this behaviour off.\r\n",
      "  --min_bit <INT>       minimum bitscore (default: 80)\r\n",
      "\r\n",
      "Phylogenetic placement:\r\n",
      "  The parameters in this group affect phylogenetic placement\r\n",
      "\r\n",
      "  --refpkg <DIR>        PATH to refpkg for pplacer\r\n",
      "  --jplace <FILE>       phylogenetic placement result from prefious pplacer\r\n",
      "                        run in *.jplace format\r\n",
      "\r\n",
      "Kraken:\r\n",
      "  The parameters in this group affect taxonomic assignment using Kraken\r\n",
      "\r\n",
      "  --jellyfish_hash_size <STR>\r\n",
      "                        jellyfish hash size to control memory usage during\r\n",
      "                        kraken database building. A table size of '6400M' will\r\n",
      "                        require ~44G of RAM. '2700M' -> 20G RAM.\r\n",
      "  --kraken_db <DIR>     PATH to a Kraken database\r\n",
      "  --kraken_score_threshold <FLOAT>\r\n",
      "                        minimum proportion of k-mers to support assignment\r\n",
      "                        (0-1; default: 0)\r\n",
      "  --rm_kraken_db        Remove Kraken database after successful completion\r\n",
      "\r\n",
      "BIOM OUTPUT:\r\n",
      "  The arguments in this groups affect the output in BIOM format\r\n",
      "\r\n",
      "  -o OUTPUT_PREFIX, --output_prefix OUTPUT_PREFIX\r\n",
      "                        prefix for BIOM output files (default='metaBEAT')\r\n",
      "  --metadata METADATA   comma delimited file containing metadata (optional)\r\n",
      "\r\n",
      "Entrez identification:\r\n",
      "  metaBEAT is querying the NCBI Entrez databases, please provide an email\r\n",
      "  address for identification\r\n",
      "\r\n",
      "  -@ <email-address>, --email <email-address>\r\n",
      "                        provide your email address for identification to NCBI\r\n"
     ]
    }
   ],
   "source": [
    "!metaBEAT_global.py -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Command to trim:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starttime: Sat Jul  4 02:12:47 UTC 2020\n",
      "\n",
      "Endtime: Sat Jul  4 07:39:33 UTC 2020\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "echo -e \"Starttime: $(date)\\n\"\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-Q Querymap_final.txt \\\n",
    "--trim_qual 30 \\\n",
    "--read_crop 110 \\\n",
    "--trim_minlength 90 \\\n",
    "--merge \\\n",
    "--product_length 106 \\\n",
    "--forward_only \\\n",
    "--length_filter 106 \\\n",
    "--length_deviation 0.2 \\\n",
    "-m 12S -o Eel2017_trim30-min90-crop110-forwonly-filter100-deviation0.2 \\\n",
    "-n 5 -v \\\n",
    "-@ nathangriffiths484@gmail.com &> log\n",
    "\n",
    "echo -e \"Endtime: $(date)\\n\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read processing will take several hours.\n",
    "\n",
    "\n",
    "# Visualise query survival after trimming\n",
    "\n",
    "metaBEAT will generate a directory with all temporary files that were created during the processing for each sample and will record useful stats summarizing the data processing in the file `metaBEAT_read_stats.csv`. You can explore the table manually or quickly plot out some of these stats here:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes.AxesSubplot at 0x7fd9a96223d0>"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAEACAYAAACj0I2EAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAFlJJREFUeJzt3X2sZHddx/H3thcipaV1KdmtFLkJiCABK0gFhPgFRIsg\nEE2IYoxVQvzDCCRidzEaNxFlrYmgIfqHRlqrokRCFQXTFneCIqDFXSwPpTx0tUK7BfuwFTSy9PrH\nObf37p3vuffMzJnzOXN+n1dyd+fMnbnnvTN3f3vudx4WzMzMzMzMzMzMzMzMzMzMzMzMzMx6sa/l\n5U4Cp4FvAF8HLgf2A38BPL7+/KuA+zovNDOzTt1OtYBvdzVwVX36EHC01yIzM5vL7cCjd5x3K3Cg\nPn2w3jYzs4H7AnAcuBl4bX3evds+v2/HtpmZDdQl9e+PAU4Az2d6Ab+n1yIzMzvLWsvL3Vn//mXg\nPVQPip6iGrXcRbXg351c74vAtyzYaGZWms8DT1zGFz4PuKA+/UjgQ8APUD0oeqg+/zD5g6Ibywha\n0BF1QOKIOiBxRB3Q4Ig6IHFEHZA4og5IHFEHJI6oAxrMtXa2OUI/QHVUvnn5PwVuoJqnvwt4DVtP\nW1wF6+qAxLo6ILGuDmiwrg5IrKsDEuvqgMS6OiCxrg7oUpsF/XbgsuT8e4Dv7zbHzMyGaogjl1AH\nJEIdkAh1QINQByRCHZAIdUAi1AGJUAc0GOLaOcwoM7OBm2vtPKfrihUQ6oBEqAMSoQ5oEOqARKgD\nEqEOSIQ6IBHqgC6VuKCbmdkcPHIxM5udRy5mZiUrcUEPdUAi1AGJUAc0CHVAItQBiVAHJEIdkAh1\nQJdKXNDNzGwOnqGbmc3OM3Qzs92tnaZaLHd8rJ2WZq2IIR6hhzogEeqARKgDGoQ6IBHqgESoAxKh\nDgA2YGPbx7H698GtVT5CNzMrWdv/JHpeGz3sw8ysrY384HffQ78MxFxrp4/QzcxGosQFPdQBiVAH\nJEId0CDUAYlQByRCHZAIdcC0iTqgUyUu6GZmo+QZupmVxDN0MzMbvhIX9FAHJEIdkAh1QINQByRC\nHZAIdUAi1AHTJuqATpW4oJuZjZJn6GZWEs/Qzcxs+Epc0EMdkAh1QCLUAQ1CHZAIdUAi1AGJUAdM\nm6gDOlXigm5mNkqeoZtZSTxDNzOz4StxQQ91QCLUAYlQBzQIdUAi1AGJUAckQh0wbaIO6FSJC7qZ\n2Sh5hm5mJfEM3czMhq/EBT3UAYlQByRCHdAg1AGJUAckQh2QCHXAtIk6oFMlLuhmZqPkGbqZlcQz\ndDMzG74SF/RQByRCHZAIdUCDUAckQh2QCHVAItQB0ybqgE6VuKCbmRXtXOA48N56ez9wI3AbcANw\nUcP1smGVmZnKBmwkH4Nbq+bqaXuE/nrgU9t2cphqQX8S8IF628zMBu5S4CbgBWwdod8KHKhPH6y3\nM0P7Vw8GOcdz0wxCHZAIdUAi1AGJUAcwdYR+rLgj9LcCvwg8uO28A8Cp+vQpthZ3MzMTWdvj8y8D\n7qaan0fDZfb61+0a4GR9+j7gBFsPLW9+zb632ePz3q4+htSzfZs9Pu/tYd5/m+epe5Isac+kPn1l\nvX1yZ1hXfgO4A7gduBP4KnAd1YjlYH2ZS1itkYuZlWvUD4rO4vvYmqFfDRyqTx8Gjqqi5hDqgESo\nAxKhDmgQ6oBEqAMSoQ5IhDoAz9DTnRwFXkz1tMUX0rygm5lZT/xeLmZWEr+Xi5mZDV+JC3qoAxKh\nDkiEOqBBqAMSoQ5IhDogEeqAaRN1QKdKXNDNzEbJM3QzK4ln6GZmNnwlLuihDkiEOiAR6oAGoQ5I\nhDogEeqARKgDpk3UAZ0qcUE3Mxslz9DNrCSeoZuZ2fCVuKCHOiAR6oBEqAMahDogEeqARKgDEqEO\nmDapf1+DrXeO3fGxdlqSNoe93j7XzKwAZ2h+P6x9F/RZsgjP0M2sJLvM0BsX9Id+6ZFn6GZmJStx\nQQ91QCLUAYlQBzQIdUAi1AGJUAckQh0wbaIO6FSJC7qZ2Sh5hm5mJfEM3czMhq/EBT3UAYlQByRC\nHdAg1AGJUAckQh2QCHXAtIk6oFMlLuhmZqPkGbqZlcQzdDMzG74SF/RQByRCHZAIdUCDUAckQh2Q\nCHVAItQB0ybqgE6VuKCbmY2SZ+hmVhLP0M3MbPhKXNBDHZAIdUAi1AENQh2QCHVAItQBiVAHTJuo\nAzpV4oJuZjZKnqGbWUk8Qzczs+ErcUEPdUAi1AGJUAc0CHVAItQBiVAHJEIdMG2iDuhUiQu6mdko\neYZuZiXxDN3MzIavxAU91AGJUAckQh3QINQBiVAHJEIdkAh1wLSJOqBTJS7oZmZF+ibgo8AJ4FPA\nW+rz9wM3ArcBNwAXNVy/aShlZqawARvJR9P5m59TdC7HefXva8BHgOcBVwNX1ecfAo72HWVmNofi\nF/RN5wH/AjwVuBU4UJ9/sN6WRM0h1AGJUAckQh3QINQBiVAHJEIdkAh1AFML97FRLehtZujnUI1c\nTgHHgE9SLean6s+fYmtxNzOzFXAh1cjlBcC9Oz53T8N1hniEbmblGvXIZW2Gy94P/C3wTKqj8oPA\nXcAlwN27XO8a4GR9+j6qo/1JvR317972tre93df2jFd7yDL7Ariy3j65c8dduZitZ7A8Avgg8CKq\nB0UP1ecfZrUeFA11QCLUAYlQBzQIdUAi1AGJUAckQh3AyGfoex2hXwJcSzVHPwe4DvgAcBx4F/Aa\nqn9NXjXPzs3MrDt+LxczK4nfy8XMzIavxAU91AGJUAckQh3QINQBiVAHJEIdkAh1wLSJOqBTJS7o\nZmaj5Bm6mZXEM3QzMxu+Ehf0UAckQh2QCHVAg1AHJEIdkAh1QCLUAdMm6oBOlbigm5mNkmfoZlYS\nz9DNzGz4SlzQQx2QCHVAItQBDUIdkAh1QCLUAYnob1drp6mOcnd+7DDpL6kHs7zbopnZijhzQfNo\nZbw8QzezMZpxVu4ZupmZDUiJC3qoAxKhDkiEOqBBqAMSoQ5IhDogEeqAaRN1QKdKXNDNzEbJM3Qz\nGyPP0M3MbHWVuKCHOiAR6oBEqAMahDogEeqARKgDEqEOmDZRB3SqxAXdzGyUPEM3szHyDN3MzFZX\niQt6qAMSoQ5IhDqgQagDEqEOSIQ6IBHqgGkTdUCnSlzQzcxGyTN0Mxsjz9DNzGx1lbighzogEeqA\nRKgDGoQ6IBHqgESoAxKhDpg2UQd0qsQF3cxslDxDN7Mx8gzdzMxWV4kLeqgDEqEOSIQ6oEGoAxKh\nDkiEOiAR6oBpE3VAp0pc0M3MRskzdDMbI8/QzcxsdZW4oIc6IBHqgESoAxqEOiAR6oBEqAMSoQ6Y\nNlEHdKrEBd3MbJQ8QzezMfIMvcHjgGPAJ4FPAK+rz98P3AjcBtwAXDTrzs3MrF8Hgcvq0+cDnwGe\nAlwNXFWffwg4mly36Z88pVAHJEIdkAh1QINQByRCHZAIdUAietzXBmwkHzvPP9Zw/s7r9G6ufbY5\nQr8LOFGf/m/g08BjgZcD19bnXwu8cp4AMzPTWAf+HbgAuHfb+ft2bG8a4hG6mY1fyyP0vc5frSP0\nWZwPfIytI/GdC/g9yXW8oJuZQpEL+lrLyz0MeDdwHXB9fd4pqvn6XcAlwN0N170GOFmfvo9qfDOp\nt6P+vc/ty4C3CfefbW+eN5Se7S1D6dnc9v23uvffG+j173/Txdi2faLOanP5pd9fV9bbJ3fuuEv7\ngD8G3rrj/KupHgwFOIwfFF1EqAMSoQ5oEOqARKgDEqEOSESP+yryQdE2ngc8SPVP2fH64wqqpy3e\nxO5PWxzigm5m41fkyMUvLDKzMfILiwoR6oBEqAMSoQ5oEOqARKgDEqEOSIQ6YNpEHdCpEhd0M7NR\n8sjFzMbIIxczM1tdJS7ooQ5IhDogEeqABqEOSIQ6IBHqgESoA6ZN1AGdKnFBNzMbJc/QzWyMPEM3\nM7PVVeKCHuqARKgDEqEOaBDqgESoAxKhDkiEOmDaRB3QqRIXdDOzUfIM3czGyDN0MzNbXSUu6KEO\nSIQ6IBHqgAahDkiEOiAR6oBEqAOmTdQBnSpxQTczGyXP0M1sjDxDNzOz1VXigh7qgESoAxKhDmgQ\n6oBEqAMSoQ5IhDpg2kQd0KkSF3QzM5uD/09RM2tp7TTVmrHjY+30HF/M/6foEvhBUTNra5cHMmde\nR/ygaCFCHZAIdUAi1AENQh2QCHVAItQBiVAHTJuoAzpV4oJuZjZKHrmY2VB45LLFIxczs5KVuKCH\nOiAR6oBEqAMahDogEeqARKgDEqEOmDZRB3SqxAXdzGyUPEM3s6HwDH2LZ+hmZiUrcUEPdUAi1AGJ\nUAc0CHVAItQBiVAHJEIdMG2iDuhUiQu6mdkoeYZuZkPhGfoWz9DNrCRNb+ZV7psClrighzogEeqA\nRKgDGoQ6IBHqgESoAxLR7Zc7c8Hi6/mk2ySxEhd0M7NR8gzdzIZi1hl6w+U3L+4ZupmZrag2C/of\nAaeAW7adtx+4EbgNuAG4qPu0pQl1QCLUAYlQBzQIdUAi1AGJUAckQh0wbaIO6FSbBf0dwBU7zjtM\ntaA/CfhAvW1mZitgnbOP0G8FDtSnD9bbmWKfPmRmM5v1//Tc4/8BLe//FJ13hn6AagxD/fuBXS5r\nZmY9WOvga+z1L9g1wMn69H3ACbYGV1H/3uf2ZcDbhPvPtjfPG0rP9pah9Gxu+/5b3fvvDez+95/8\nZnxIy8vvtb39uifqrN0uvwacSda4c74GD750hh3vth3AlfX2yel9dWud6ZHLwfr0JazWyCXUAYlQ\nByRCHdAg1AGJUAckQh2QiD0+Lxi5HGs5cul9FLPUtXOdsxf0q4FD9enDwNGG6w1xQTezYRrwDH08\nC/o7gS8B/wfcAfw01dMWb2Lvpy16QTeztrygn/VnG54NOP8rycevCZtCuO8moQ5IhDqgQagDEqEO\nSIQ6IBF7fN4jl7P+bLPr4kHRPdzy6LO33wO8+WnL36+ZdWPtdPVGWFPnPwBnHtXD/skflLS+Jf+q\nXbcB+69Xh5lZa30dofY0JhnvyMXv5WJmNhIlLuihDkiEOiAR6oAGoQ5IhDogEeqARKgDpk3UAZ0q\ncUE3M7M5eIZutvo8Q/cM3czM+lTigh7qgESoAxKhDmgQ6oBEqAMSoQ5IhDpg2kQd0KkSF3QzM5uD\nZ+hmq88z9PT8tc3bIPlYO734bTG7Hl4pamY2RmdoXnf3Ja+sXb4SRy6hDkiEOiAR6oAGoQ5IhDog\nEeqARKgDpk3UAZ0qcUE3M7M5eIZutvo8Q5/rOoveFrPzEbqZ2UiUuKCHOiAR6oBEqAMahDogEeqA\nRKgDEqEOmDZRB3SqxAXdzMzm4Bm62eqbY068dprZn5/tGfpZt8Xs/Dx0M1uCMxfka5Lm+dmlKHHk\nEuqARKgDEqEOaBDqgESoAxKhDpg2UQckJuqATpW4oJuZ2Rw8QzdbffPMiTu+jmfobXiGbmZzWgPO\nLLpwWYdKHLmEOiAR6oBEqAMahDogEeqARCx/F5tvTpV9ZCbLT5rZRB3QqRIXdDMzm4Nn6Garr+O5\nc1f78Qx9pxU6Qp/rhQq2dI33S9PP4r7PGpXwPb4Gs81pbAYrtKBvvlBh58eZWV+oEB2HdSHUAYlo\nd7HG+yU5b+77bI6uXkU3X6az7/EOm7o0Yfa5ex9N47FCC7qZmSkls6W5Z+jLmlXZYhTzxbEa6u3V\n49x5DF/LM3QzM1vQCBb03R5kSR9Mil2+VsODUrs9yLfbA1atH+SKFvvv4YGxoe7/rH3HYl9rKX+W\nlk29CnXAtIk6IDFRB3RqBAt6lw++NT0otbHLfnbbx6wPcu26/x7epW77/o+J97/ovtW3pVn/9i35\n629Mj4L+BHj9X8E9r1z8a0H1R2gaN+176JfF9rHbfnbdxy7N6XX22L/i/tpz/13eL7PeXrtR35bz\n6PLP36U57+NlX2eoX2vzcwvdZxvzXH8ER+hmZgaLL+hXALcCnwUOtb/aAy+ll/lm03x9cC/UiL0v\n0vhnWdLtOGm5/96FYJ97iemz1C8SOuerDftvmkPt9rmOTLr7Up2ZqAM6tciCfi7wdqpF/TuAHwee\n0u6qX1/rZ76Zzb3f2vE+OnHZ3hdpnOE3nb/gn/FEy/33rsVt1bukqdMXCc3hwfNm+37Z7XNd2fk9\nNQRDbJrfIgv65cDngJPA14E/B17RQdOS3acOyFykDpg2yNsJBnlbDbJpgIb4PTXEpvktsqA/Frhj\n2/Z/1ueZmZnAIv/BRcufxV54/9nbdz4ceMQC+13QSd2um62rA6adVAc0WVcHJNbVAavhpDogcVId\n0KlFnlbzbOAI1Qwd4E3Ag8BvbrvM54AnLLAPM7MSfR54Yp87XKt3ug48nOrRhZYPipqZ2dC8BPgM\n1ZH4m8QtZmZmZmYG7V5c9Lv15z8OfNdAup4MfBj4X+AXBtL0E1S30b8BHwKePoCmV9RNx4GPAS8c\nQNOmZ1E9efpHemiCvbsCuJ/qtjoO/PIAmja7jgOfoJ9X0uzV9Ea2bqNbqO7DZT/1c6+mi4G/oxod\nfwK4csk9bZq+GXgP1d+/jwJPXXbQuVTjlnXgYeRz9B8C3lef/h7gI8uOatn1GOC7gTfTz4Lepuk5\nwIX16StY/m3VpumR204/rb68umnzcn8P/A3wo0tuatsVwF/30DJL00XAJ4FL6+2LB9C03cuAmwbQ\ndAR4S336YuC/WOxZgF00/RbwK/Xpb6fF7bToS//bvLjo5cC19emPUn2DHVhwv110fRm4uf58H9o0\nfZjqCA+q2+pSlqtN01e3nT4f+MoAmgB+HvhLqvuxD227+nwTrTZNrwbeTfU6ERjO/bfp1cA7B9B0\nJ/Co+vSjqBb0M+Kmp1C97SlUj1WuUx2INlp0QW/z4qLsMsteqIb4oqdZm17D1k82y9K26ZXAp4H3\nA68bQNNjqb75f7/e7uM9CNp0bQDPpfoR+X1Ub4mhbvo2YD/VwnAz8JMDaNp0HvCDVP/gqJv+gGqk\n8SWq++/1A2j6OFvjxMuBx7PH2rnojxRt/yLtPGpZ9l9AyZuM7GGWphcAPwN875JaNrVtur7+eD5w\nHdWPf8vSpultwGG23mK0j6PiNl3/CjwO+BrVM8CuB54kbnoY8AzgRVQL6IepRnmfFTZt+mHgH1n+\n6+/bNP0S1dgjqF47cyPwncADwqajwO+w9VjDceAbu11h0QX9i1TfwJsex9aPdk2XubQ+b5nadPWt\nbdPTqY4WrgDuHUjTpn+g+p55NNWPpKqmZ1L9iArVvPMlVD+2LnN+3aZr+1/+9wO/R3V0fI+w6Q6q\nMcv/1B8fpFqolrWgz/I99WMsf9wC7ZqeC/x6ffrzwO1UBy43C5seoDqw23Q78IUl9QDtXly0/UHR\nZ9PPg6KzvOjpCP08KNqm6Vup5mrP7qGnbdMT2DoCfkZ9eXXTdu+gn2e5tOk6wNZtdTnLf115m6Yn\nUz2Ydi7VEfotLHcU1Pb+u5DqoKCPtwFp0/TbwK/Wpw9QLa77xU0X1p8DeC1wzRJ7HpK9uOhn649N\nb68//3GqRWEIXQepjl7upzoS/g+qB/2UTX9I9U2++ZSuf15yT5umq6iexnWc6gj9WQNo2q6vBb1N\n189R3VYngH+in3+Y29xWb6R6psstLP8xkLZNPwX8WQ8tbZsuBt5LtUbdQvVgrbrpOfXnb6V6AsCF\nO7+AmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmRn/D/pEpkg0FKCeAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7fd9aefae490>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('Eel2017_trim30-min90-crop110-forwonly-filter100-deviation0.2_read_stats.csv', index_col=0)\n",
    "df['fraction'] = df['queries']/(df['total']*0.5)\n",
    "df.fraction.hist(bins=50)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Detailed information on what metaBEAT did to each sample is contained in the `log` file. It contains the exact commands that were run for each sample during each step of the process."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool\r\n",
      "version: v.0.97.11-global\r\n",
      "\r\n",
      "\r\n",
      "Sat Jul  4 02:12:47 2020\r\n",
      "\r\n",
      "/usr/bin/metaBEAT_global.py -Q Querymap_final.txt --trim_qual 30 --read_crop 110 --trim_minlength 90 --merge --product_length 106 --forward_only --length_filter 106 --length_deviation 0.2 -m 12S -o Eel2017_trim30-min90-crop110-forwonly-filter100-deviation0.2 -n 5 -v -@ nathangriffiths484@gmail.com\r\n",
      "\r\n",
      "\r\n",
      "metaBEAT may be querying NCBI's Entrez databases to fetch/verify taxonomic ids. Entrez User requirements state that you need to identify yourself by providing an email address so that NCBI can contact you in case there is a problem.\r\n",
      "\r\n",
      "You have specified: 'nathangriffiths484@gmail.com'\r\n",
      "\r\n",
      "\r\n",
      "Parsing querylist file\r\n",
      "\r\n",
      "Number of samples to process: 232\r\n",
      "Sequence input format: defaultdict(<type 'int'>, {'fastq': 232})\r\n",
      "Barcodes for demultiplexing provided for 0 samples\r\n",
      "Cropping instructions provided for 232 samples\r\n",
      "\r\n",
      "\r\n",
      "Sat Jul  4 02:12:47 2020\r\n",
      "\r\n",
      "\r\n",
      "##### processing query ID: 24-S1 #####\r\n",
      "\r\n",
      "\r\n",
      "### READ QUALITY TRIMMING ###\r\n",
      "\r\n",
      "\r\n",
      "trimming PE reads with trimmomatic\r\n",
      "java -jar /usr/bin/trimmomatic-0.32.jar PE -threads 5 -phred33 -trimlog trimmomatic.log /home/working/raw_reads/24-S1.R1.fastq.gz /home/working/raw_reads/24-S1.R2.fastq.gz 24-S1_forward.paired.fastq.gz 24-S1_forward.singletons.fastq.gz 24-S1_reverse.paired.fastq.gz 24-S1_reverse.singletons.fastq.gz ILLUMINACLIP:None:5:5:5 TRAILING:30 LEADING:30 SLIDINGWINDOW:5:30 CROP:110 MINLEN:90\r\n",
      "\r\n",
      "TrimmomaticPE: Started with arguments: -threads 5 -phred33 -trimlog trimmomatic.log /home/working/raw_reads/24-S1.R1.fastq.gz /home/working/raw_reads/24-S1.R2.fastq.gz 24-S1_forward.paired.fastq.gz 24-S1_forward.singletons.fastq.gz 24-S1_reverse.paired.fastq.gz 24-S1_reverse.singletons.fastq.gz ILLUMINACLIP:None:5:5:5 TRAILING:30 LEADING:30 SLIDINGWINDOW:5:30 CROP:110 MINLEN:90\r\n",
      "Jul 04, 2020 2:13:06 AM org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer makeIlluminaClippingTrimmer\r\n",
      "SEVERE: null\r\n",
      "java.io.FileNotFoundException: /home/working/1-trimming/24-S1/None (No such file or directory)\r\n",
      "\tat java.io.FileInputStream.open(Native Method)\r\n",
      "\tat java.io.FileInputStream.<init>(FileInputStream.java:146)\r\n",
      "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\r\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:107)\r\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:70)\r\n",
      "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:27)\r\n",
      "\tat org.usadellab.trimmomatic.TrimmomaticPE.run(TrimmomaticPE.java:495)\r\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:35)\r\n",
      "\r\n",
      "Input Read Pairs: 149662 Both Surviving: 97959 (65.45%) Forward Only Surviving: 22519 (15.05%) Reverse Only Surviving: 6250 (4.18%) Dropped: 22934 (15.32%)\r\n",
      "TrimmomaticPE: Completed successfully\r\n",
      "\r\n",
      "\r\n",
      "### READ CLIPPING ###\r\n",
      "\r\n",
      "Cropping the first 18 bases off reads in file: 24-S1_forward.paired.fastq.gz\r\n",
      "Cropping the first 18 bases off reads in file: 24-S1_forward.singletons.fastq.gz\r\n",
      "Cropping the first 18 bases off reads in file: 24-S1_reverse.paired.fastq.gz\r\n",
      "Cropping the first 18 bases off reads in file: 24-S1_reverse.singletons.fastq.gz\r\n",
      "\r\n",
      "### MERGING READ PAIRS ###\r\n",
      "\r\n",
      "merging paired-end reads with flash\r\n",
      "\r\n",
      "flash 24-S1_forward.paired.headclipped.fastq.gz 24-S1_reverse.paired.headclipped.fastq.gz -M 106 -t 5 -p 33 -o 24-S1 -z\r\n",
      "[FLASH] Starting FLASH v1.2.11\r\n",
      "[FLASH] Fast Length Adjustment of SHort reads\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Input files:\r\n",
      "[FLASH]     24-S1_forward.paired.headclipped.fastq.gz\r\n",
      "[FLASH]     24-S1_reverse.paired.headclipped.fastq.gz\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Output files:\r\n",
      "[FLASH]     ./24-S1.extendedFrags.fastq.gz\r\n",
      "[FLASH]     ./24-S1.notCombined_1.fastq.gz\r\n",
      "[FLASH]     ./24-S1.notCombined_2.fastq.gz\r\n",
      "[FLASH]     ./24-S1.hist\r\n",
      "[FLASH]     ./24-S1.histogram\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Parameters:\r\n",
      "[FLASH]     Min overlap:           10\r\n",
      "[FLASH]     Max overlap:           106\r\n",
      "[FLASH]     Max mismatch density:  0.250000\r\n",
      "[FLASH]     Allow \"outie\" pairs:   false\r\n",
      "[FLASH]     Cap mismatch quals:    false\r\n",
      "[FLASH]     Combiner threads:      5\r\n",
      "[FLASH]     Input format:          FASTQ, phred_offset=33\r\n",
      "[FLASH]     Output format:         FASTQ, phred_offset=33, gzip\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Starting reader and writer threads\r\n",
      "[FLASH] Starting 5 combiner threads\r\n",
      "[FLASH] Processed 25000 read pairs\r\n",
      "[FLASH] Processed 50000 read pairs\r\n",
      "[FLASH] Processed 75000 read pairs\r\n",
      "[FLASH] Processed 97959 read pairs\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Read combination statistics:\r\n",
      "[FLASH]     Total pairs:      97959\r\n",
      "[FLASH]     Combined pairs:   97743\r\n",
      "[FLASH]     Uncombined pairs: 216\r\n",
      "[FLASH]     Percent combined: 99.78%\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 100 log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The next steps in the processing will be chimera detection, and global clustering of the centroids from all clusters from all samples to produce denovo OTUs. The temporary files from the global clustering and the final OTU table were written to the directory `./GLOBAL`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Eel2017_trim30-min90-crop110-forwonly-filter100-deviation0.2-OTU-denovo.biom\r\n",
      "Eel2017_trim30-min90-crop110-forwonly-filter100-deviation0.2-OTU-denovo.tsv\r\n",
      "global_centroids.fasta\r\n",
      "global_queries.fasta\r\n",
      "global.uc\r\n"
     ]
    }
   ],
   "source": [
    "!ls GLOBAL/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The denovo OTU table (numbers are reads) can be viewed to see how OTUs are distributed across your samples. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Chimera detection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some stats on the read counts before/after trimming, merging etc. are summarised for you in `metaBEAT_read_stats.csv`.\n",
    "\n",
    "Next stage of the processing is chimera detection and removal of putative chimeric sequences. We'll do that using uchime as implemented in vsearch."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/1-trimming\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir 2-chimera_detection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/2-chimera_detection\n"
     ]
    }
   ],
   "source": [
    "cd 2-chimera_detection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Convert reference database from GenBank to fasta format to be used in chimera detection.\n",
    "\n",
    "Prepare `REFmap.txt` file, i.e. text file that specifies the location and the format of the reference to be used.\n",
    "The reference sequences in GenBank format are present in subdirectories for each vertebrate group in the `../Reference_database` directory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!echo '../supplementary_data/Reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb\\tgb\\n' \\\n",
    "'../supplementary_data/Reference_DBs/M.zebra.gb\\tgb'> REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!echo '../supplementary_data/Reference_DBs/12S_UKmammals_SATIVA_cleaned_June_2018.gb\\tgb\\n' \\\n",
    "'../supplementary_data/Reference_DBs/12S_UKamphibians_SATIVA_cleaned.gb\\tgb\\n' \\\n",
    "'../supplementary_data/Reference_DBs/12S_UKreptiles_SATIVA_cleaned.gb\\tgb\\n' \\\n",
    "'../supplementary_data/Reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb\\tgb\\n' \\\n",
    "'../supplementary_data/Reference_DBs/12S_UKbirds_SATIVA_cleaned.gb\\tgb\\n' \\\n",
    "'../supplementary_data/Reference_DBs/M.zebra.gb\\tgb'> REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../supplementary_data/Reference_DBs/12S_UKmammals_SATIVA_cleaned_June_2018.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/12S_UKamphibians_SATIVA_cleaned.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/12S_UKreptiles_SATIVA_cleaned.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/12S_UKbirds_SATIVA_cleaned.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/M.zebra.gb\tgb\r\n"
     ]
    }
   ],
   "source": [
    "!cat REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: metaBEAT.py [-h] [-Q <FILE>] [-B <FILE>] [--g_queries <FILE>] [-v] [-s]\r\n",
      "                   [-f] [-p] [-k] [-t] [-b] [-m <string>] [-n <INT>] [-E] [-e]\r\n",
      "                   [--read_stats_off] [--PCR_primer <FILE>] [--bc_dist <INT>]\r\n",
      "                   [--trim_adapter <FILE>] [--trim_qual <INT>] [--phred <INT>]\r\n",
      "                   [--trim_window <INT>] [--read_crop <INT>]\r\n",
      "                   [--trim_minlength <INT>] [--merge] [--product_length <INT>]\r\n",
      "                   [--merged_only] [--forward_only] [--length_filter <INT>]\r\n",
      "                   [--length_deviation <FLOAT>] [-R <FILE>] [--gb_out <FILE>]\r\n",
      "                   [--rec_check] [--gb_to_taxid <FILE>] [--cluster]\r\n",
      "                   [--clust_match <FLOAT>] [--clust_cov <INT>]\r\n",
      "                   [--blast_db <PATH>] [--blast_xml <PATH>]\r\n",
      "                   [--update_taxonomy] [--taxonomy_db <FILE>]\r\n",
      "                   [--min_ident <FLOAT>] [--min_ali_length <FLOAT>]\r\n",
      "                   [--bitscore_skim_LCA <FLOAT>] [--bitscore_skim_adjust_off]\r\n",
      "                   [--min_bit <INT>] [--refpkg <DIR>] [--jplace <FILE>]\r\n",
      "                   [--jellyfish_hash_size <STR>] [--kraken_db <DIR>]\r\n",
      "                   [--kraken_score_threshold <FLOAT>] [--rm_kraken_db]\r\n",
      "                   [-o OUTPUT_PREFIX] [--metadata METADATA]\r\n",
      "                   [-@ <email-address>] [--version]\r\n",
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool version:\r\n",
      "v.0.97.11-global\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -Q <FILE>, --querylist <FILE>\r\n",
      "                        file containing a list of query files\r\n",
      "  -B <FILE>, --BIOM_input <FILE>\r\n",
      "                        OTU table in BIOM format\r\n",
      "  --g_queries <FILE>    fasta file containing query sequences (in combination\r\n",
      "                        with '-B' sequence headers are expected to match)\r\n",
      "  -v, --verbose         turn verbose output on\r\n",
      "  -s, --seqinfo         write out seq_info.csv file\r\n",
      "  -f, --fasta           write out ref.fasta file\r\n",
      "  -p, --pplace          perform phylogenetic placement\r\n",
      "  -k, --kraken          perform phylogenetic placement\r\n",
      "  -t, --taxids          write out taxid.txt file\r\n",
      "  -b, --blast           compile local blast db and blast queries\r\n",
      "  -m <string>, --marker <string>\r\n",
      "                        marker ID (default: marker)\r\n",
      "  -n <INT>, --n_threads <INT>\r\n",
      "                        Number of threads (default: 1)\r\n",
      "  -E, --extract_centroid_reads\r\n",
      "                        extract centroid reads to files\r\n",
      "  -e, --extract_all_reads\r\n",
      "                        extract reads to files\r\n",
      "  --read_stats_off      ommit writing read stats to file\r\n",
      "  --version             show program's version number and exit\r\n",
      "\r\n",
      "Query preprocessing:\r\n",
      "  The parameters in this group affect how the query sequences are processed\r\n",
      "\r\n",
      "  --PCR_primer <FILE>   PCR primers (provided in fasta file) to be clipped\r\n",
      "                        from reads\r\n",
      "  --bc_dist <INT>       Number of mismatches allowed in barcode sequences\r\n",
      "  --trim_adapter <FILE>\r\n",
      "                        trim adapters provided in file\r\n",
      "  --trim_qual <INT>     minimum phred quality score (default: 30)\r\n",
      "  --phred <INT>         phred quality score offset - 33 or 64 (default: 33)\r\n",
      "  --trim_window <INT>   sliding window size (default: 5) for trimming; if\r\n",
      "                        average quality drops below the specified minimum\r\n",
      "                        quality all subsequent bases are removed from the\r\n",
      "                        reads\r\n",
      "  --read_crop <INT>     Crop reads to this length if they are longer than that\r\n",
      "                        (default: off)\r\n",
      "  --trim_minlength <INT>\r\n",
      "                        minimum length of reads to be retained after trimming\r\n",
      "                        (default: 50)\r\n",
      "  --merge               attempt to merge paired-end reads\r\n",
      "  --product_length <INT>\r\n",
      "                        estimated length of PCR product (specifying this\r\n",
      "                        option increases merging efficiency)\r\n",
      "  --merged_only         only process successfully merged read-pairs\r\n",
      "  --forward_only        only process sequences that contain forward reads\r\n",
      "                        (i.e. unmerged forward reads and merged reads)\r\n",
      "  --length_filter <INT>\r\n",
      "                        only process reads, which are within +/- 10 percent of\r\n",
      "                        this length\r\n",
      "  --length_deviation <FLOAT>\r\n",
      "                        allowed deviation (in percent) from length specified\r\n",
      "                        by --length_filter (default=0.1)\r\n",
      "\r\n",
      "Reference:\r\n",
      "  The parameters in this group affect the reference to be used in the\r\n",
      "  analyses\r\n",
      "\r\n",
      "  -R <FILE>, --REFlist <FILE>\r\n",
      "                        file containing a list of files to be used as\r\n",
      "                        reference sequences\r\n",
      "  --gb_out <FILE>       output the corrected gb file\r\n",
      "  --rec_check           check records to be used as reference\r\n",
      "  --gb_to_taxid <FILE>  comma delimited file containing 'gb accession,taxid'\r\n",
      "                        for a list of taxa\r\n",
      "\r\n",
      "Query clustering options:\r\n",
      "  The parameters in this group affect read clustering\r\n",
      "\r\n",
      "  --cluster             perform clustering of query sequences using vsearch\r\n",
      "  --clust_match <FLOAT>\r\n",
      "                        identity threshold for clustering in percent (default:\r\n",
      "                        1)\r\n",
      "  --clust_cov <INT>     minimum number of records in cluster (default: 1)\r\n",
      "\r\n",
      "BLAST search:\r\n",
      "  The parameters in this group affect BLAST search and BLAST based taxonomic\r\n",
      "  assignment\r\n",
      "\r\n",
      "  --blast_db <PATH>     path to precompiled blast database\r\n",
      "  --blast_xml <PATH>    path to Blast result in xml format\r\n",
      "  --update_taxonomy     Download/update taxonomy database. Database will be\r\n",
      "                        called 'taxonomy.db' and will be compiled in the same\r\n",
      "                        location as the metaBEAT.py script.\r\n",
      "  --taxonomy_db <FILE>  taxonomy database file location. In case it's not the\r\n",
      "                        default, which is 'taxonomy.db' in the same directory\r\n",
      "                        as the metaBEAT.py script.\r\n",
      "  --min_ident <FLOAT>   minimum identity threshold in percent (default: 0.80)\r\n",
      "  --min_ali_length <FLOAT>\r\n",
      "                        minimum alignment length in percent of total query\r\n",
      "                        length (default: 0.95)\r\n",
      "  --bitscore_skim_LCA <FLOAT>\r\n",
      "                        Only BLAST hits with bitscores differing by less than\r\n",
      "                        this factor from the top hit (bitscore skim window)\r\n",
      "                        will be considered for LCA (0-1; default: 0.1)\r\n",
      "  --bitscore_skim_adjust_off\r\n",
      "                        Per default a 100% identity BLAST top hit across the\r\n",
      "                        minimum alignment length triggers an adjustment of the\r\n",
      "                        bitscore skim window to '0', i.e. only hits with\r\n",
      "                        bitscores as good as the top hit are considered for\r\n",
      "                        LCA. This flag switches this behaviour off.\r\n",
      "  --min_bit <INT>       minimum bitscore (default: 80)\r\n",
      "\r\n",
      "Phylogenetic placement:\r\n",
      "  The parameters in this group affect phylogenetic placement\r\n",
      "\r\n",
      "  --refpkg <DIR>        PATH to refpkg for pplacer\r\n",
      "  --jplace <FILE>       phylogenetic placement result from prefious pplacer\r\n",
      "                        run in *.jplace format\r\n",
      "\r\n",
      "Kraken:\r\n",
      "  The parameters in this group affect taxonomic assignment using Kraken\r\n",
      "\r\n",
      "  --jellyfish_hash_size <STR>\r\n",
      "                        jellyfish hash size to control memory usage during\r\n",
      "                        kraken database building. A table size of '6400M' will\r\n",
      "                        require ~44G of RAM. '2700M' -> 20G RAM.\r\n",
      "  --kraken_db <DIR>     PATH to a Kraken database\r\n",
      "  --kraken_score_threshold <FLOAT>\r\n",
      "                        minimum proportion of k-mers to support assignment\r\n",
      "                        (0-1; default: 0)\r\n",
      "  --rm_kraken_db        Remove Kraken database after successful completion\r\n",
      "\r\n",
      "BIOM OUTPUT:\r\n",
      "  The arguments in this groups affect the output in BIOM format\r\n",
      "\r\n",
      "  -o OUTPUT_PREFIX, --output_prefix OUTPUT_PREFIX\r\n",
      "                        prefix for BIOM output files (default='metaBEAT')\r\n",
      "  --metadata METADATA   comma delimited file containing metadata (optional)\r\n",
      "\r\n",
      "Entrez identification:\r\n",
      "  metaBEAT is querying the NCBI Entrez databases, please provide an email\r\n",
      "  address for identification\r\n",
      "\r\n",
      "  -@ <email-address>, --email <email-address>\r\n",
      "                        provide your email address for identification to NCBI\r\n"
     ]
    }
   ],
   "source": [
    "!metaBEAT_global.py -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool\n",
      "version: v.0.97.11-global\n",
      "\n",
      "\n",
      "Sat Jul  4 12:13:54 2020\n",
      "\n",
      "/usr/bin/metaBEAT_global.py -R REFmap.txt -f -@ nathangriffiths484@gmail.com\n",
      "\n",
      "\n",
      "metaBEAT may be querying NCBI's Entrez databases to fetch/verify taxonomic ids. Entrez User requirements state that you need to identify yourself by providing an email address so that NCBI can contact you in case there is a problem.\n",
      "\n",
      "You have specified: 'nathangriffiths484@gmail.com'\n",
      "\n",
      "\n",
      "######## PROCESSING REFERENCE DATA ########\n",
      "\n",
      "\n",
      "processing ../supplementary_data/Reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb (containing 340 records)\n",
      "\n",
      "total number of valid records: 340\n",
      "\n",
      "\n",
      "processing ../supplementary_data/Reference_DBs/12S_UKbirds_SATIVA_cleaned.gb (containing 622 records)\n",
      "\n",
      "total number of valid records: 962\n",
      "\n",
      "\n",
      "processing ../supplementary_data/Reference_DBs/M.zebra.gb (containing 1 records)\n",
      "\n",
      "total number of valid records: 963\n",
      "\n",
      "\n",
      "processing ../supplementary_data/Reference_DBs/12S_UKmammals_SATIVA_cleaned_June_2018.gb (containing 1082 records)\n",
      "\n",
      "total number of valid records: 2045\n",
      "\n",
      "\n",
      "processing ../supplementary_data/Reference_DBs/12S_UKamphibians_SATIVA_cleaned.gb (containing 198 records)\n",
      "\n",
      "total number of valid records: 2243\n",
      "\n",
      "\n",
      "processing ../supplementary_data/Reference_DBs/12S_UKreptiles_SATIVA_cleaned.gb (containing 112 records)\n",
      "\n",
      "total number of valid records: 2355\n",
      "\n",
      "write out reference sequences to refs.fasta\n",
      "\n",
      "\n",
      "Sat Jul  4 12:13:55 2020\n",
      "\n",
      "\n",
      "### GLOBAL CLUSTERING ###\n",
      "\n",
      "\n",
      "Concatenating for global clustering\n",
      "\n",
      "\n",
      "Performing global clustering\n",
      "\n",
      "vsearch --cluster_fast /home/working/2-chimera_detection/GLOBAL/global_queries.fasta --id 1.00 --strand both --threads 1 --centroids global_centroids.fasta --uc global.uc\n",
      "vsearch v1.1.0_linux_x86_64, 31.4GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "\n",
      "\n",
      "parse vsearch uc file\n",
      "\n",
      "\n",
      "writing denovo OTU table\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/usr/bin/metaBEAT_global.py\", line 2575, in <module>\n",
      "    out.write(BIOM_tables_per_method['OTU_denovo'].to_tsv()) #to_json('generaged by test', direct_io=out)\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/biom/table.py\", line 4027, in to_tsv\n",
      "    observation_column_name)\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/biom/table.py\", line 1268, in delimited_self\n",
      "    raise TableException(\"Cannot delimit self if I don't have data...\")\n",
      "biom.exception.TableException: Cannot delimit self if I don't have data...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-R REFmap.txt \\\n",
    "-f \\\n",
    "-@ nathangriffiths484@gmail.com"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will produce `refs.fasta`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">KC292935|7962|Cyprinus carpio\r\n",
      "CCAGAAAAGAGAGATTTTAACTCTCACCCCTGGCTCCCAAAGCCAGAATTCTAAACTAAACTATTTTCTGGGGAGGTAGCACTCCCTTTATGGTATAGTACATATTATGCATAATATTACATTAATGTATTAGTACATATATGTATTATCACCAACTCACTATTTTAACCATAAAGCAGGTACATAATATTAAGGTGGGCATAAAGCATATTATTAAGACTCACAAATTCTATTATTTGGACTTGAGTAATATATTAATCCCCAAAAATTTGTCCTCAAATTTTTCCTTGAAATAATCAACTATAATTCCATTTAAACATATTAATGTAGTAAGAGACCACCAACCAGTTTATATAAAGGTATATCATGAATGATAGAATCAAGGACAATAATTGTGAGGGTTACACAATATGAACTATTACTGGCATCTGGTTCCTATTTCAGGGACATAACTGTAATATTCCACCCTCGGATAATTATACTGGCATCTGATTAATGGTGTAAGTACATACGTTTCATTACCCCCCATGCCGAGCATTCTTTTATATGCATAAGGTATTTTTTATTGGTTTCCTTTCATCTGGCATTTCAGAGTGCAGGCTCAAATGTTAAATCAAGGTTGAACATTTTCCTTGTATGTGATAATATATATTAATTATCGTAAGACATAATTTAAGAATTACATACTTTTATCTCAAGTGCATAATATATCTGTCTCTAGTTCAACTTATCCTTACATAGTGCCCCCTTTGGTTTTTGCGCGACAAACCCCCTTACCCCCTACGCTCAGCGAATCCTGTTATCCTTGTCAAACCCCGAAACCAAGGAGGACCCAAGAACGTGTAAACCAACGAGTTGAGGTATAAATTGGCATCCCATTATATATATATATATATATGCATCGGTTTTTTTAACCGCAACTTACCACTTACCTAAAAGCCCCTACCAAAAATCCCCAAAAAGAGGCTCGGCACTAAATACTCTAATATAATTAATCAGCTAGCGTAGCTTAACACAAAGCATAGCACTGAAGATGCTAAGATGAGTCCTAAAAAACTCCGCATGCACAAAGGCATGGTCCCGACCTTATTATCAGCTCTAACTCAACTTACACATGCAAGTCTCCGCAACCCAGTGAATATGCCCTCAATCCCCCCACCCGGGGACGAGGAGCGGGCATCAGGCACAAACATTAGCCCAAGACGCCTAGCCAAGCCACACCCCCAAGGGAATTCAGCAGTGATAAACATTAAGCCATAAGTGAAAACTTGACTCAGTTAGTGTTAAGAGGGCCGGTAAAACTCGTGCCAGCCACCGCGGTTAGACGAGAGGCCCTAGTTGATATTACAACGGCGTAAAGGGTGGTTAAGGATAAACAAAAATAAAGTCAAATGGCCCCTTGGCCGTCATACGCTTCTAGGAGTCCGAAGCCCTAATACGAAAGTAACTTAATAAACCCACC\r\n",
      ">AJ001798|98395|Cobitis taenia\r\n",
      "TACCCCACTATGTTCAGCCATAAACTTAGACGTTGTTATACAATAAACGTCCGCCAGGGTACTACGAGCATTAGCTTAAAACCCAAAGGACTTGGCGGTGCCTCAGACCCCCCTAGAGGAGCCTGTTCTAGAACCGATAACCCCCGTTAAACCTCACCACTTCTGGTCATCCCCGCCTATATACCGCCGTCGTCAGCTTACCCTGTGAAGGACTGATAGTAAGCAAAATGAATAAAATCCAAAACGTCAGGTCGAGGTGTAGCGAACGAAGTGGGAAGAAATGGGCTACATTTTCTTTAATAAGAATAATACGAACAACACTATGAAAACTAGTGCTCGAAGGAGGATTTAGTAGTAAAAAGGAAATAGAGTGTCCTTTTGAACCCGGCTCTGAG\r\n",
      ">KF856964|7962|Cyprinus carpio\r\n",
      "GCTAGCGTAGCTTAACACAAAGCATAGCACTGAAGATGCTAAGATGAGTCCTAAAAAACTCCGCATGCACAAAGGCATGGTCCCGACCTTATTATCAGCTCTAACTCAACTTACACATGCAAGTCTCCGCAACCCAGTGAATATGCCCTCAATCCCCCCACCCGGGGACGAGGAGCGGGCATCAGGCACAAACATTAGCCCAAGACGCCTAGCCAAGCCACACCCCCAAGGGAATTCAGCAGTGATAAACATTAAGCCATAAGTGAAAACTTGACCCAGTTAGTGTTAAGAGGGCCGGTAAAACTCGTGCCAGCCACCGCGGTTAGACGAGAGGCCCTAGTTGATATTACAACGGCGTAAAGGGTGGTTAAGGATAAACAAAAATAAAGTCAAATGGCCCCTTGGCCGTCATACGCTTCTAGGAGTCCGAAGCCCTAATACGAAAGTAACTTTAATAAACCCACCTGACCCCACGAAAGCTGAGAAACAAACTGGGATTAGATACCCCACTATGCTCAGCCGTAAACTCAGACATCCAGCTACAATTAGATGTCCGCCAGGGTACTACGAGCATTAGCTTAAAACCCAAAGGACCTGACGGTGTCTCAGACCCCCCTAGAGGAGCCTGTTCTAGAACCGATAACCCCCGTTCAACCTCACCACTTCTAGCCACCCCAGCCTATATACCGCCGTCGTCAGCTTACCCTGTGAAGGTAATAAAAGTAAGCAAAATGGGCACAACCCAAAACGTCAGGTCGAGGTGTAGCGCATGAAGTGGGAAGAAATGGGCTACATTTTCTAATATAGAATATTACGAACATGCACCATGAAACAATGCTTGAAGGAGGATTTAGTAGTAAAAGGGAAATAGAGTGTCCCTTTGAACCCGGCTCTGAGACGCGTACACACCGCCCGTCACTCTCCCCTGTCAAAACGCACCAAAAATACATAATACAACAGCACTGACAAGGGGAGGCAAGTCGTAACACGGTAAGTGTACCGGAAGGTGCACTTGGATCAAACCCAGGGTGTGGCTGAGTTAGTCAAGCATCTCACTTACACCGAGAAGACATCCATGCAAGTTGGATCGCCCTGAGCCAAACAGCTAGCTTAACTACTTAATAACTAAACAATATAAATAAAACAAAATAAGCCTAACACCAAAAACTAAACCATTTTTTTACCTGAGTATGGGAGACAGAAAAGGTTCCACAAAGCGATAGAAATAGTACCGCAAGGGAAAGCTGAAAGAGAAATGAAATAACCCATATAAGCACTAAAAAGCAAAGATTAAACCTCGTACCTTTTGCATCATGATTTAGCCAGTACACCCAAGCAAAGAGACCTTTAGTTTGAAACCCCGAAACCAGGTGAGCTACCCCGAGACAGCCTATTAAGGGCCAACCCGTCTCTGTGGCAAAAGAGTGGGAAGAGCTCCGGGTAGAAGTGACAGACCTACCGAACCTGGTGATAGCTGGTTGCCTAAGAAATGAATAGAAGTTCAGCCTCGTGCACCTCAAATCACACAAACAAAACAAGACTAAGAGAAACACATGAGAGTTAGTTAAAGGGGGTACAGCCCCTTTAACAAAGGACACAACCTTTCCAGGAGGATAAAGATCATAATACATAAAACATACTGTTCTAGTGGGCCTAAAAGCAGCCATCTAAACAGAAAGCGTTAAAGCTCAGACAGAAAAAAGTTTATTATCCCGATAAAAAATCTTACTCCCCTAAGTACTATTAGGCCAACCCATGCCCCCATGGAAGAGATTATGCTAAAATGAGTAACAAGAAGGCCCGCCCTTCTCCAAGCACAAGTGTAAGCCAAACCGGACCAACCATTGGCAACTAACGAACTCAATCAAAGAGAGCAATGTGAATTACAAAAAAACCAAGAAAAACCCACAACTAAGCTATCGTTACCCCCACACTGGAGTGCCCTAAAGGAAAGACTAAAAGAAAAGGAAGGAACTCGGCAAACACAAGCCTCGCCTGTTTACCAAAAACATCGCCTCCTGCAACGCAACCAAGTATAGGAGGTCCAGCCTGCCCAGTGACTACAAGTTCAACGGCCGCGGTATTTTGACCGTGCAAAGGTAGCGCAATCACTTGTCTTTTAAATAGAGACCTGTATGAATGGCTAAACGAGGGCTTAACTGTCTCCCCTTTCAAGTCAGTGAAATTGATCTACCCGTGCAGAAGCGGGTATAATACTACAAGACGAGAAGACCCTTTGGAGCTTAAGGTACAAAACTCAACCACGTTAAGCAACTCAATAAAAAGCAAAAACCTTGTGGACCATGAGATTTTACCTTCGGTTGGGGCGACCACGGAGGAAAGAAAAGCCTCCAGGTGGACTGGGAAAACCTCCTAAAACCAAGAGAGACATCTCTAAGCCACAGAACATCTGACCAAATATGATCCGGCTAACACATAGCCGATCAACGAACCAAGTTACCCTAGGGATAACAGCGCAATCCTCTCCCAGAGTCCATATCGACGAGGGGGTTTACGACCTCGATGTTGGATCAGGACATCCTAATGGTGCAGCCGCTATTAAGGGTTCGTTTGTTCAACGATTAAAGTCCTACGTGATCTGAGTTCAGACCGGAGCAATCCAGGTCAGTTTCTATCTGTAACGCTACTTTTCCTAGTACGAAAGGATCGGAAAAGAGGGGCCCATACTTAAAGCACGCCCCACCCCTAATTTATGAAAACAAATAAATAAAATAAAGGGAGAGCCAAAATCCCAGCTGGCCAAAATAAGGACATACTGGGGTGGCAGAGCATGGTAAATTGCGAAAGGCCTAAGCCCTTTTACCCAGAGGTTCAAATCCTCTTCCCAGTTTATGCTAAACATCCTAATAACTCACCTAATCAACCCTCTAGCCTATATCGTGCCTGTACTTCTAGCAGTAGCCTTCCTAACACTAATCGAACGAAAAGTACTAGGATATATGCAACTACGAAAAGGACCAAACGTGGTAGGCCCCTACGGACTACTACAACCCATCGCCGACGGAGTAAAACTCTTCATTAAAGAACCCGTCCGCCCATCCACATCATCCCCATTCCTATTCCTAGCCGCCCCAGTACTTGCATTAACCCTAGCCATAACCTTATGAGCACCAATACCTATACCCCACCCAGTAACTGACCTCAACCTAGGAATCCTATTTATTCTAGCCCTATCAAGCCTTGCGGTATACTCAATCCTAGGGTCAGGATGAGCATCAAATTCAAAATATGCATTAATCGGAGCCCTACGGGCCGTAGCCCAAACAATTTCGTATGAAGTTAGCCTAGGACTAATCCTCTTATCCGTAATTATCTTCTCAGGAGGATATACCCTACAAACATTTAACATCACCCAAGAAAGCATTTGACTTCTCATCCCTGCCTGACCTTTAGCCGCAATATGATATATCTCAACATTAGCCGAAACAAACCGAGCACCATTCGACCTAACAGAAGGAGAATCAGAACTGGTATCCGGCTTCAACGTAGAATATGCAGGAGGACCCTTCGCACTCTTCTTCCTAGCCGAATACGCCAACATCCTTCTAATAAATACCCTCTCAGCCGTCCTATTTTTAGGAGCCTCACATATTCCAAGCATACCAGAACTTACGACAATTAATCTCATAACTAAAGCCGCACTACTCTCTATTATATTCCTATGAGTACGAGCCTCGTACCCACGATTCCGATACGACCAACTAATACACTTAGTATGAAAAAACTTCCTTCCCCTCACACTCGCCTTTGTACTATGACACACTGCCCTACCAATTGCACTAGCAGGGCTCCCTCCACAACTATAATTAAGGAACTGTGCCTGAGCACCTAAGGACCACTTTGATAGAGTGAATTACAGGGGTTAAAATCCCCTCAGTTCCTAGAAAGAAGGGAATTGAACCCATACTCAAGAGATCAAAACTCTTGGTGCTTCCTTTACACCACTTTCTAAAGGCGGGGTCAGCTAATAAAGCTTTCGGGCCCATACCCCGAACATGACGGTTAAAGTCCCTCCTCCGCCAATGAACCCATATGTACTCGCAACCCTATTATCCAGCCTAGGGCTAGGAACCACCCTAACCTTTGCTAGCTCTCACTGACTCCTAGCTTGAATGGGCCTAGAAATTAATACGCTAGCAATCACCCCCCTAATGGCACAACATCACCACCCCCGTGCAGTAGAAGCAACTACAAAATATTTCTTAACCCAAGCCACCGCCGCGGCAATAATCCTGTTCGCAAGCACAACAAATGCCTGAATAACAGGAGAATGAAGCATCAATGACCTGTCGAACCCCATCGCCAGCACAATATTTATAGCTGCCTTAGCACTTAAAATTGGACTTGCACCAATACACTTCTGAATACCAGAGGTTCTACAAGGATTAGACCTGCTAACAGGACTCATCCTATCTACCTGACAAAAACTCGCCCCATTCGCGCTTATTATCCAAACAGCACAAACCATCGACCCACTGCTACTAACACTATTAGGAATTTCATCCACATTAGTTGGAGGATGAGGGGGATTAAATCAAACCCAACTACGAAAAATCCTAGCCTACTCTTCAATCGCACACATAGGATGAATAATTATTGTAATCCAATACGCCCCACAACTCACCCTAATTGCACTAGGAACATATATTATCATAACCTCCGCAGCATTCCTGACCCTAAAAATATCACTAACAACCAAAATCAGCACACTCGCAACAACCTGATCAAAAAGCCCTATCCTGACATCAACAACTGCCCTAGTCCTACTCTCACTAGGAGGCCTACCACCATTAACAGGATTTATACCAAAATGACTAATTCTACAAGAACTAACAAAACAAGACCTCCCAATTATTGCCACAGCCATAGCCCTAACCGCCCTAATTAGCCTATACTTCTACCTACGCCTATGCTACGCAATAACACTGACTATTTCCCCCAATATGATTAACTCAACAACCCCCTGACGAACCCAAACAACCCAAACCTCTCTACCCTTAGCCCTATTTACCACAGCCGCCCTCGGACTCCTACCAATAGCCCCAGCTATCCTAATACTAACCACCTAGGGATTTAGGATAACATTAGACCAAAAGCCTTCAAAGCTTTAAGTAGAAGTGAAAATCTTCTAATCCCTGATTAAGGCCTACAAGATATTAACTTGCATCTCCTGATTGCAAATCAGACACTTTTATTAAGCTAAGGCCTTACTAGATGGGAAGGCCTCGATCCTACAAACTTTTAGTTAACAGCTAAACGCTCAAGCCAGCGAGCATCCATCTACTTTTCCCGCCGTTTAACTCAGTAAGGCGGGAAAAGCCCCGGCAGAGTATTAATCTGCGTCTTCGGATTTGCAATCCAATGTGTTCTTCACCACGGAGCTGATAGGAAGAGGACTCAAACCTCTGTCTTCGGGGCTACAACCCACCGCCTAAGCACTCGGCTACCCTACCTGTGGCAATCACGCGCTGATTCTTCTCTACCAACCACAAAGACATTGGCACCCTTTATCTTGTATTTGGTGCCTGAGCCGGAATAGTAGGAACCGCCTTAAGCCTCCTCATTCGGGCCGAACTTAGCCAACCCGGGTCGCTTCTAGGTGATGACCAAATTTATAACGTTATCGTCACTGCCCACGCCTTTGTAATAATTTTCTTTATAGTAATGCCTATCCTTATTGGAGGATTTGGAAACTGACTTGTACCACTAATAATCGGAGCCCCAGACATAGCATTCCCACGAATAAATAACATAAGCTTCTGACTACTACCCCCATCATTCCTTCTACTCCTAGCTTCTTCTGGTGTTGAAGCTGGAGCTGGAACAGGATGAACCGTATACCCACCTCTTGCAGGGAACTTAGCCCACGCAGGAGCATCAGTAGACCTAACAATTTTCTCACTTCACCTAGCAGGTGTTTCATCAATTCTAGGGGCAATCAACTTTATTACTACAACCATCAACATGAAACCCCCAGCCATCTCTCAATACCAAACACCCCTGTTCGTCTGATCCGTGCTTGTAACCGCCGTATTGCTCCTTCTATCATTACCTGTTTTAGCCGCAGGAATTACAATGCTCCTAACAGACCGAAACCTTAATACCACATTCTTTGACCCGGCAGGAGGAGGAGACCCAATCCTTTATCAACACTTATTCTGATTCTTCGGCCACCCAGAAGTTTATATTCTCATCCTCCCAGGATTCGGTATCATCTCCCATGTTGTAGCCTATTATTCAGGCAAAAAAGAACCGTTCGGTTACATAGGAATGGTCTGAGCCATGATGGCCATCGGTCTACTAGGGTTTATTGTGTGAGCTCACCACATGTTCACTGTTGGAATAGACGTAGACACTCGAGCATATTTTACATCCGCAACAATAATTATCGCAATTCCAACAGGTGTAAAAGTATTTAGCTGATTAGCCACACTCCACGGAGGATCCATTAAATGAGAAACACCAATACTATGGGCTCTAGGGTTTATTTTCCTATTCACAGTAGGGGGACTCACAGGGATCGTCCTATCCAATTCATCACTTGATATTGTCCTTCACGACACTTACTACGTAGTAGCACACTTCCACTATGTACTATCAATGGGTGCTGTATTCGCTATTATGGCAGCCTTTGTGCATTGATTCCCCCTATTAACTGGGTATACTCTCCACAGCACCTGAACAAAAATCCACTTTGGAGTTATGTTTATTGGAGTTAACCTCACGTTCTTCCCACAACACTTCCTAGGCTTAGCCGGTATGCCACGGCGATACTCCGACTATCCAGATGCCTACGCCCTGTGAAATACAGTATCATCCATTGGATCGCTAATTTCCCTAGTAGCAGTAATCATATTCCTATTTATCCTATGAGAAGCCTTCGCCGCTAAACGAGAAGTACTATCTGTAGAACTAACAGCAACAAATGTAGAATGACTTCACGGCTGCCCTCCTCCTTATCACACATACGAGGAACCAGCATTCGTTCAAATTCAATCAAATTAACGAGAAAGGGAGGAATTGAACCCCCATATACTGGTTTCAAGCCAGCCACATAACCACTCTGTCACTTCCTTCTAAAGACATTAGTAAAACGTAAATTACATCACCTTGTCAAGGTGAAATCGTAGGTTAAATCCCTGCATGTCTTAACCCAAAGACTTAATGGCACACCCAACGCAACTAGGATTTCAAGACGCGGCATCACCCGTTATAGAAGAACTTCTTCACTTCCACGACCACGCATTAATAATTGTGCTCCTAATTAGCACTTTAGTGTTATATATTATTACTGCAATGGTATCAACTAAACTTACTAATAAATATATTCTAGACTCCCAAGAAATCGAAATCGTATGAACCATTCTACCAGCCGTCATTTTAGTACTAATCGCCCTGCCCTCCCTACGCATCCTGTACCTTATAGACGAAATTAACGACCCTCACCTGACAATTAAAGCAATAGGACACCAATGATACTGAAGTTACGAGTATACAGACTATGAAAATCTAGGATTCGACTCCTATATAGTACCAACCCAAGACCTTGCCCCCGGACAATTCCGACTTCTGGAAACAGACCACCGAATAGTTGTTCCAATAGAATCCCCAGTCCGTGTCCTAGTATCTGCTGAAGACGTGCTACATTCTTGAGCTGTTCCATCCCTTGGCGTAAAAATGGACGCAGTCCCAGGACGACTAAATCAAGCCGCCTTTATTGCCTCACGCCCAGGGGTGTTTTACGGACAATGCTCTGAAATTTGTGGAGCTAATCACAGCTTTATACCAATTGTAGTTGAAGCAGTACCTCTCGAACACTTCGAAAACTGATCCTCATTAATACTAGAAGACGCCTCGCTAGGAAGCTAAATATTGGACAAAGCGTTGGCCTTTTAAGCCAAAGATTGGTGATTCCCGACCACCTCTAGTGAAATGCCACAATTAAACCCCGGCCCTTGATTCGCAATTTTAGTATTTTCTTGACTAATTTTCTTAACTATTATTCCAACTAAAATCTTAAGCCATATTTCACCAAATGAACCAACCCCAGTAAGTGCTGAAAAACACAAAACTGAATCCTGAGACTGACCATGATAGTAAGCTTTTTCGACCAATTTGCAAGCCCATCATATCTAGGAATTCCCCTAATTGCTATCGCAATCGCACTACCCTGAGTACTTTACCCAACCCCACCAGCTCGATGAATTAATAACCGGCTCATCACGATCCAAGGATGATTTATCAACCGATTTACTAACCAACTAATACTACCACTAAATGTAGGAGGACATAAATGAGCACTACTACTAGCCTCATTAATAATTTTCCTAATTACAATTAATATGCTTGGCTTACTACCGTATACCTTCACACCCACAACACAACTATCACTCAATATGGGATTTGCCGTACCTCTATGACTCGCTACTGTAATTATTGGGATACGAAATCAACCAACAGTTGCCCTAGGACACCTACTACCAGAAGGAACACCCATTCCACTGATCCCAGTACTAATTATCATCGAAACAATTAGCCTATTTATCCGCCCATTAGCCCTGGGAGTCCGACTAACAGCCAATCTAACCGCAGGTCACCTGTTAATCCAACTCATCGCCACAGCTGTATTTGTTCTCCTACCAATAATACCAACAGTAGCAATCCTAACCGCTGCCGTACTCTTCCTCCTCACATTACTAGAAGTTGCAGTAGCAATAATCCAAGCTTATGTATTTGTACTTCTTCTAAGCCTATATCTACAAGAAAACGTTTAATGGCCCACCAAGCACATGCCTATCATATAGTTGATCCAAGCCCATGACCACTAACCGGAGCTATCGCTGCCCTCCTAATAACATCCGGCTTAGCAATCTGATTCCACTTCCACTCAACAACATTAATAACTCTAGGACTAATTCTCCTACTTCTTACCATATATCAATGATGACGTGACATTATCCGAGAAGGGACCTTCCAAGGCCACCACACACCGCCTGTACAAAAAGGGCTACGATATGGAATAATCCTATTTATCACCTCTGAAGTATTCTTCTTCCTAGGATTTTTCTGAGCCTTCTACCACTCAAGCCTAGCACCAACACCAGAATTAGGAGGATGCTGACCTCCCACAGGAATTACACCACTAGATCCCTTCGAAGTACCACTCCTTAATACAGCCGTTCTACTAGCATCAGGAGTCACAGTAACATGAGCTCATCACAGCATCATAGAGGGAGAACGAAAACAAGCTATTCAATCCTTAGCATTAACCATTTTACTAGGACTCTACTTCACCGCCCTCCAAGCCATGGAATACTATGAAGCACCCTTCACAATTGCAGATGGGGTTTACGGTTCAACATTCTTCGTAGCCACAGGTTTCCACGGTCTCCATGTTATTATTGGATCAACCTTCCTAGCAGTATGCCTTCTTCGCCAAATCCAATACCACTTTACATCTGAACACCATTTTGGCTTTGAAGCCGCTGCCTGATACTGACATTTTGTCGACGTAGTATGACTGTTCCTCTACGTATCCATCTATTGATGAGGCTCATAATCTTTCTAGTATCAAAGTTAGTACAAGTGACTTCCAATCACACAGTCTTGGTTAAACCCCAAGGAAAGATAATGAATCTAATTATAACTATTCTAACCATTACCGTGGCCCTATCCCTAATTCTAGCAACCGTATCCTTTTGACTACCACAAATAAATCCAGACGCAGAAAAACTATCACCATACGAATGCGGGTTTGATCCACTAGGATCTGCCCGACTACCATTTTCCCTACGCTTCTTCCTAGTAGCAATCCTATTCCTACTCTTCGATCTAGAAATTGCCCTCCTCCTCCCACTGCCCTGAGGAGACCAACTCCACAACCCCACCGGAACATTCTTCTGGGCCACAACAGTTCTAATTTTATTAACTCTAGGACTAATTTATGAATGAACACAAGGCGGCCTAGAATGAGCAGAATAGGGAGTTAGTCCAAAACAAGACCTCTGATTTCGGCTCAGAAAATCATGGTTTAATTCCATGACCCCCTTATGACACCCGTACATTTCAGCTTTAGCTCAGCATTCATTTTAGGTCTAATGGGACTAGCATTCCACCGAACCCACCTACTCTCCGCGCTACTATGCTTAGAAGGAATAATACTGTCCCTATTCATTGCACTAGCCCTATGAGCATTACAATTTGAATCCACAGGATTCTCAACAGCCCCTATACTACTACTGGCCTTTTCTGCTTGTGAAGCTAGCACCGGCTTAGCACTTTTAGTTGCTACCGCTCGCACTCACGGAACCGACCGGCTACAAAACCTTAATCTCCTACAATGCTAAAAGTACTAATCCCTACAATCATATTATTTCCAACAATTTGATTAACCTCCCCTAAATGACTGTGGACAACTACAACTGCACACAGCCTCTTAATTGCCTCCATCAGCCTAACATGACTGAAATGAACATCCGAAACTGGGTGAACTTCCTCCAACATATACCTAGCCACAGACCCACTATCAACCCCCCTATTAGTCCTAACATGCTGACTACTCCCACTTATAATCCTAGCCAGCCAAAACCATATTAATCCCGAACCAATTAGCCGGCAACGCCTATACATCACACTCCTCGCCTCACTACAAACCTTTCTGATCATAGCATTCGGTGCCACAGAAATCATTATATTCTACATCATATTTGAAGCCACACTCATTCCAACCCTCATTATTATTACCCGCTGAGGGAACCAAACCGAACGCCTTAACGCCGGAACCTACTTCTTATTCTATACTCTAGCAGGATCCCTTCCGCTTTTAGTCGCCCTACTCCTTCTTCAACAATCCACAGGTACACTATCAATACTTGTACTACAATATTCACAACCCCTACAACTCAACTCTTGAGGTCACATGTTTTGATGAGCTGGCTGCCTAATCGCATTCCTAGTCAAGATACCACTATATGGTGTCCACCTGTGATTACCAAAAGCACACGTAGAAGCCCCCGTAGCAGGATCGATAGTACTAGCAGCAGTATTACTAAAACTTGGTGGATACGGAATAATACGTATAATAGTAATACTAGATCCCCTATCAAAAGAACTCGCCTACCCATTTATTATCCTAGCCCTCTGAGGAATCATTATAACCGGATCAATTTGCCTCCGACAAACAGACCTAAAGTCACTAATCGCTTACTCATCCGTTAGCCACATAGGATTAGTAGCAGGAGGAATCCTAATTCAAACCCCATGAGGATTCTCAGGAGCAATCATCCTAATAATTGCTCACGGACTAGTATCCTCGGCACTTTTCTGCTTAGCCAACACAGCCTATGAACGAACCCATAGCCGAACAATAATTCTCGCCCGAGGCCTACAAGTAATTTTCCCACTAACTGCAGTATGATGGTTTATCGCAAATCTAGCTAACTTAGCACTCCCGCCTCTACCCAACCTAATAGGAGAACTTATAATCATCACAACATTATTCAATTGATCCCCCTGAACAATTTTACTCACTGGCCTGGGAACACTAATCACAGCTGGCTACTCCCTATACATGTTCCTCATGTCACAACGGGGCCCAACACCAAACCATATTATAGGCCTTCAACCATTCCACACCCGAGAACACCTACTAATAACTCTACACCTAATCCCTGTTATCCTCTTAGTAACAAAACCGGAACTCATGTGAGGATGATGCTATTGTAAGTATAGTTTAACTAAAACATTAGATTGTGATTCTAAAAACAGGGGTTAAAGTCCCCTTACTCACCAAGGAAGAACAGAAAATAGTAAGCACTGCTAATCCTTACGTTCCATGGTTAAAATCCGTGGCTTCCTTGCGCTTTCAAAGGATAACAGTTCATCCGTTGGTCTTAGGAACCAAAAACTCTTGGTGCAAATCCAAGTGGAAGCTAAAATGACACTAATTATACACTCATCACTCCTTCTAATCTTTTTTATCTTAGTTTACCCCCTACTCACCACATTAAACCCTAACCAGCAAGACTCCAACATAGCAGGTATAACCAAAATCGCCGTTAGCTCTGCATTCTTCGTCAGCCTCCTGCCCCTTATAATTTTCCTGAACCTAAAAACAGAAGGCATTATTACGAACTGACAATGAATAAACACCCAAACATTTGACGTAAACATCAGTTTCAAATTCGACCACTACTCCCTAATCTTCGTCCCAATCGCCCTATACGTCACCTGATCAATTTTAGAATTCGCACTATGATATATACACTCCGACCCCAACATTGACCGATTCTTCAAATACTTACTCACATTCCTAGTAGCCATAATTATCCTAGTTACAGCCAACAACATATTCCAACTATTTATCGGCTGAGAAGGCGTAGGCATCATATCATTCCTACTCATCGGATGATGACATGGACGAGCAGACGCCAACACCGCGGCTCTCCAAGCTGTTATTTACAACCGAGTAGGAGACATTGGACTAATTATAACTATAGCCTGACTAGCAATAAACCTTAATTCCTGAGAAATTCAACAAATCTTTGCCCTGTCAAAAAACTTTGATATAACAATTCCCCTAATAGGACTTGCCCTAGCGGCAACAGGAAAATCAGCCCAATTTGGCCTCCACCCATGACTCCCGTCCGCCATGGAGGGCCCTACACCAGTATCCGCCCTACTCCATTCAAGTACTATAGTCGTCGCAGGAATTTTCCTACTAATCCGCCTTCACCCCCTCATAGAAAACAATCAACTAGCCCTAACAACCTGCCTCTGCCTCGGAGCATTAACCTCACTATTTACAGCCACCTGCGCCCTAACCCAAAACGACATCAAAAAAATTGTAGCCTTCTCAACATCCAGCCAACTAGGACTAATAATAGTTACAATCGGACTAAACCAACCACAACTAGCATTCCTCCACATTTGCACCCATGCCTTCTTCAAGGCAATGCTCTTCCTGTGTTCAGGGTCAATCATCCACAGCCTAAATGACGAACAAGACATCCGAAAAATAGGAGGCCTATTCAACATTATGCCCGCCACCTCAACCTATTTTACAATTGGCAGCCTAGCCTTAACAGGAACCCCCTTCCTAGCAGGATTCTTCTCAAAGGACGCAATTATTGAAGCCCTAAACACCTCCTACCTAAACGCCTGAGCCCTAACCCTAACACTAATCGCTACATCATTCACCGCAGTATACAGTTTTCGACTAGTATACTTTGTAATTATAGGAACCCCACGATTCCTGCCCCTATCTCCAATCAACGAAAATAACCCACTAGTAATTAACTCCATCAAACGACTTGCCTGAGGAAGCATTATTGCAGGACTCATTATTACACAAAACTTCCCACCAATAAAAACACCAATCATAACAATATCAATCACTTTAAAAATAGCAGCCCTCATAGTAACAATTGCAGGCCTGCTAGTAGCCATAGAACTAGCCAATTTAACAAGCAAACAAGTAAAAATCACCCCAATAATCTCCACACACCACTTCTCAAACATATTAGGATTCTACCCTATAATTATCCACCGACTTATTCCAAAACTTAAACTTACTCTAGGACAATCAGCCGCCACCCAACTAGACAAAACATGACTCGAAACCGTTGGGCCGAAAGGTCTAGCACTCACACAAATAGCCATAGCAAAAATTACAAATGACATCACACGAGGAATGATCAAAACATACCTAACCATCTTCCTCCTAACTCTAATCCTGGCCACCATCCCCGTCCTCCTTTAAACCGCACGAAGAGCTCCACGACTCAGGCCCCGAGTAAGCTCCAACACAACAAGTAAGGTTAAAAGCAATACCCAAGCACAAATAACCAACATACCCCCACCAGACGAATACATTACAGCTACCCCACTAATATCCCCACGTAAAACAGAAAATTCCTTTAACCCATCTACAACCACCCATGAACCTTCGTATCAGCCCCCTCAAAAAATCCCTGCCACTAAACCAACTCCTAATAAGTAAACTAAAACATAACCCAACACAGAACGACTCCCCCAAGCTTCCGGGAAAGGCTCGGCGGCCAAAGCTGCCGAATAAGCAAAAACCACAAGCATCCCTCCTAAATAAATTAAGAAAAGAACCAAAGATAAAAAAGAACCTCCATGACCAACCAATACCCCACACCCAACCCCAGCTGCAACCACTAAACCAAGAGCAGCAAAATAAGGCGTAGGATTAGAAGCAACAGCAACTAAGCCCACAACCAAAGCTATTAATAACAAAAACATAAAATAGGTCATAATTCTTGCTCAGACTTTAACCGAGACCAATGACTTGAAGAACCACCGTTGTTATTCAACTACAAGAACCACTAATGGCAAGCCTACGAAAAACACACCCTCTCATTAAAATCGCTAACGACGCACTAGTTGACCTACCAACACCATCCAACATCTCAGCATGATGAAACTTTGGATCCCTCCTAGGACTATGCTTAATTACCCAAATTTTAACCGGCCTATTCCTAGCCATACACTACACCTCAGACATTTCAACCGCATTCTCATCTGTTACCCACATCTGCCGAGACGTAAATTACGGCTGACTAATCCGTAATGTACACGCCAACGGAGCATCATTCTTCTTCATTTGCATTTACATACACATCGCCCGAGGCCTATACTACGGATCATACCTCTACAAAGAAACCTGAAACATTGGTGTAGTCCTTCTACTACTAGTCATGATAACAGCCTTCGTTGGCTATGTTCTTCCATGAGGACAAATATCCTTTTGAGGCGCCACAGTAATCACAAACCTCCTATCTGCCGTACCATACATGGGAGACATGTTAGTCCAATGAATCTGAGGTGGGTTCTCAGTAGACAATGCAACACTAACACGATTCTTCGCATTCCACTTCCTACTACCATTTGTTATTGCCGCCGCAACCATCATCCACCTACTGTTCCTCCACGAAACAGGATCAAACAACCCGATCGGACTAAACTCAGACGCAGACAAAGTCTCTTTCCACCCGTACTTCTCATACAAAGACCTCCTTGGGTTCGTAATTATACTCCTAGCTCTTACACTACTAGCACTATTCTCCCCTAACTTACTAGGAGACCCAGAAAACTTCACCCCCGCAAACCCTCTAGTTACACCACCCCACATCAAACCAGAATGATACTTCCTATTTGCCTACGCCATCCTACGATCAATTCCAAACAAACTCGGAGGTGTCCTTGCACTCCTATTCTCCATTCTGGTATTAATAGTAGTACCACTACTACACACCTCAAAACAACGAGGACTAACATTCCGCCCCATCACCCAATTCCTATTCTGAACCCTAGTAGCGGACATAATTATCCTAACATGAATTGGAGGCATACCAGTAGAGCATCCCTTCATCATTATTGGACAAATTGCATCCGTCCTATACTTCGCACTATTCCTCATTTTTATGCCACTAGCAGGATGGTTAGAAAATAAAGCACTAAAATGAGCTTGCCCTAGTAGCTTAGCCTAAAAGCATCGGTCTTGTAATCCGAAGATCGGAGGTTAAATTCCTCCCTAGCGCCCAGAAAAGAGAGATTTTAACTCTCACCCCTGGCTCCCAAAGCCAGAATTCTAAACTAAACTATTTTCTGGGGAGGTAGCACTCCCTTTATGGTATAGTACATATTATGCATAATATTACATTAATGTATTAGTACATATATGTATTATCACCAACTCACTATTTTAACCATAAAGCAGGTACATAATATTAAGGTGGGCATAAAGCATATTATTAAGACTCACAAATTCTATTATTTGAACTTGAGTAATATATTAATCCCCAAAAATTTGTCCTCAAATTTTTCCTTGAAATAATCAACTATAATTCCATTCAAACATATTAATGTAGTAAGAGACCACCAACCAGTTTATATAAAGGCATATCATGAATGATAGAATCAAGGACAATAATTGTGAGGGTTACACAATATGAACTATTACTGGCATCTGGTTCCTATTTCAGGGACATAACTGTAATATTCCACCCTCGGATAATTATACTGGCATCTGATTAATGGTGTAGTACATACATTTCATTACCCCCCATGCCGAGCATTCTTTTATATGCATAGGGTATTTTTTATTGGTTTCCTTTCATCTGGCATTTCAGAGTGCAGGCTCAAATGTTAAATTAAGGTTGAACATTTTTCTTGTATGTGATAATATATATTAATTATCGTAAGACATAATTTAAGAATTACATACTTTTATCTCAAGTGCATAATATATCTGTCTCTAGTTCAACTTATCCTTACATAGTGCCCCCTTTGGTTTTTGCGCGACAAACCCCCTTACCCCCTACGCTCAGAGAATCCTGTTATCCTTGTCAAACCCCGAAACCAAGGAGGACCCAAGAACGTGTAAACCAACGAGTTGAGGTATAAATTGGCATCCCATTATATATATATATATATATGCATCGCTTTTTTTAACCGCAACTTACCACTTACCTAAAAGTCCCTACCAAAAATCCCCAAAAAGAGGCTCGGCACTAAATACTCTAATATAATTAATCA\r\n",
      ">KJ627918|134920|Pungitius pungitius\r\n",
      "AAACTGGGATTAGATACCCCACTATGCCTAGCCCTAAACATTGATAGAATTTTACAACCTCTATCCGCCCGGGGACTACAAGCATCAGCTTGAAACCCAAAGGACTTGGCGGTGCTTTAGATCCACCTAGAGGAGCCTGTTCTAGAACCGATAACCCCCGTTCAACCTCACCCCTCTTTGTTAATACCGCCTATATACCACCGTCGTCAGCTTACCCTGTGAGGGATCAATAGTAAGCTTAACTGGTACAACCCTAAACGTCAGGTCGAGGTGTAGCGTATGGAGGGGGAAGAAATGGGCTACATTCGCTACAACAGCGAACACGAATGATGTCCTGAAACGTACATCTGAAGGAGGATTTAGCAGTAAGTAGAAAATAGAGTGTTCTACTGAAATTGGCCCTGAAGCGCGCACACACCGCC\r\n",
      ">HM177473|8010|Esox lucius\r\n",
      "TACAATTTTATCCGCCAGGGAACTACAAGCATCAGCTTAAAACCCAAAGGACTTGGCGGTGCTTCAGACCCACCTAGAGGAGCCTGTTCTATAACCGATAACCCCCGCTTAACCTCACCAGCCCTTGTTCACCCCGCCTATATACCACCGTCGCCAGCTTATCCTGTGAAGGTCTCATAATAAGCAAAATGGGTAAAACCCAGAACGTCAGGTCGAGGTGTAGCGCATGAGCTGGGAAGAAATGGGCTACATTTTCTAATTCAGAATATACGAACTGCACTGTGAAACCAGTGCCCAAAGGTGGATTTAGCAGTAAACAGAAAATAGAGTGTTCTCTTGAAACTGGCTCTGAAGCGCGCACACACCGCCCGTCACCCTAA\r\n"
     ]
    }
   ],
   "source": [
    "!head refs.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now run chimera detection."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "### Detecting chimeras in 24-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 24-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 24-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 24-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 24-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 24-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 25-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 25-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 25-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 25-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 25-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 25-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 26-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 26-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 26-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 26-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 26-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in 26-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL3-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL3-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL3-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL3-S3 ###\n",
      "\n",
      "BL3-S4 is empty\n",
      "\n",
      "### Detecting chimeras in BL3-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL4-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL4-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL4-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL4-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL4-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in BL4-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in C-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in C-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in C-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in C-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in C-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in D-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in D-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in D-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in D-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in D-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in D-B1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-06 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-07 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-08 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-09 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-10 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LB-B1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LH-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LH-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LH-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LH-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in LH-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N+C-B1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in neg1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in neg2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in neg3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in neg4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in neg5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-06 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-07 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-08 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-09 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-10 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in PH-B1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pos1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pos2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pos3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pos4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pos5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in S-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in S-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in S-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in S-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in S-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG10-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG10-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG10-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG10-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG10-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG10-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG1-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG1-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG1-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG1-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG1-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG1-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG21-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG21-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG21-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG21-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG21-S4 ###\n",
      "\n",
      "SG21-S5 is empty\n",
      "\n",
      "### Detecting chimeras in SG22-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG22-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG22-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG22-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG22-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG22-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG23-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG23-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG23-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG23-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG23-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG23-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG2-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG2-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG2-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG2-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG2-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG2-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG5-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG5-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG5-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG5-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG5-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG5-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG6-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG6-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG6-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG6-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG6-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG6-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG7-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG7-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG7-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG7-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG7-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG7-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG8-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG8-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG8-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG8-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG8-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG8-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG9-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG9-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG9-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG9-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG9-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in SG9-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T-B1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-01 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-02 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-03 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-04 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-05 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-06 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-07 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-08 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-09 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-10 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W11-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W11-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W11-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W11-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W11-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W11-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W13-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W13-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W13-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W13-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W13-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W13-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W14-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W14-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W14-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W14-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W14-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W14-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W15-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W15-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W15-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W15-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W15-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W15-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W16-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W16-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W16-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W16-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W16-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W16-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W17-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W17-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W17-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W17-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W17-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W17-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W18-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W18-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W18-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W18-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W18-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W18-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W19-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W19-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W19-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W19-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W19-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W19-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W20-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W20-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W20-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W20-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W20-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W20-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W27-blank ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W27-S1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W27-S2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W27-S3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W27-S4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W27-S5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in W-B1 ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "\n",
    "for a in $(cut -f 1 ../1-trimming/Querymap.txt)\n",
    "do\n",
    "    if [ -s ../1-trimming/$a/$a\\_trimmed.fasta ]\n",
    "    then\n",
    "        echo -e \"\\n### Detecting chimeras in $a ###\\n\"\n",
    "        mkdir $a\n",
    "        cd $a\n",
    "        vsearch --uchime_ref ../../1-trimming/$a/$a\\_trimmed.fasta --db ../refs.fasta \\\n",
    "        --nonchimeras $a-nonchimeras.fasta --chimeras $a-chimeras.fasta &> log \n",
    "        cd ..\n",
    "\n",
    "    else\n",
    "        echo -e \"$a is empty\"\n",
    "    fi\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clustering and taxonomic assignment against UK fish database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/4-taxonomic_assignment_vertebrates\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘3-taxonomic_assignment_fish’: File exists\r\n"
     ]
    }
   ],
   "source": [
    "!mkdir 3-taxonomic_assignment_fish"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/3-taxonomic_assignment_fish\n"
     ]
    }
   ],
   "source": [
    "cd 3-taxonomic_assignment_fish/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Produce the text file containing the fish reference sequences using the command line - we call it `REFmap.txt`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!echo '../supplementary_data/Reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb\\tgb\\n' \\\n",
    "'../supplementary_data/Reference_DBs/M.zebra.gb\\tgb'> REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../supplementary_data/Reference_DBs/12S_UKmammals_SATIVA_cleaned_June_2018.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/12S_UKamphibians_SATIVA_cleaned.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/12S_UKreptiles_SATIVA_cleaned.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/12S_UKbirds_SATIVA_cleaned.gb\tgb\r\n",
      " ../supplementary_data/Reference_DBs/M.zebra.gb\tgb\r\n"
     ]
    }
   ],
   "source": [
    "!cat REFmap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Produce the text file containing non-chimera query sequences - `Querymap.txt`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Querymap\n",
    "for a in $(ls -l ../2-chimera_detection/ | grep \"^d\" | perl -ne 'chomp; @a=split(\" \"); print \"$a[-1]\\n\"')\n",
    "do\n",
    "    echo -e \"$a-nc\\tfasta\\t../2-chimera_detection/$a/$a-nonchimeras.fasta\"\n",
    "done > Querymap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24-blank-nc\tfasta\t../2-chimera_detection/24-blank/24-blank-nonchimeras.fasta\r\n",
      "24-S1-nc\tfasta\t../2-chimera_detection/24-S1/24-S1-nonchimeras.fasta\r\n",
      "24-S2-nc\tfasta\t../2-chimera_detection/24-S2/24-S2-nonchimeras.fasta\r\n",
      "24-S3-nc\tfasta\t../2-chimera_detection/24-S3/24-S3-nonchimeras.fasta\r\n",
      "24-S4-nc\tfasta\t../2-chimera_detection/24-S4/24-S4-nonchimeras.fasta\r\n",
      "24-S5-nc\tfasta\t../2-chimera_detection/24-S5/24-S5-nonchimeras.fasta\r\n",
      "25-blank-nc\tfasta\t../2-chimera_detection/25-blank/25-blank-nonchimeras.fasta\r\n",
      "25-S1-nc\tfasta\t../2-chimera_detection/25-S1/25-S1-nonchimeras.fasta\r\n",
      "25-S2-nc\tfasta\t../2-chimera_detection/25-S2/25-S2-nonchimeras.fasta\r\n",
      "25-S3-nc\tfasta\t../2-chimera_detection/25-S3/25-S3-nonchimeras.fasta\r\n",
      "25-S4-nc\tfasta\t../2-chimera_detection/25-S4/25-S4-nonchimeras.fasta\r\n",
      "25-S5-nc\tfasta\t../2-chimera_detection/25-S5/25-S5-nonchimeras.fasta\r\n",
      "26-blank-nc\tfasta\t../2-chimera_detection/26-blank/26-blank-nonchimeras.fasta\r\n",
      "26-S1-nc\tfasta\t../2-chimera_detection/26-S1/26-S1-nonchimeras.fasta\r\n",
      "26-S2-nc\tfasta\t../2-chimera_detection/26-S2/26-S2-nonchimeras.fasta\r\n",
      "26-S3-nc\tfasta\t../2-chimera_detection/26-S3/26-S3-nonchimeras.fasta\r\n",
      "26-S4-nc\tfasta\t../2-chimera_detection/26-S4/26-S4-nonchimeras.fasta\r\n",
      "26-S5-nc\tfasta\t../2-chimera_detection/26-S5/26-S5-nonchimeras.fasta\r\n",
      "BL3-blank-nc\tfasta\t../2-chimera_detection/BL3-blank/BL3-blank-nonchimeras.fasta\r\n",
      "BL3-S1-nc\tfasta\t../2-chimera_detection/BL3-S1/BL3-S1-nonchimeras.fasta\r\n",
      "BL3-S2-nc\tfasta\t../2-chimera_detection/BL3-S2/BL3-S2-nonchimeras.fasta\r\n",
      "BL3-S3-nc\tfasta\t../2-chimera_detection/BL3-S3/BL3-S3-nonchimeras.fasta\r\n",
      "BL3-S5-nc\tfasta\t../2-chimera_detection/BL3-S5/BL3-S5-nonchimeras.fasta\r\n",
      "BL4-blank-nc\tfasta\t../2-chimera_detection/BL4-blank/BL4-blank-nonchimeras.fasta\r\n",
      "BL4-S1-nc\tfasta\t../2-chimera_detection/BL4-S1/BL4-S1-nonchimeras.fasta\r\n",
      "BL4-S2-nc\tfasta\t../2-chimera_detection/BL4-S2/BL4-S2-nonchimeras.fasta\r\n",
      "BL4-S3-nc\tfasta\t../2-chimera_detection/BL4-S3/BL4-S3-nonchimeras.fasta\r\n",
      "BL4-S4-nc\tfasta\t../2-chimera_detection/BL4-S4/BL4-S4-nonchimeras.fasta\r\n",
      "BL4-S5-nc\tfasta\t../2-chimera_detection/BL4-S5/BL4-S5-nonchimeras.fasta\r\n",
      "C-01-nc\tfasta\t../2-chimera_detection/C-01/C-01-nonchimeras.fasta\r\n",
      "C-02-nc\tfasta\t../2-chimera_detection/C-02/C-02-nonchimeras.fasta\r\n",
      "C-03-nc\tfasta\t../2-chimera_detection/C-03/C-03-nonchimeras.fasta\r\n",
      "C-04-nc\tfasta\t../2-chimera_detection/C-04/C-04-nonchimeras.fasta\r\n",
      "C-05-nc\tfasta\t../2-chimera_detection/C-05/C-05-nonchimeras.fasta\r\n",
      "D-01-nc\tfasta\t../2-chimera_detection/D-01/D-01-nonchimeras.fasta\r\n",
      "D-02-nc\tfasta\t../2-chimera_detection/D-02/D-02-nonchimeras.fasta\r\n",
      "D-03-nc\tfasta\t../2-chimera_detection/D-03/D-03-nonchimeras.fasta\r\n",
      "D-04-nc\tfasta\t../2-chimera_detection/D-04/D-04-nonchimeras.fasta\r\n",
      "D-05-nc\tfasta\t../2-chimera_detection/D-05/D-05-nonchimeras.fasta\r\n",
      "D-B1-nc\tfasta\t../2-chimera_detection/D-B1/D-B1-nonchimeras.fasta\r\n",
      "GLOBAL-nc\tfasta\t../2-chimera_detection/GLOBAL/GLOBAL-nonchimeras.fasta\r\n",
      "LB-01-nc\tfasta\t../2-chimera_detection/LB-01/LB-01-nonchimeras.fasta\r\n",
      "LB-02-nc\tfasta\t../2-chimera_detection/LB-02/LB-02-nonchimeras.fasta\r\n",
      "LB-03-nc\tfasta\t../2-chimera_detection/LB-03/LB-03-nonchimeras.fasta\r\n",
      "LB-04-nc\tfasta\t../2-chimera_detection/LB-04/LB-04-nonchimeras.fasta\r\n",
      "LB-05-nc\tfasta\t../2-chimera_detection/LB-05/LB-05-nonchimeras.fasta\r\n",
      "LB-06-nc\tfasta\t../2-chimera_detection/LB-06/LB-06-nonchimeras.fasta\r\n",
      "LB-07-nc\tfasta\t../2-chimera_detection/LB-07/LB-07-nonchimeras.fasta\r\n",
      "LB-08-nc\tfasta\t../2-chimera_detection/LB-08/LB-08-nonchimeras.fasta\r\n",
      "LB-09-nc\tfasta\t../2-chimera_detection/LB-09/LB-09-nonchimeras.fasta\r\n",
      "LB-10-nc\tfasta\t../2-chimera_detection/LB-10/LB-10-nonchimeras.fasta\r\n",
      "LB-B1-nc\tfasta\t../2-chimera_detection/LB-B1/LB-B1-nonchimeras.fasta\r\n",
      "LH-01-nc\tfasta\t../2-chimera_detection/LH-01/LH-01-nonchimeras.fasta\r\n",
      "LH-02-nc\tfasta\t../2-chimera_detection/LH-02/LH-02-nonchimeras.fasta\r\n",
      "LH-03-nc\tfasta\t../2-chimera_detection/LH-03/LH-03-nonchimeras.fasta\r\n",
      "LH-04-nc\tfasta\t../2-chimera_detection/LH-04/LH-04-nonchimeras.fasta\r\n",
      "LH-05-nc\tfasta\t../2-chimera_detection/LH-05/LH-05-nonchimeras.fasta\r\n",
      "N-01-nc\tfasta\t../2-chimera_detection/N-01/N-01-nonchimeras.fasta\r\n",
      "N-02-nc\tfasta\t../2-chimera_detection/N-02/N-02-nonchimeras.fasta\r\n",
      "N-03-nc\tfasta\t../2-chimera_detection/N-03/N-03-nonchimeras.fasta\r\n",
      "N-04-nc\tfasta\t../2-chimera_detection/N-04/N-04-nonchimeras.fasta\r\n",
      "N-05-nc\tfasta\t../2-chimera_detection/N-05/N-05-nonchimeras.fasta\r\n",
      "N+C-B1-nc\tfasta\t../2-chimera_detection/N+C-B1/N+C-B1-nonchimeras.fasta\r\n",
      "neg1-nc\tfasta\t../2-chimera_detection/neg1/neg1-nonchimeras.fasta\r\n",
      "neg2-nc\tfasta\t../2-chimera_detection/neg2/neg2-nonchimeras.fasta\r\n",
      "neg3-nc\tfasta\t../2-chimera_detection/neg3/neg3-nonchimeras.fasta\r\n",
      "neg4-nc\tfasta\t../2-chimera_detection/neg4/neg4-nonchimeras.fasta\r\n",
      "neg5-nc\tfasta\t../2-chimera_detection/neg5/neg5-nonchimeras.fasta\r\n",
      "PH-01-nc\tfasta\t../2-chimera_detection/PH-01/PH-01-nonchimeras.fasta\r\n",
      "PH-02-nc\tfasta\t../2-chimera_detection/PH-02/PH-02-nonchimeras.fasta\r\n",
      "PH-03-nc\tfasta\t../2-chimera_detection/PH-03/PH-03-nonchimeras.fasta\r\n",
      "PH-04-nc\tfasta\t../2-chimera_detection/PH-04/PH-04-nonchimeras.fasta\r\n",
      "PH-05-nc\tfasta\t../2-chimera_detection/PH-05/PH-05-nonchimeras.fasta\r\n",
      "PH-06-nc\tfasta\t../2-chimera_detection/PH-06/PH-06-nonchimeras.fasta\r\n",
      "PH-07-nc\tfasta\t../2-chimera_detection/PH-07/PH-07-nonchimeras.fasta\r\n",
      "PH-08-nc\tfasta\t../2-chimera_detection/PH-08/PH-08-nonchimeras.fasta\r\n",
      "PH-09-nc\tfasta\t../2-chimera_detection/PH-09/PH-09-nonchimeras.fasta\r\n",
      "PH-10-nc\tfasta\t../2-chimera_detection/PH-10/PH-10-nonchimeras.fasta\r\n",
      "PH-B1-nc\tfasta\t../2-chimera_detection/PH-B1/PH-B1-nonchimeras.fasta\r\n",
      "pos1-nc\tfasta\t../2-chimera_detection/pos1/pos1-nonchimeras.fasta\r\n",
      "pos2-nc\tfasta\t../2-chimera_detection/pos2/pos2-nonchimeras.fasta\r\n",
      "pos3-nc\tfasta\t../2-chimera_detection/pos3/pos3-nonchimeras.fasta\r\n",
      "pos4-nc\tfasta\t../2-chimera_detection/pos4/pos4-nonchimeras.fasta\r\n",
      "pos5-nc\tfasta\t../2-chimera_detection/pos5/pos5-nonchimeras.fasta\r\n",
      "S-01-nc\tfasta\t../2-chimera_detection/S-01/S-01-nonchimeras.fasta\r\n",
      "S-02-nc\tfasta\t../2-chimera_detection/S-02/S-02-nonchimeras.fasta\r\n",
      "S-03-nc\tfasta\t../2-chimera_detection/S-03/S-03-nonchimeras.fasta\r\n",
      "S-04-nc\tfasta\t../2-chimera_detection/S-04/S-04-nonchimeras.fasta\r\n",
      "S-05-nc\tfasta\t../2-chimera_detection/S-05/S-05-nonchimeras.fasta\r\n",
      "SG10-blank-nc\tfasta\t../2-chimera_detection/SG10-blank/SG10-blank-nonchimeras.fasta\r\n",
      "SG10-S1-nc\tfasta\t../2-chimera_detection/SG10-S1/SG10-S1-nonchimeras.fasta\r\n",
      "SG10-S2-nc\tfasta\t../2-chimera_detection/SG10-S2/SG10-S2-nonchimeras.fasta\r\n",
      "SG10-S3-nc\tfasta\t../2-chimera_detection/SG10-S3/SG10-S3-nonchimeras.fasta\r\n",
      "SG10-S4-nc\tfasta\t../2-chimera_detection/SG10-S4/SG10-S4-nonchimeras.fasta\r\n",
      "SG10-S5-nc\tfasta\t../2-chimera_detection/SG10-S5/SG10-S5-nonchimeras.fasta\r\n",
      "SG1-blank-nc\tfasta\t../2-chimera_detection/SG1-blank/SG1-blank-nonchimeras.fasta\r\n",
      "SG1-S1-nc\tfasta\t../2-chimera_detection/SG1-S1/SG1-S1-nonchimeras.fasta\r\n",
      "SG1-S2-nc\tfasta\t../2-chimera_detection/SG1-S2/SG1-S2-nonchimeras.fasta\r\n",
      "SG1-S3-nc\tfasta\t../2-chimera_detection/SG1-S3/SG1-S3-nonchimeras.fasta\r\n",
      "SG1-S4-nc\tfasta\t../2-chimera_detection/SG1-S4/SG1-S4-nonchimeras.fasta\r\n",
      "SG1-S5-nc\tfasta\t../2-chimera_detection/SG1-S5/SG1-S5-nonchimeras.fasta\r\n",
      "SG21-blank-nc\tfasta\t../2-chimera_detection/SG21-blank/SG21-blank-nonchimeras.fasta\r\n",
      "SG21-S1-nc\tfasta\t../2-chimera_detection/SG21-S1/SG21-S1-nonchimeras.fasta\r\n",
      "SG21-S2-nc\tfasta\t../2-chimera_detection/SG21-S2/SG21-S2-nonchimeras.fasta\r\n",
      "SG21-S3-nc\tfasta\t../2-chimera_detection/SG21-S3/SG21-S3-nonchimeras.fasta\r\n",
      "SG21-S4-nc\tfasta\t../2-chimera_detection/SG21-S4/SG21-S4-nonchimeras.fasta\r\n",
      "SG22-blank-nc\tfasta\t../2-chimera_detection/SG22-blank/SG22-blank-nonchimeras.fasta\r\n",
      "SG22-S1-nc\tfasta\t../2-chimera_detection/SG22-S1/SG22-S1-nonchimeras.fasta\r\n",
      "SG22-S2-nc\tfasta\t../2-chimera_detection/SG22-S2/SG22-S2-nonchimeras.fasta\r\n",
      "SG22-S3-nc\tfasta\t../2-chimera_detection/SG22-S3/SG22-S3-nonchimeras.fasta\r\n",
      "SG22-S4-nc\tfasta\t../2-chimera_detection/SG22-S4/SG22-S4-nonchimeras.fasta\r\n",
      "SG22-S5-nc\tfasta\t../2-chimera_detection/SG22-S5/SG22-S5-nonchimeras.fasta\r\n",
      "SG23-blank-nc\tfasta\t../2-chimera_detection/SG23-blank/SG23-blank-nonchimeras.fasta\r\n",
      "SG23-S1-nc\tfasta\t../2-chimera_detection/SG23-S1/SG23-S1-nonchimeras.fasta\r\n",
      "SG23-S2-nc\tfasta\t../2-chimera_detection/SG23-S2/SG23-S2-nonchimeras.fasta\r\n",
      "SG23-S3-nc\tfasta\t../2-chimera_detection/SG23-S3/SG23-S3-nonchimeras.fasta\r\n",
      "SG23-S4-nc\tfasta\t../2-chimera_detection/SG23-S4/SG23-S4-nonchimeras.fasta\r\n",
      "SG23-S5-nc\tfasta\t../2-chimera_detection/SG23-S5/SG23-S5-nonchimeras.fasta\r\n",
      "SG2-blank-nc\tfasta\t../2-chimera_detection/SG2-blank/SG2-blank-nonchimeras.fasta\r\n",
      "SG2-S1-nc\tfasta\t../2-chimera_detection/SG2-S1/SG2-S1-nonchimeras.fasta\r\n",
      "SG2-S2-nc\tfasta\t../2-chimera_detection/SG2-S2/SG2-S2-nonchimeras.fasta\r\n",
      "SG2-S3-nc\tfasta\t../2-chimera_detection/SG2-S3/SG2-S3-nonchimeras.fasta\r\n",
      "SG2-S4-nc\tfasta\t../2-chimera_detection/SG2-S4/SG2-S4-nonchimeras.fasta\r\n",
      "SG2-S5-nc\tfasta\t../2-chimera_detection/SG2-S5/SG2-S5-nonchimeras.fasta\r\n",
      "SG5-blank-nc\tfasta\t../2-chimera_detection/SG5-blank/SG5-blank-nonchimeras.fasta\r\n",
      "SG5-S1-nc\tfasta\t../2-chimera_detection/SG5-S1/SG5-S1-nonchimeras.fasta\r\n",
      "SG5-S2-nc\tfasta\t../2-chimera_detection/SG5-S2/SG5-S2-nonchimeras.fasta\r\n",
      "SG5-S3-nc\tfasta\t../2-chimera_detection/SG5-S3/SG5-S3-nonchimeras.fasta\r\n",
      "SG5-S4-nc\tfasta\t../2-chimera_detection/SG5-S4/SG5-S4-nonchimeras.fasta\r\n",
      "SG5-S5-nc\tfasta\t../2-chimera_detection/SG5-S5/SG5-S5-nonchimeras.fasta\r\n",
      "SG6-blank-nc\tfasta\t../2-chimera_detection/SG6-blank/SG6-blank-nonchimeras.fasta\r\n",
      "SG6-S1-nc\tfasta\t../2-chimera_detection/SG6-S1/SG6-S1-nonchimeras.fasta\r\n",
      "SG6-S2-nc\tfasta\t../2-chimera_detection/SG6-S2/SG6-S2-nonchimeras.fasta\r\n",
      "SG6-S3-nc\tfasta\t../2-chimera_detection/SG6-S3/SG6-S3-nonchimeras.fasta\r\n",
      "SG6-S4-nc\tfasta\t../2-chimera_detection/SG6-S4/SG6-S4-nonchimeras.fasta\r\n",
      "SG6-S5-nc\tfasta\t../2-chimera_detection/SG6-S5/SG6-S5-nonchimeras.fasta\r\n",
      "SG7-blank-nc\tfasta\t../2-chimera_detection/SG7-blank/SG7-blank-nonchimeras.fasta\r\n",
      "SG7-S1-nc\tfasta\t../2-chimera_detection/SG7-S1/SG7-S1-nonchimeras.fasta\r\n",
      "SG7-S2-nc\tfasta\t../2-chimera_detection/SG7-S2/SG7-S2-nonchimeras.fasta\r\n",
      "SG7-S3-nc\tfasta\t../2-chimera_detection/SG7-S3/SG7-S3-nonchimeras.fasta\r\n",
      "SG7-S4-nc\tfasta\t../2-chimera_detection/SG7-S4/SG7-S4-nonchimeras.fasta\r\n",
      "SG7-S5-nc\tfasta\t../2-chimera_detection/SG7-S5/SG7-S5-nonchimeras.fasta\r\n",
      "SG8-blank-nc\tfasta\t../2-chimera_detection/SG8-blank/SG8-blank-nonchimeras.fasta\r\n",
      "SG8-S1-nc\tfasta\t../2-chimera_detection/SG8-S1/SG8-S1-nonchimeras.fasta\r\n",
      "SG8-S2-nc\tfasta\t../2-chimera_detection/SG8-S2/SG8-S2-nonchimeras.fasta\r\n",
      "SG8-S3-nc\tfasta\t../2-chimera_detection/SG8-S3/SG8-S3-nonchimeras.fasta\r\n",
      "SG8-S4-nc\tfasta\t../2-chimera_detection/SG8-S4/SG8-S4-nonchimeras.fasta\r\n",
      "SG8-S5-nc\tfasta\t../2-chimera_detection/SG8-S5/SG8-S5-nonchimeras.fasta\r\n",
      "SG9-blank-nc\tfasta\t../2-chimera_detection/SG9-blank/SG9-blank-nonchimeras.fasta\r\n",
      "SG9-S1-nc\tfasta\t../2-chimera_detection/SG9-S1/SG9-S1-nonchimeras.fasta\r\n",
      "SG9-S2-nc\tfasta\t../2-chimera_detection/SG9-S2/SG9-S2-nonchimeras.fasta\r\n",
      "SG9-S3-nc\tfasta\t../2-chimera_detection/SG9-S3/SG9-S3-nonchimeras.fasta\r\n",
      "SG9-S4-nc\tfasta\t../2-chimera_detection/SG9-S4/SG9-S4-nonchimeras.fasta\r\n",
      "SG9-S5-nc\tfasta\t../2-chimera_detection/SG9-S5/SG9-S5-nonchimeras.fasta\r\n",
      "T-01-nc\tfasta\t../2-chimera_detection/T-01/T-01-nonchimeras.fasta\r\n",
      "T-02-nc\tfasta\t../2-chimera_detection/T-02/T-02-nonchimeras.fasta\r\n",
      "T-03-nc\tfasta\t../2-chimera_detection/T-03/T-03-nonchimeras.fasta\r\n",
      "T-04-nc\tfasta\t../2-chimera_detection/T-04/T-04-nonchimeras.fasta\r\n",
      "T-05-nc\tfasta\t../2-chimera_detection/T-05/T-05-nonchimeras.fasta\r\n",
      "T-B1-nc\tfasta\t../2-chimera_detection/T-B1/T-B1-nonchimeras.fasta\r\n",
      "W-01-nc\tfasta\t../2-chimera_detection/W-01/W-01-nonchimeras.fasta\r\n",
      "W-02-nc\tfasta\t../2-chimera_detection/W-02/W-02-nonchimeras.fasta\r\n",
      "W-03-nc\tfasta\t../2-chimera_detection/W-03/W-03-nonchimeras.fasta\r\n",
      "W-04-nc\tfasta\t../2-chimera_detection/W-04/W-04-nonchimeras.fasta\r\n",
      "W-05-nc\tfasta\t../2-chimera_detection/W-05/W-05-nonchimeras.fasta\r\n",
      "W-06-nc\tfasta\t../2-chimera_detection/W-06/W-06-nonchimeras.fasta\r\n",
      "W-07-nc\tfasta\t../2-chimera_detection/W-07/W-07-nonchimeras.fasta\r\n",
      "W-08-nc\tfasta\t../2-chimera_detection/W-08/W-08-nonchimeras.fasta\r\n",
      "W-09-nc\tfasta\t../2-chimera_detection/W-09/W-09-nonchimeras.fasta\r\n",
      "W-10-nc\tfasta\t../2-chimera_detection/W-10/W-10-nonchimeras.fasta\r\n",
      "W11-blank-nc\tfasta\t../2-chimera_detection/W11-blank/W11-blank-nonchimeras.fasta\r\n",
      "W11-S1-nc\tfasta\t../2-chimera_detection/W11-S1/W11-S1-nonchimeras.fasta\r\n",
      "W11-S2-nc\tfasta\t../2-chimera_detection/W11-S2/W11-S2-nonchimeras.fasta\r\n",
      "W11-S3-nc\tfasta\t../2-chimera_detection/W11-S3/W11-S3-nonchimeras.fasta\r\n",
      "W11-S4-nc\tfasta\t../2-chimera_detection/W11-S4/W11-S4-nonchimeras.fasta\r\n",
      "W11-S5-nc\tfasta\t../2-chimera_detection/W11-S5/W11-S5-nonchimeras.fasta\r\n",
      "W13-blank-nc\tfasta\t../2-chimera_detection/W13-blank/W13-blank-nonchimeras.fasta\r\n",
      "W13-S1-nc\tfasta\t../2-chimera_detection/W13-S1/W13-S1-nonchimeras.fasta\r\n",
      "W13-S2-nc\tfasta\t../2-chimera_detection/W13-S2/W13-S2-nonchimeras.fasta\r\n",
      "W13-S3-nc\tfasta\t../2-chimera_detection/W13-S3/W13-S3-nonchimeras.fasta\r\n",
      "W13-S4-nc\tfasta\t../2-chimera_detection/W13-S4/W13-S4-nonchimeras.fasta\r\n",
      "W13-S5-nc\tfasta\t../2-chimera_detection/W13-S5/W13-S5-nonchimeras.fasta\r\n",
      "W14-blank-nc\tfasta\t../2-chimera_detection/W14-blank/W14-blank-nonchimeras.fasta\r\n",
      "W14-S1-nc\tfasta\t../2-chimera_detection/W14-S1/W14-S1-nonchimeras.fasta\r\n",
      "W14-S2-nc\tfasta\t../2-chimera_detection/W14-S2/W14-S2-nonchimeras.fasta\r\n",
      "W14-S3-nc\tfasta\t../2-chimera_detection/W14-S3/W14-S3-nonchimeras.fasta\r\n",
      "W14-S4-nc\tfasta\t../2-chimera_detection/W14-S4/W14-S4-nonchimeras.fasta\r\n",
      "W14-S5-nc\tfasta\t../2-chimera_detection/W14-S5/W14-S5-nonchimeras.fasta\r\n",
      "W15-blank-nc\tfasta\t../2-chimera_detection/W15-blank/W15-blank-nonchimeras.fasta\r\n",
      "W15-S1-nc\tfasta\t../2-chimera_detection/W15-S1/W15-S1-nonchimeras.fasta\r\n",
      "W15-S2-nc\tfasta\t../2-chimera_detection/W15-S2/W15-S2-nonchimeras.fasta\r\n",
      "W15-S3-nc\tfasta\t../2-chimera_detection/W15-S3/W15-S3-nonchimeras.fasta\r\n",
      "W15-S4-nc\tfasta\t../2-chimera_detection/W15-S4/W15-S4-nonchimeras.fasta\r\n",
      "W15-S5-nc\tfasta\t../2-chimera_detection/W15-S5/W15-S5-nonchimeras.fasta\r\n",
      "W16-blank-nc\tfasta\t../2-chimera_detection/W16-blank/W16-blank-nonchimeras.fasta\r\n",
      "W16-S1-nc\tfasta\t../2-chimera_detection/W16-S1/W16-S1-nonchimeras.fasta\r\n",
      "W16-S2-nc\tfasta\t../2-chimera_detection/W16-S2/W16-S2-nonchimeras.fasta\r\n",
      "W16-S3-nc\tfasta\t../2-chimera_detection/W16-S3/W16-S3-nonchimeras.fasta\r\n",
      "W16-S4-nc\tfasta\t../2-chimera_detection/W16-S4/W16-S4-nonchimeras.fasta\r\n",
      "W16-S5-nc\tfasta\t../2-chimera_detection/W16-S5/W16-S5-nonchimeras.fasta\r\n",
      "W17-blank-nc\tfasta\t../2-chimera_detection/W17-blank/W17-blank-nonchimeras.fasta\r\n",
      "W17-S1-nc\tfasta\t../2-chimera_detection/W17-S1/W17-S1-nonchimeras.fasta\r\n",
      "W17-S2-nc\tfasta\t../2-chimera_detection/W17-S2/W17-S2-nonchimeras.fasta\r\n",
      "W17-S3-nc\tfasta\t../2-chimera_detection/W17-S3/W17-S3-nonchimeras.fasta\r\n",
      "W17-S4-nc\tfasta\t../2-chimera_detection/W17-S4/W17-S4-nonchimeras.fasta\r\n",
      "W17-S5-nc\tfasta\t../2-chimera_detection/W17-S5/W17-S5-nonchimeras.fasta\r\n",
      "W18-blank-nc\tfasta\t../2-chimera_detection/W18-blank/W18-blank-nonchimeras.fasta\r\n",
      "W18-S1-nc\tfasta\t../2-chimera_detection/W18-S1/W18-S1-nonchimeras.fasta\r\n",
      "W18-S2-nc\tfasta\t../2-chimera_detection/W18-S2/W18-S2-nonchimeras.fasta\r\n",
      "W18-S3-nc\tfasta\t../2-chimera_detection/W18-S3/W18-S3-nonchimeras.fasta\r\n",
      "W18-S4-nc\tfasta\t../2-chimera_detection/W18-S4/W18-S4-nonchimeras.fasta\r\n",
      "W18-S5-nc\tfasta\t../2-chimera_detection/W18-S5/W18-S5-nonchimeras.fasta\r\n",
      "W19-blank-nc\tfasta\t../2-chimera_detection/W19-blank/W19-blank-nonchimeras.fasta\r\n",
      "W19-S1-nc\tfasta\t../2-chimera_detection/W19-S1/W19-S1-nonchimeras.fasta\r\n",
      "W19-S2-nc\tfasta\t../2-chimera_detection/W19-S2/W19-S2-nonchimeras.fasta\r\n",
      "W19-S3-nc\tfasta\t../2-chimera_detection/W19-S3/W19-S3-nonchimeras.fasta\r\n",
      "W19-S4-nc\tfasta\t../2-chimera_detection/W19-S4/W19-S4-nonchimeras.fasta\r\n",
      "W19-S5-nc\tfasta\t../2-chimera_detection/W19-S5/W19-S5-nonchimeras.fasta\r\n",
      "W20-blank-nc\tfasta\t../2-chimera_detection/W20-blank/W20-blank-nonchimeras.fasta\r\n",
      "W20-S1-nc\tfasta\t../2-chimera_detection/W20-S1/W20-S1-nonchimeras.fasta\r\n",
      "W20-S2-nc\tfasta\t../2-chimera_detection/W20-S2/W20-S2-nonchimeras.fasta\r\n",
      "W20-S3-nc\tfasta\t../2-chimera_detection/W20-S3/W20-S3-nonchimeras.fasta\r\n",
      "W20-S4-nc\tfasta\t../2-chimera_detection/W20-S4/W20-S4-nonchimeras.fasta\r\n",
      "W20-S5-nc\tfasta\t../2-chimera_detection/W20-S5/W20-S5-nonchimeras.fasta\r\n",
      "W27-blank-nc\tfasta\t../2-chimera_detection/W27-blank/W27-blank-nonchimeras.fasta\r\n",
      "W27-S1-nc\tfasta\t../2-chimera_detection/W27-S1/W27-S1-nonchimeras.fasta\r\n",
      "W27-S2-nc\tfasta\t../2-chimera_detection/W27-S2/W27-S2-nonchimeras.fasta\r\n",
      "W27-S3-nc\tfasta\t../2-chimera_detection/W27-S3/W27-S3-nonchimeras.fasta\r\n",
      "W27-S4-nc\tfasta\t../2-chimera_detection/W27-S4/W27-S4-nonchimeras.fasta\r\n",
      "W27-S5-nc\tfasta\t../2-chimera_detection/W27-S5/W27-S5-nonchimeras.fasta\r\n",
      "W-B1-nc\tfasta\t../2-chimera_detection/W-B1/W-B1-nonchimeras.fasta\r\n"
     ]
    }
   ],
   "source": [
    "!cat Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Querymap.txt file has been made but includes the `./GLOBAL` directory in which all centroids and queries are contained. This will cause metaBEAT to fail so it must be removed manually from the `Querymap.txt` file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!sed '/GLOBAL/d' Querymap.txt > Querymap_final.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24-blank-nc\tfasta\t../2-chimera_detection/24-blank/24-blank-nonchimeras.fasta\r\n",
      "24-S1-nc\tfasta\t../2-chimera_detection/24-S1/24-S1-nonchimeras.fasta\r\n",
      "24-S2-nc\tfasta\t../2-chimera_detection/24-S2/24-S2-nonchimeras.fasta\r\n",
      "24-S3-nc\tfasta\t../2-chimera_detection/24-S3/24-S3-nonchimeras.fasta\r\n",
      "24-S4-nc\tfasta\t../2-chimera_detection/24-S4/24-S4-nonchimeras.fasta\r\n",
      "24-S5-nc\tfasta\t../2-chimera_detection/24-S5/24-S5-nonchimeras.fasta\r\n",
      "25-blank-nc\tfasta\t../2-chimera_detection/25-blank/25-blank-nonchimeras.fasta\r\n",
      "25-S1-nc\tfasta\t../2-chimera_detection/25-S1/25-S1-nonchimeras.fasta\r\n",
      "25-S2-nc\tfasta\t../2-chimera_detection/25-S2/25-S2-nonchimeras.fasta\r\n",
      "25-S3-nc\tfasta\t../2-chimera_detection/25-S3/25-S3-nonchimeras.fasta\r\n",
      "25-S4-nc\tfasta\t../2-chimera_detection/25-S4/25-S4-nonchimeras.fasta\r\n",
      "25-S5-nc\tfasta\t../2-chimera_detection/25-S5/25-S5-nonchimeras.fasta\r\n",
      "26-blank-nc\tfasta\t../2-chimera_detection/26-blank/26-blank-nonchimeras.fasta\r\n",
      "26-S1-nc\tfasta\t../2-chimera_detection/26-S1/26-S1-nonchimeras.fasta\r\n",
      "26-S2-nc\tfasta\t../2-chimera_detection/26-S2/26-S2-nonchimeras.fasta\r\n",
      "26-S3-nc\tfasta\t../2-chimera_detection/26-S3/26-S3-nonchimeras.fasta\r\n",
      "26-S4-nc\tfasta\t../2-chimera_detection/26-S4/26-S4-nonchimeras.fasta\r\n",
      "26-S5-nc\tfasta\t../2-chimera_detection/26-S5/26-S5-nonchimeras.fasta\r\n",
      "BL3-blank-nc\tfasta\t../2-chimera_detection/BL3-blank/BL3-blank-nonchimeras.fasta\r\n",
      "BL3-S1-nc\tfasta\t../2-chimera_detection/BL3-S1/BL3-S1-nonchimeras.fasta\r\n",
      "BL3-S2-nc\tfasta\t../2-chimera_detection/BL3-S2/BL3-S2-nonchimeras.fasta\r\n",
      "BL3-S3-nc\tfasta\t../2-chimera_detection/BL3-S3/BL3-S3-nonchimeras.fasta\r\n",
      "BL3-S5-nc\tfasta\t../2-chimera_detection/BL3-S5/BL3-S5-nonchimeras.fasta\r\n",
      "BL4-blank-nc\tfasta\t../2-chimera_detection/BL4-blank/BL4-blank-nonchimeras.fasta\r\n",
      "BL4-S1-nc\tfasta\t../2-chimera_detection/BL4-S1/BL4-S1-nonchimeras.fasta\r\n",
      "BL4-S2-nc\tfasta\t../2-chimera_detection/BL4-S2/BL4-S2-nonchimeras.fasta\r\n",
      "BL4-S3-nc\tfasta\t../2-chimera_detection/BL4-S3/BL4-S3-nonchimeras.fasta\r\n",
      "BL4-S4-nc\tfasta\t../2-chimera_detection/BL4-S4/BL4-S4-nonchimeras.fasta\r\n",
      "BL4-S5-nc\tfasta\t../2-chimera_detection/BL4-S5/BL4-S5-nonchimeras.fasta\r\n",
      "C-01-nc\tfasta\t../2-chimera_detection/C-01/C-01-nonchimeras.fasta\r\n",
      "C-02-nc\tfasta\t../2-chimera_detection/C-02/C-02-nonchimeras.fasta\r\n",
      "C-03-nc\tfasta\t../2-chimera_detection/C-03/C-03-nonchimeras.fasta\r\n",
      "C-04-nc\tfasta\t../2-chimera_detection/C-04/C-04-nonchimeras.fasta\r\n",
      "C-05-nc\tfasta\t../2-chimera_detection/C-05/C-05-nonchimeras.fasta\r\n",
      "D-01-nc\tfasta\t../2-chimera_detection/D-01/D-01-nonchimeras.fasta\r\n",
      "D-02-nc\tfasta\t../2-chimera_detection/D-02/D-02-nonchimeras.fasta\r\n",
      "D-03-nc\tfasta\t../2-chimera_detection/D-03/D-03-nonchimeras.fasta\r\n",
      "D-04-nc\tfasta\t../2-chimera_detection/D-04/D-04-nonchimeras.fasta\r\n",
      "D-05-nc\tfasta\t../2-chimera_detection/D-05/D-05-nonchimeras.fasta\r\n",
      "D-B1-nc\tfasta\t../2-chimera_detection/D-B1/D-B1-nonchimeras.fasta\r\n",
      "LB-01-nc\tfasta\t../2-chimera_detection/LB-01/LB-01-nonchimeras.fasta\r\n",
      "LB-02-nc\tfasta\t../2-chimera_detection/LB-02/LB-02-nonchimeras.fasta\r\n",
      "LB-03-nc\tfasta\t../2-chimera_detection/LB-03/LB-03-nonchimeras.fasta\r\n",
      "LB-04-nc\tfasta\t../2-chimera_detection/LB-04/LB-04-nonchimeras.fasta\r\n",
      "LB-05-nc\tfasta\t../2-chimera_detection/LB-05/LB-05-nonchimeras.fasta\r\n",
      "LB-06-nc\tfasta\t../2-chimera_detection/LB-06/LB-06-nonchimeras.fasta\r\n",
      "LB-07-nc\tfasta\t../2-chimera_detection/LB-07/LB-07-nonchimeras.fasta\r\n",
      "LB-08-nc\tfasta\t../2-chimera_detection/LB-08/LB-08-nonchimeras.fasta\r\n",
      "LB-09-nc\tfasta\t../2-chimera_detection/LB-09/LB-09-nonchimeras.fasta\r\n",
      "LB-10-nc\tfasta\t../2-chimera_detection/LB-10/LB-10-nonchimeras.fasta\r\n",
      "LB-B1-nc\tfasta\t../2-chimera_detection/LB-B1/LB-B1-nonchimeras.fasta\r\n",
      "LH-01-nc\tfasta\t../2-chimera_detection/LH-01/LH-01-nonchimeras.fasta\r\n",
      "LH-02-nc\tfasta\t../2-chimera_detection/LH-02/LH-02-nonchimeras.fasta\r\n",
      "LH-03-nc\tfasta\t../2-chimera_detection/LH-03/LH-03-nonchimeras.fasta\r\n",
      "LH-04-nc\tfasta\t../2-chimera_detection/LH-04/LH-04-nonchimeras.fasta\r\n",
      "LH-05-nc\tfasta\t../2-chimera_detection/LH-05/LH-05-nonchimeras.fasta\r\n",
      "N-01-nc\tfasta\t../2-chimera_detection/N-01/N-01-nonchimeras.fasta\r\n",
      "N-02-nc\tfasta\t../2-chimera_detection/N-02/N-02-nonchimeras.fasta\r\n",
      "N-03-nc\tfasta\t../2-chimera_detection/N-03/N-03-nonchimeras.fasta\r\n",
      "N-04-nc\tfasta\t../2-chimera_detection/N-04/N-04-nonchimeras.fasta\r\n",
      "N-05-nc\tfasta\t../2-chimera_detection/N-05/N-05-nonchimeras.fasta\r\n",
      "N+C-B1-nc\tfasta\t../2-chimera_detection/N+C-B1/N+C-B1-nonchimeras.fasta\r\n",
      "neg1-nc\tfasta\t../2-chimera_detection/neg1/neg1-nonchimeras.fasta\r\n",
      "neg2-nc\tfasta\t../2-chimera_detection/neg2/neg2-nonchimeras.fasta\r\n",
      "neg3-nc\tfasta\t../2-chimera_detection/neg3/neg3-nonchimeras.fasta\r\n",
      "neg4-nc\tfasta\t../2-chimera_detection/neg4/neg4-nonchimeras.fasta\r\n",
      "neg5-nc\tfasta\t../2-chimera_detection/neg5/neg5-nonchimeras.fasta\r\n",
      "PH-01-nc\tfasta\t../2-chimera_detection/PH-01/PH-01-nonchimeras.fasta\r\n",
      "PH-02-nc\tfasta\t../2-chimera_detection/PH-02/PH-02-nonchimeras.fasta\r\n",
      "PH-03-nc\tfasta\t../2-chimera_detection/PH-03/PH-03-nonchimeras.fasta\r\n",
      "PH-04-nc\tfasta\t../2-chimera_detection/PH-04/PH-04-nonchimeras.fasta\r\n",
      "PH-05-nc\tfasta\t../2-chimera_detection/PH-05/PH-05-nonchimeras.fasta\r\n",
      "PH-06-nc\tfasta\t../2-chimera_detection/PH-06/PH-06-nonchimeras.fasta\r\n",
      "PH-07-nc\tfasta\t../2-chimera_detection/PH-07/PH-07-nonchimeras.fasta\r\n",
      "PH-08-nc\tfasta\t../2-chimera_detection/PH-08/PH-08-nonchimeras.fasta\r\n",
      "PH-09-nc\tfasta\t../2-chimera_detection/PH-09/PH-09-nonchimeras.fasta\r\n",
      "PH-10-nc\tfasta\t../2-chimera_detection/PH-10/PH-10-nonchimeras.fasta\r\n",
      "PH-B1-nc\tfasta\t../2-chimera_detection/PH-B1/PH-B1-nonchimeras.fasta\r\n",
      "pos1-nc\tfasta\t../2-chimera_detection/pos1/pos1-nonchimeras.fasta\r\n",
      "pos2-nc\tfasta\t../2-chimera_detection/pos2/pos2-nonchimeras.fasta\r\n",
      "pos3-nc\tfasta\t../2-chimera_detection/pos3/pos3-nonchimeras.fasta\r\n",
      "pos4-nc\tfasta\t../2-chimera_detection/pos4/pos4-nonchimeras.fasta\r\n",
      "pos5-nc\tfasta\t../2-chimera_detection/pos5/pos5-nonchimeras.fasta\r\n",
      "S-01-nc\tfasta\t../2-chimera_detection/S-01/S-01-nonchimeras.fasta\r\n",
      "S-02-nc\tfasta\t../2-chimera_detection/S-02/S-02-nonchimeras.fasta\r\n",
      "S-03-nc\tfasta\t../2-chimera_detection/S-03/S-03-nonchimeras.fasta\r\n",
      "S-04-nc\tfasta\t../2-chimera_detection/S-04/S-04-nonchimeras.fasta\r\n",
      "S-05-nc\tfasta\t../2-chimera_detection/S-05/S-05-nonchimeras.fasta\r\n",
      "SG10-blank-nc\tfasta\t../2-chimera_detection/SG10-blank/SG10-blank-nonchimeras.fasta\r\n",
      "SG10-S1-nc\tfasta\t../2-chimera_detection/SG10-S1/SG10-S1-nonchimeras.fasta\r\n",
      "SG10-S2-nc\tfasta\t../2-chimera_detection/SG10-S2/SG10-S2-nonchimeras.fasta\r\n",
      "SG10-S3-nc\tfasta\t../2-chimera_detection/SG10-S3/SG10-S3-nonchimeras.fasta\r\n",
      "SG10-S4-nc\tfasta\t../2-chimera_detection/SG10-S4/SG10-S4-nonchimeras.fasta\r\n",
      "SG10-S5-nc\tfasta\t../2-chimera_detection/SG10-S5/SG10-S5-nonchimeras.fasta\r\n",
      "SG1-blank-nc\tfasta\t../2-chimera_detection/SG1-blank/SG1-blank-nonchimeras.fasta\r\n",
      "SG1-S1-nc\tfasta\t../2-chimera_detection/SG1-S1/SG1-S1-nonchimeras.fasta\r\n",
      "SG1-S2-nc\tfasta\t../2-chimera_detection/SG1-S2/SG1-S2-nonchimeras.fasta\r\n",
      "SG1-S3-nc\tfasta\t../2-chimera_detection/SG1-S3/SG1-S3-nonchimeras.fasta\r\n",
      "SG1-S4-nc\tfasta\t../2-chimera_detection/SG1-S4/SG1-S4-nonchimeras.fasta\r\n",
      "SG1-S5-nc\tfasta\t../2-chimera_detection/SG1-S5/SG1-S5-nonchimeras.fasta\r\n",
      "SG21-blank-nc\tfasta\t../2-chimera_detection/SG21-blank/SG21-blank-nonchimeras.fasta\r\n",
      "SG21-S1-nc\tfasta\t../2-chimera_detection/SG21-S1/SG21-S1-nonchimeras.fasta\r\n",
      "SG21-S2-nc\tfasta\t../2-chimera_detection/SG21-S2/SG21-S2-nonchimeras.fasta\r\n",
      "SG21-S3-nc\tfasta\t../2-chimera_detection/SG21-S3/SG21-S3-nonchimeras.fasta\r\n",
      "SG21-S4-nc\tfasta\t../2-chimera_detection/SG21-S4/SG21-S4-nonchimeras.fasta\r\n",
      "SG22-blank-nc\tfasta\t../2-chimera_detection/SG22-blank/SG22-blank-nonchimeras.fasta\r\n",
      "SG22-S1-nc\tfasta\t../2-chimera_detection/SG22-S1/SG22-S1-nonchimeras.fasta\r\n",
      "SG22-S2-nc\tfasta\t../2-chimera_detection/SG22-S2/SG22-S2-nonchimeras.fasta\r\n",
      "SG22-S3-nc\tfasta\t../2-chimera_detection/SG22-S3/SG22-S3-nonchimeras.fasta\r\n",
      "SG22-S4-nc\tfasta\t../2-chimera_detection/SG22-S4/SG22-S4-nonchimeras.fasta\r\n",
      "SG22-S5-nc\tfasta\t../2-chimera_detection/SG22-S5/SG22-S5-nonchimeras.fasta\r\n",
      "SG23-blank-nc\tfasta\t../2-chimera_detection/SG23-blank/SG23-blank-nonchimeras.fasta\r\n",
      "SG23-S1-nc\tfasta\t../2-chimera_detection/SG23-S1/SG23-S1-nonchimeras.fasta\r\n",
      "SG23-S2-nc\tfasta\t../2-chimera_detection/SG23-S2/SG23-S2-nonchimeras.fasta\r\n",
      "SG23-S3-nc\tfasta\t../2-chimera_detection/SG23-S3/SG23-S3-nonchimeras.fasta\r\n",
      "SG23-S4-nc\tfasta\t../2-chimera_detection/SG23-S4/SG23-S4-nonchimeras.fasta\r\n",
      "SG23-S5-nc\tfasta\t../2-chimera_detection/SG23-S5/SG23-S5-nonchimeras.fasta\r\n",
      "SG2-blank-nc\tfasta\t../2-chimera_detection/SG2-blank/SG2-blank-nonchimeras.fasta\r\n",
      "SG2-S1-nc\tfasta\t../2-chimera_detection/SG2-S1/SG2-S1-nonchimeras.fasta\r\n",
      "SG2-S2-nc\tfasta\t../2-chimera_detection/SG2-S2/SG2-S2-nonchimeras.fasta\r\n",
      "SG2-S3-nc\tfasta\t../2-chimera_detection/SG2-S3/SG2-S3-nonchimeras.fasta\r\n",
      "SG2-S4-nc\tfasta\t../2-chimera_detection/SG2-S4/SG2-S4-nonchimeras.fasta\r\n",
      "SG2-S5-nc\tfasta\t../2-chimera_detection/SG2-S5/SG2-S5-nonchimeras.fasta\r\n",
      "SG5-blank-nc\tfasta\t../2-chimera_detection/SG5-blank/SG5-blank-nonchimeras.fasta\r\n",
      "SG5-S1-nc\tfasta\t../2-chimera_detection/SG5-S1/SG5-S1-nonchimeras.fasta\r\n",
      "SG5-S2-nc\tfasta\t../2-chimera_detection/SG5-S2/SG5-S2-nonchimeras.fasta\r\n",
      "SG5-S3-nc\tfasta\t../2-chimera_detection/SG5-S3/SG5-S3-nonchimeras.fasta\r\n",
      "SG5-S4-nc\tfasta\t../2-chimera_detection/SG5-S4/SG5-S4-nonchimeras.fasta\r\n",
      "SG5-S5-nc\tfasta\t../2-chimera_detection/SG5-S5/SG5-S5-nonchimeras.fasta\r\n",
      "SG6-blank-nc\tfasta\t../2-chimera_detection/SG6-blank/SG6-blank-nonchimeras.fasta\r\n",
      "SG6-S1-nc\tfasta\t../2-chimera_detection/SG6-S1/SG6-S1-nonchimeras.fasta\r\n",
      "SG6-S2-nc\tfasta\t../2-chimera_detection/SG6-S2/SG6-S2-nonchimeras.fasta\r\n",
      "SG6-S3-nc\tfasta\t../2-chimera_detection/SG6-S3/SG6-S3-nonchimeras.fasta\r\n",
      "SG6-S4-nc\tfasta\t../2-chimera_detection/SG6-S4/SG6-S4-nonchimeras.fasta\r\n",
      "SG6-S5-nc\tfasta\t../2-chimera_detection/SG6-S5/SG6-S5-nonchimeras.fasta\r\n",
      "SG7-blank-nc\tfasta\t../2-chimera_detection/SG7-blank/SG7-blank-nonchimeras.fasta\r\n",
      "SG7-S1-nc\tfasta\t../2-chimera_detection/SG7-S1/SG7-S1-nonchimeras.fasta\r\n",
      "SG7-S2-nc\tfasta\t../2-chimera_detection/SG7-S2/SG7-S2-nonchimeras.fasta\r\n",
      "SG7-S3-nc\tfasta\t../2-chimera_detection/SG7-S3/SG7-S3-nonchimeras.fasta\r\n",
      "SG7-S4-nc\tfasta\t../2-chimera_detection/SG7-S4/SG7-S4-nonchimeras.fasta\r\n",
      "SG7-S5-nc\tfasta\t../2-chimera_detection/SG7-S5/SG7-S5-nonchimeras.fasta\r\n",
      "SG8-blank-nc\tfasta\t../2-chimera_detection/SG8-blank/SG8-blank-nonchimeras.fasta\r\n",
      "SG8-S1-nc\tfasta\t../2-chimera_detection/SG8-S1/SG8-S1-nonchimeras.fasta\r\n",
      "SG8-S2-nc\tfasta\t../2-chimera_detection/SG8-S2/SG8-S2-nonchimeras.fasta\r\n",
      "SG8-S3-nc\tfasta\t../2-chimera_detection/SG8-S3/SG8-S3-nonchimeras.fasta\r\n",
      "SG8-S4-nc\tfasta\t../2-chimera_detection/SG8-S4/SG8-S4-nonchimeras.fasta\r\n",
      "SG8-S5-nc\tfasta\t../2-chimera_detection/SG8-S5/SG8-S5-nonchimeras.fasta\r\n",
      "SG9-blank-nc\tfasta\t../2-chimera_detection/SG9-blank/SG9-blank-nonchimeras.fasta\r\n",
      "SG9-S1-nc\tfasta\t../2-chimera_detection/SG9-S1/SG9-S1-nonchimeras.fasta\r\n",
      "SG9-S2-nc\tfasta\t../2-chimera_detection/SG9-S2/SG9-S2-nonchimeras.fasta\r\n",
      "SG9-S3-nc\tfasta\t../2-chimera_detection/SG9-S3/SG9-S3-nonchimeras.fasta\r\n",
      "SG9-S4-nc\tfasta\t../2-chimera_detection/SG9-S4/SG9-S4-nonchimeras.fasta\r\n",
      "SG9-S5-nc\tfasta\t../2-chimera_detection/SG9-S5/SG9-S5-nonchimeras.fasta\r\n",
      "T-01-nc\tfasta\t../2-chimera_detection/T-01/T-01-nonchimeras.fasta\r\n",
      "T-02-nc\tfasta\t../2-chimera_detection/T-02/T-02-nonchimeras.fasta\r\n",
      "T-03-nc\tfasta\t../2-chimera_detection/T-03/T-03-nonchimeras.fasta\r\n",
      "T-04-nc\tfasta\t../2-chimera_detection/T-04/T-04-nonchimeras.fasta\r\n",
      "T-05-nc\tfasta\t../2-chimera_detection/T-05/T-05-nonchimeras.fasta\r\n",
      "T-B1-nc\tfasta\t../2-chimera_detection/T-B1/T-B1-nonchimeras.fasta\r\n",
      "W-01-nc\tfasta\t../2-chimera_detection/W-01/W-01-nonchimeras.fasta\r\n",
      "W-02-nc\tfasta\t../2-chimera_detection/W-02/W-02-nonchimeras.fasta\r\n",
      "W-03-nc\tfasta\t../2-chimera_detection/W-03/W-03-nonchimeras.fasta\r\n",
      "W-04-nc\tfasta\t../2-chimera_detection/W-04/W-04-nonchimeras.fasta\r\n",
      "W-05-nc\tfasta\t../2-chimera_detection/W-05/W-05-nonchimeras.fasta\r\n",
      "W-06-nc\tfasta\t../2-chimera_detection/W-06/W-06-nonchimeras.fasta\r\n",
      "W-07-nc\tfasta\t../2-chimera_detection/W-07/W-07-nonchimeras.fasta\r\n",
      "W-08-nc\tfasta\t../2-chimera_detection/W-08/W-08-nonchimeras.fasta\r\n",
      "W-09-nc\tfasta\t../2-chimera_detection/W-09/W-09-nonchimeras.fasta\r\n",
      "W-10-nc\tfasta\t../2-chimera_detection/W-10/W-10-nonchimeras.fasta\r\n",
      "W11-blank-nc\tfasta\t../2-chimera_detection/W11-blank/W11-blank-nonchimeras.fasta\r\n",
      "W11-S1-nc\tfasta\t../2-chimera_detection/W11-S1/W11-S1-nonchimeras.fasta\r\n",
      "W11-S2-nc\tfasta\t../2-chimera_detection/W11-S2/W11-S2-nonchimeras.fasta\r\n",
      "W11-S3-nc\tfasta\t../2-chimera_detection/W11-S3/W11-S3-nonchimeras.fasta\r\n",
      "W11-S4-nc\tfasta\t../2-chimera_detection/W11-S4/W11-S4-nonchimeras.fasta\r\n",
      "W11-S5-nc\tfasta\t../2-chimera_detection/W11-S5/W11-S5-nonchimeras.fasta\r\n",
      "W13-blank-nc\tfasta\t../2-chimera_detection/W13-blank/W13-blank-nonchimeras.fasta\r\n",
      "W13-S1-nc\tfasta\t../2-chimera_detection/W13-S1/W13-S1-nonchimeras.fasta\r\n",
      "W13-S2-nc\tfasta\t../2-chimera_detection/W13-S2/W13-S2-nonchimeras.fasta\r\n",
      "W13-S3-nc\tfasta\t../2-chimera_detection/W13-S3/W13-S3-nonchimeras.fasta\r\n",
      "W13-S4-nc\tfasta\t../2-chimera_detection/W13-S4/W13-S4-nonchimeras.fasta\r\n",
      "W13-S5-nc\tfasta\t../2-chimera_detection/W13-S5/W13-S5-nonchimeras.fasta\r\n",
      "W14-blank-nc\tfasta\t../2-chimera_detection/W14-blank/W14-blank-nonchimeras.fasta\r\n",
      "W14-S1-nc\tfasta\t../2-chimera_detection/W14-S1/W14-S1-nonchimeras.fasta\r\n",
      "W14-S2-nc\tfasta\t../2-chimera_detection/W14-S2/W14-S2-nonchimeras.fasta\r\n",
      "W14-S3-nc\tfasta\t../2-chimera_detection/W14-S3/W14-S3-nonchimeras.fasta\r\n",
      "W14-S4-nc\tfasta\t../2-chimera_detection/W14-S4/W14-S4-nonchimeras.fasta\r\n",
      "W14-S5-nc\tfasta\t../2-chimera_detection/W14-S5/W14-S5-nonchimeras.fasta\r\n",
      "W15-blank-nc\tfasta\t../2-chimera_detection/W15-blank/W15-blank-nonchimeras.fasta\r\n",
      "W15-S1-nc\tfasta\t../2-chimera_detection/W15-S1/W15-S1-nonchimeras.fasta\r\n",
      "W15-S2-nc\tfasta\t../2-chimera_detection/W15-S2/W15-S2-nonchimeras.fasta\r\n",
      "W15-S3-nc\tfasta\t../2-chimera_detection/W15-S3/W15-S3-nonchimeras.fasta\r\n",
      "W15-S4-nc\tfasta\t../2-chimera_detection/W15-S4/W15-S4-nonchimeras.fasta\r\n",
      "W15-S5-nc\tfasta\t../2-chimera_detection/W15-S5/W15-S5-nonchimeras.fasta\r\n",
      "W16-blank-nc\tfasta\t../2-chimera_detection/W16-blank/W16-blank-nonchimeras.fasta\r\n",
      "W16-S1-nc\tfasta\t../2-chimera_detection/W16-S1/W16-S1-nonchimeras.fasta\r\n",
      "W16-S2-nc\tfasta\t../2-chimera_detection/W16-S2/W16-S2-nonchimeras.fasta\r\n",
      "W16-S3-nc\tfasta\t../2-chimera_detection/W16-S3/W16-S3-nonchimeras.fasta\r\n",
      "W16-S4-nc\tfasta\t../2-chimera_detection/W16-S4/W16-S4-nonchimeras.fasta\r\n",
      "W16-S5-nc\tfasta\t../2-chimera_detection/W16-S5/W16-S5-nonchimeras.fasta\r\n",
      "W17-blank-nc\tfasta\t../2-chimera_detection/W17-blank/W17-blank-nonchimeras.fasta\r\n",
      "W17-S1-nc\tfasta\t../2-chimera_detection/W17-S1/W17-S1-nonchimeras.fasta\r\n",
      "W17-S2-nc\tfasta\t../2-chimera_detection/W17-S2/W17-S2-nonchimeras.fasta\r\n",
      "W17-S3-nc\tfasta\t../2-chimera_detection/W17-S3/W17-S3-nonchimeras.fasta\r\n",
      "W17-S4-nc\tfasta\t../2-chimera_detection/W17-S4/W17-S4-nonchimeras.fasta\r\n",
      "W17-S5-nc\tfasta\t../2-chimera_detection/W17-S5/W17-S5-nonchimeras.fasta\r\n",
      "W18-blank-nc\tfasta\t../2-chimera_detection/W18-blank/W18-blank-nonchimeras.fasta\r\n",
      "W18-S1-nc\tfasta\t../2-chimera_detection/W18-S1/W18-S1-nonchimeras.fasta\r\n",
      "W18-S2-nc\tfasta\t../2-chimera_detection/W18-S2/W18-S2-nonchimeras.fasta\r\n",
      "W18-S3-nc\tfasta\t../2-chimera_detection/W18-S3/W18-S3-nonchimeras.fasta\r\n",
      "W18-S4-nc\tfasta\t../2-chimera_detection/W18-S4/W18-S4-nonchimeras.fasta\r\n",
      "W18-S5-nc\tfasta\t../2-chimera_detection/W18-S5/W18-S5-nonchimeras.fasta\r\n",
      "W19-blank-nc\tfasta\t../2-chimera_detection/W19-blank/W19-blank-nonchimeras.fasta\r\n",
      "W19-S1-nc\tfasta\t../2-chimera_detection/W19-S1/W19-S1-nonchimeras.fasta\r\n",
      "W19-S2-nc\tfasta\t../2-chimera_detection/W19-S2/W19-S2-nonchimeras.fasta\r\n",
      "W19-S3-nc\tfasta\t../2-chimera_detection/W19-S3/W19-S3-nonchimeras.fasta\r\n",
      "W19-S4-nc\tfasta\t../2-chimera_detection/W19-S4/W19-S4-nonchimeras.fasta\r\n",
      "W19-S5-nc\tfasta\t../2-chimera_detection/W19-S5/W19-S5-nonchimeras.fasta\r\n",
      "W20-blank-nc\tfasta\t../2-chimera_detection/W20-blank/W20-blank-nonchimeras.fasta\r\n",
      "W20-S1-nc\tfasta\t../2-chimera_detection/W20-S1/W20-S1-nonchimeras.fasta\r\n",
      "W20-S2-nc\tfasta\t../2-chimera_detection/W20-S2/W20-S2-nonchimeras.fasta\r\n",
      "W20-S3-nc\tfasta\t../2-chimera_detection/W20-S3/W20-S3-nonchimeras.fasta\r\n",
      "W20-S4-nc\tfasta\t../2-chimera_detection/W20-S4/W20-S4-nonchimeras.fasta\r\n",
      "W20-S5-nc\tfasta\t../2-chimera_detection/W20-S5/W20-S5-nonchimeras.fasta\r\n",
      "W27-blank-nc\tfasta\t../2-chimera_detection/W27-blank/W27-blank-nonchimeras.fasta\r\n",
      "W27-S1-nc\tfasta\t../2-chimera_detection/W27-S1/W27-S1-nonchimeras.fasta\r\n",
      "W27-S2-nc\tfasta\t../2-chimera_detection/W27-S2/W27-S2-nonchimeras.fasta\r\n",
      "W27-S3-nc\tfasta\t../2-chimera_detection/W27-S3/W27-S3-nonchimeras.fasta\r\n",
      "W27-S4-nc\tfasta\t../2-chimera_detection/W27-S4/W27-S4-nonchimeras.fasta\r\n",
      "W27-S5-nc\tfasta\t../2-chimera_detection/W27-S5/W27-S5-nonchimeras.fasta\r\n",
      "W-B1-nc\tfasta\t../2-chimera_detection/W-B1/W-B1-nonchimeras.fasta\r\n"
     ]
    }
   ],
   "source": [
    "!cat Querymap_final.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Update the taxonomy database in the current metaBEAT image."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!taxit new_database \\\n",
    "--taxdump-url ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2018-10-01.zip \\\n",
    "    --download-dir /usr/bin/ /usr/bin/taxonomy.db"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "That's almost it. Now start the pipeline to do sequence clustering and taxonomic assignment of non-chimera queries via metaBEAT. As input, `Querymap.txt` containing samples that have been trimmed, merged and checked for chimeras, and the `REFmap.txt` file must be specified. metaBEAT will be asked to attempt taxonomic assignment using BLAST.\n",
    "\n",
    "metaBEAT will automatically wrangle the data into the particular file formats that are required by each of the methods, run all necessary steps, and finally convert the outputs of each program to a standardized BIOM table.\n",
    "\n",
    "GO!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: metaBEAT.py [-h] [-Q <FILE>] [-B <FILE>] [--g_queries <FILE>] [-v] [-s]\r\n",
      "                   [-f] [-p] [-k] [-t] [-b] [-m <string>] [-n <INT>] [-E] [-e]\r\n",
      "                   [--read_stats_off] [--PCR_primer <FILE>] [--bc_dist <INT>]\r\n",
      "                   [--trim_adapter <FILE>] [--trim_qual <INT>] [--phred <INT>]\r\n",
      "                   [--trim_window <INT>] [--read_crop <INT>]\r\n",
      "                   [--trim_minlength <INT>] [--merge] [--product_length <INT>]\r\n",
      "                   [--merged_only] [--forward_only] [--length_filter <INT>]\r\n",
      "                   [--length_deviation <FLOAT>] [-R <FILE>] [--gb_out <FILE>]\r\n",
      "                   [--rec_check] [--gb_to_taxid <FILE>] [--cluster]\r\n",
      "                   [--clust_match <FLOAT>] [--clust_cov <INT>]\r\n",
      "                   [--blast_db <PATH>] [--blast_xml <PATH>]\r\n",
      "                   [--update_taxonomy] [--taxonomy_db <FILE>]\r\n",
      "                   [--min_ident <FLOAT>] [--min_ali_length <FLOAT>]\r\n",
      "                   [--bitscore_skim_LCA <FLOAT>] [--bitscore_skim_adjust_off]\r\n",
      "                   [--min_bit <INT>] [--refpkg <DIR>] [--jplace <FILE>]\r\n",
      "                   [--jellyfish_hash_size <STR>] [--kraken_db <DIR>]\r\n",
      "                   [--kraken_score_threshold <FLOAT>] [--rm_kraken_db]\r\n",
      "                   [-o OUTPUT_PREFIX] [--metadata METADATA]\r\n",
      "                   [-@ <email-address>] [--version]\r\n",
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool version:\r\n",
      "v.0.97.11-global\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -Q <FILE>, --querylist <FILE>\r\n",
      "                        file containing a list of query files\r\n",
      "  -B <FILE>, --BIOM_input <FILE>\r\n",
      "                        OTU table in BIOM format\r\n",
      "  --g_queries <FILE>    fasta file containing query sequences (in combination\r\n",
      "                        with '-B' sequence headers are expected to match)\r\n",
      "  -v, --verbose         turn verbose output on\r\n",
      "  -s, --seqinfo         write out seq_info.csv file\r\n",
      "  -f, --fasta           write out ref.fasta file\r\n",
      "  -p, --pplace          perform phylogenetic placement\r\n",
      "  -k, --kraken          perform phylogenetic placement\r\n",
      "  -t, --taxids          write out taxid.txt file\r\n",
      "  -b, --blast           compile local blast db and blast queries\r\n",
      "  -m <string>, --marker <string>\r\n",
      "                        marker ID (default: marker)\r\n",
      "  -n <INT>, --n_threads <INT>\r\n",
      "                        Number of threads (default: 1)\r\n",
      "  -E, --extract_centroid_reads\r\n",
      "                        extract centroid reads to files\r\n",
      "  -e, --extract_all_reads\r\n",
      "                        extract reads to files\r\n",
      "  --read_stats_off      ommit writing read stats to file\r\n",
      "  --version             show program's version number and exit\r\n",
      "\r\n",
      "Query preprocessing:\r\n",
      "  The parameters in this group affect how the query sequences are processed\r\n",
      "\r\n",
      "  --PCR_primer <FILE>   PCR primers (provided in fasta file) to be clipped\r\n",
      "                        from reads\r\n",
      "  --bc_dist <INT>       Number of mismatches allowed in barcode sequences\r\n",
      "  --trim_adapter <FILE>\r\n",
      "                        trim adapters provided in file\r\n",
      "  --trim_qual <INT>     minimum phred quality score (default: 30)\r\n",
      "  --phred <INT>         phred quality score offset - 33 or 64 (default: 33)\r\n",
      "  --trim_window <INT>   sliding window size (default: 5) for trimming; if\r\n",
      "                        average quality drops below the specified minimum\r\n",
      "                        quality all subsequent bases are removed from the\r\n",
      "                        reads\r\n",
      "  --read_crop <INT>     Crop reads to this length if they are longer than that\r\n",
      "                        (default: off)\r\n",
      "  --trim_minlength <INT>\r\n",
      "                        minimum length of reads to be retained after trimming\r\n",
      "                        (default: 50)\r\n",
      "  --merge               attempt to merge paired-end reads\r\n",
      "  --product_length <INT>\r\n",
      "                        estimated length of PCR product (specifying this\r\n",
      "                        option increases merging efficiency)\r\n",
      "  --merged_only         only process successfully merged read-pairs\r\n",
      "  --forward_only        only process sequences that contain forward reads\r\n",
      "                        (i.e. unmerged forward reads and merged reads)\r\n",
      "  --length_filter <INT>\r\n",
      "                        only process reads, which are within +/- 10 percent of\r\n",
      "                        this length\r\n",
      "  --length_deviation <FLOAT>\r\n",
      "                        allowed deviation (in percent) from length specified\r\n",
      "                        by --length_filter (default=0.1)\r\n",
      "\r\n",
      "Reference:\r\n",
      "  The parameters in this group affect the reference to be used in the\r\n",
      "  analyses\r\n",
      "\r\n",
      "  -R <FILE>, --REFlist <FILE>\r\n",
      "                        file containing a list of files to be used as\r\n",
      "                        reference sequences\r\n",
      "  --gb_out <FILE>       output the corrected gb file\r\n",
      "  --rec_check           check records to be used as reference\r\n",
      "  --gb_to_taxid <FILE>  comma delimited file containing 'gb accession,taxid'\r\n",
      "                        for a list of taxa\r\n",
      "\r\n",
      "Query clustering options:\r\n",
      "  The parameters in this group affect read clustering\r\n",
      "\r\n",
      "  --cluster             perform clustering of query sequences using vsearch\r\n",
      "  --clust_match <FLOAT>\r\n",
      "                        identity threshold for clustering in percent (default:\r\n",
      "                        1)\r\n",
      "  --clust_cov <INT>     minimum number of records in cluster (default: 1)\r\n",
      "\r\n",
      "BLAST search:\r\n",
      "  The parameters in this group affect BLAST search and BLAST based taxonomic\r\n",
      "  assignment\r\n",
      "\r\n",
      "  --blast_db <PATH>     path to precompiled blast database\r\n",
      "  --blast_xml <PATH>    path to Blast result in xml format\r\n",
      "  --update_taxonomy     Download/update taxonomy database. Database will be\r\n",
      "                        called 'taxonomy.db' and will be compiled in the same\r\n",
      "                        location as the metaBEAT.py script.\r\n",
      "  --taxonomy_db <FILE>  taxonomy database file location. In case it's not the\r\n",
      "                        default, which is 'taxonomy.db' in the same directory\r\n",
      "                        as the metaBEAT.py script.\r\n",
      "  --min_ident <FLOAT>   minimum identity threshold in percent (default: 0.80)\r\n",
      "  --min_ali_length <FLOAT>\r\n",
      "                        minimum alignment length in percent of total query\r\n",
      "                        length (default: 0.95)\r\n",
      "  --bitscore_skim_LCA <FLOAT>\r\n",
      "                        Only BLAST hits with bitscores differing by less than\r\n",
      "                        this factor from the top hit (bitscore skim window)\r\n",
      "                        will be considered for LCA (0-1; default: 0.1)\r\n",
      "  --bitscore_skim_adjust_off\r\n",
      "                        Per default a 100% identity BLAST top hit across the\r\n",
      "                        minimum alignment length triggers an adjustment of the\r\n",
      "                        bitscore skim window to '0', i.e. only hits with\r\n",
      "                        bitscores as good as the top hit are considered for\r\n",
      "                        LCA. This flag switches this behaviour off.\r\n",
      "  --min_bit <INT>       minimum bitscore (default: 80)\r\n",
      "\r\n",
      "Phylogenetic placement:\r\n",
      "  The parameters in this group affect phylogenetic placement\r\n",
      "\r\n",
      "  --refpkg <DIR>        PATH to refpkg for pplacer\r\n",
      "  --jplace <FILE>       phylogenetic placement result from prefious pplacer\r\n",
      "                        run in *.jplace format\r\n",
      "\r\n",
      "Kraken:\r\n",
      "  The parameters in this group affect taxonomic assignment using Kraken\r\n",
      "\r\n",
      "  --jellyfish_hash_size <STR>\r\n",
      "                        jellyfish hash size to control memory usage during\r\n",
      "                        kraken database building. A table size of '6400M' will\r\n",
      "                        require ~44G of RAM. '2700M' -> 20G RAM.\r\n",
      "  --kraken_db <DIR>     PATH to a Kraken database\r\n",
      "  --kraken_score_threshold <FLOAT>\r\n",
      "                        minimum proportion of k-mers to support assignment\r\n",
      "                        (0-1; default: 0)\r\n",
      "  --rm_kraken_db        Remove Kraken database after successful completion\r\n",
      "\r\n",
      "BIOM OUTPUT:\r\n",
      "  The arguments in this groups affect the output in BIOM format\r\n",
      "\r\n",
      "  -o OUTPUT_PREFIX, --output_prefix OUTPUT_PREFIX\r\n",
      "                        prefix for BIOM output files (default='metaBEAT')\r\n",
      "  --metadata METADATA   comma delimited file containing metadata (optional)\r\n",
      "\r\n",
      "Entrez identification:\r\n",
      "  metaBEAT is querying the NCBI Entrez databases, please provide an email\r\n",
      "  address for identification\r\n",
      "\r\n",
      "  -@ <email-address>, --email <email-address>\r\n",
      "                        provide your email address for identification to NCBI\r\n"
     ]
    }
   ],
   "source": [
    "!metaBEAT_global.py -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starttime: Sat Jul  4 20:36:09 UTC 2020\n",
      "\n",
      "Endtime: Sat Jul  4 20:55:40 UTC 2020\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "echo -e \"Starttime: $(date)\\n\"\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-Q Querymap_final.txt \\\n",
    "-R REFmap.txt \\\n",
    "--cluster --clust_match 1 --clust_cov 3 \\\n",
    "--blast --min_ident 0.98 \\\n",
    "-m 12S -n 5 \\\n",
    "-E -v \\\n",
    "-@ nathangriffiths484@gmail.com \\\n",
    "-o Eel2017_12S-trim30-min90-crop110-mergeforwonly-filt100-dev0.2_nonchimera_c1cov3_blast0.98_fish &> log_fish\n",
    "\n",
    "echo -e \"Endtime: $(date)\\n\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "attempting LCA assignment for W20-S1-nc|M04282:74:000000000-BLMMB:1:1108:11991:13655_ex\r\n",
      "found LCA 7953 at level family\r\n",
      "assigned LCA Cyprinidae (taxid 7953) at level family\r\n",
      "\r\n",
      "attempting LCA assignment for LH-05-nc|M04282:74:000000000-BLMMB:1:1108:9831:20874_ex\r\n",
      "found LCA 69291 at level family\r\n",
      "assigned LCA Gasterosteidae (taxid 69291) at level family\r\n",
      "\r\n",
      "direct assignment for pos1-nc|M04282:74:000000000-BLMMB:1:1101:15422:9455_ex -> 106582\r\n",
      "\r\n",
      "attempting LCA assignment for 25-S4-nc|M04282:74:000000000-BLMMB:1:1102:26065:10135_ex\r\n",
      "found LCA 57865 at level species\r\n",
      "assigned LCA Gymnocephalus cernua (taxid 57865) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for W16-S5-nc|M04282:74:000000000-BLMMB:1:1107:23001:16333_ex\r\n",
      "found LCA 7953 at level family\r\n",
      "assigned LCA Cyprinidae (taxid 7953) at level family\r\n",
      "\r\n",
      "attempting LCA assignment for T-01-nc|M04282:74:000000000-BLMMB:1:1106:19495:16556_ex\r\n",
      "found LCA 69291 at level family\r\n",
      "assigned LCA Gasterosteidae (taxid 69291) at level family\r\n",
      "\r\n",
      "attempting LCA assignment for 24-S1-nc|M04282:74:000000000-BLMMB:1:2114:8054:13324_ex\r\n",
      "found LCA 8010 at level species\r\n",
      "assigned LCA Esox lucius (taxid 8010) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for SG7-S5-nc|M04282:74:000000000-BLMMB:1:1117:22150:2449_ex\r\n",
      "found LCA 7953 at level family\r\n",
      "assigned LCA Cyprinidae (taxid 7953) at level family\r\n",
      "\r\n",
      "all queries have been successfully assigned to a taxonomy\r\n",
      "\r\n",
      "\r\n",
      "##### FORMATTING AND WRITING BIOM OUTPUT #####\r\n",
      "\r\n",
      "\r\n",
      "##### BLAST ANALYSIS DONE #####\r\n",
      "\r\n",
      "Find result tables in '/home/working/3-taxonomic_assignment_fish/GLOBAL/BLAST_0.98'\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "Sat Jul  4 20:55:40 2020\r\n",
      "\r\n",
      "\r\n",
      "##### DONE! #####\r\n",
      "\r\n",
      "\r\n",
      "Sat Jul  4 20:55:40 2020\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!tail -n 50 log_fish"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**DONE, Output file under GLOBAL/BLAST_0.98"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
